// xero-extension/xero-proto.js
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var paths_1 = require('./paths');
var XeroProto = (function (_super) {
    __extends(XeroProto, _super);
    function XeroProto(config, endpoint) {
        _super.call(this, config, endpoint);
        this.endpoint = endpoint;
    }
    XeroProto.prototype.findById = function (id, opts) {
        if (opts === void 0) { opts = {}; }
        opts.singular = true;
        return _super.prototype.get.call(this, "/" + this.endpoint + "/" + id, opts);
    };
    XeroProto.prototype.findAll = function (opts) {
        if (opts === void 0) { opts = {}; }
        return _super.prototype.get.call(this, "/" + this.endpoint, opts);
    };
    XeroProto.prototype.create = function (body, opts) {
        if (opts === void 0) { opts = {}; }
        opts.singular = true;
        return _super.prototype.put.call(this, "/" + this.endpoint, body);
    };
    XeroProto.prototype.bulkCreate = function (body, opts) {
        if (opts === void 0) { opts = {}; }
        return _super.prototype.put.call(this, "/" + this.endpoint + "?summarizeErrors=false", body, opts);
    };
    XeroProto.prototype.createOrUpdate = function (body, opts) {
        if (opts === void 0) { opts = {}; }
        opts.singular = true;
        return _super.prototype.put.call(this, "/" + this.endpoint, body, opts);
    };
    XeroProto.prototype.bulkCreateOrUpdate = function (body, opts) {
        if (opts === void 0) { opts = {}; }
        return _super.prototype.post.call(this, "/" + this.endpoint, body, opts);
    };
    return XeroProto;
})(paths_1.Methods);
exports.XeroProto = XeroProto;
