// generator-hughes/generators/app/mixins/save-package/writing.js
/**
 * Save package.json
 */

'use strict';

var mkdirp      = require('mkdirp'),
    fs          = require('fs'),
    util        = require('util'),
    jsonfile    = require('jsonfile'),
    createJson  = require('./../../helpers/createJson');


module.exports = {
	savePackage: function() {	
			var packageJson = {
				name: this.projectNameJson,
				version: this.projectVersion,
				description: this.projectDescription,
				authors: [
					this.projectAuthor
				],
	//       homepage: this.authorURI,
				repository: {
	//         type: 'git',
	//         url: 'https://github.com/michaeldeboeve/generator-robonkey'
				},
				main: 'gulpfile.js',
				dependencies: {},
				devDependencies: {}
			}
			
			var cssBase = this.cssBase;


		// Not optional

				packageJson.dependencies["autoprefixer"] =  "~6.3.3";
				packageJson.dependencies["autoprefixer-core"] =  "^6.0.1";
				packageJson.dependencies["del"] =  "^2.1.0";
				packageJson.dependencies["graceful-fs"] =  "^4.1.3";
				packageJson.dependencies["css-modularscale"] = "^1.0.0";

				packageJson.dependencies["gulp"] =  "~3.9.0";
				packageJson.dependencies["gulp-autoprefixer"] =  "^3.1.0";
				packageJson.dependencies["gulp-concat"] =  "^2.6.0";
				packageJson.dependencies["gulp-cssmin"] =  "^0.1.7";
				packageJson.dependencies["gulp-cwebp"] =  "^1.0.1";
				packageJson.dependencies["gulp-htmlmin"] =  "^1.2.0";
				packageJson.dependencies["gulp-imagemin"] =  "^2.4.0";
				packageJson.dependencies["gulp-jade-php"] =  "^2.0.0-0";
				packageJson.dependencies["gulp-newer"] =  "^1.0.0";
				packageJson.dependencies["gulp-postcss"] =  "^6.0.1";
				packageJson.dependencies["gulp-rename"] =  "^1.2.2";
				packageJson.dependencies["gulp-replace"] =  "^0.5.4";
				packageJson.dependencies["gulp-ssh"] =  "^0.5.5";
				packageJson.dependencies["gulp-shell"] =  "^0.5.2";
				packageJson.dependencies["gulp-uglify"] =  "^1.5.1";
				packageJson.dependencies["gulp-util"] =  "^3.0.7";
				packageJson.dependencies["gulp-watch"] =  "^4.3.5";
				packageJson.dependencies["gulp-concat"] =  "^2.6.0";
				packageJson.dependencies["gulp-uglify"] =  "^1.5.1";
				packageJson.dependencies["gulp-rename"] =  "^1.2.2";
				packageJson.dependencies["imagemin-pngquant"] =  "^4.2.0";
				packageJson.dependencies["lost"] =  "^8.0.0";
				
				// CSS Next
				packageJson.dependencies["postcss"] =  "^5.0.16";
				packageJson.dependencies["postcss-color-gray"] =  "^3.0.0";
				packageJson.dependencies["postcss-cssnext"] =  "~2.4.0";
				packageJson.dependencies["postcss-custom-media"] =  "^5.0.1";
				packageJson.dependencies["postcss-custom-properties"] =  "^5.0.0";
				packageJson.dependencies["postcss-custom-selectors"] =  "^3.0.0";
				packageJson.dependencies["postcss-import"] =  "^7.1.3";
				packageJson.dependencies["postcss-mixins"] =  "^4.0.0";
				packageJson.dependencies["postcss-neat"] =  "^2.4.1";
				packageJson.dependencies["postcss-nested"] =  "^1.0.0";
				packageJson.dependencies["postcss-object-fit-images"] =  "^1.1.2";
				
				// Browserify
				packageJson.dependencies["browserify"] =  "^14.4.0";
				packageJson.dependencies["babelify"] =  "^7.3.0";
				packageJson.dependencies["vinyl-source-stream"] =  "^1.1.0";
				//  Components
				packageJson.dependencies["swiper"] =  "^3.3.1";
				packageJson.dependencies["require-dir"] =  "^0.3.0";
				packageJson.dependencies["smoothscroll-for-websites"] =  "^1.4.4"
				
		// CSS Base
		if (cssBase === 'Normalize') {
			packageJson.dependencies["normalize-css"] =  "^2.3.1";
		}
		
		if (cssBase === 'Reset') {
			packageJson.dependencies["reset.css"] =  "^2.0.2";
		}


		// Write File
			var packageFile = './package.json';
			createJson(packageFile, packageJson);

	}}
