// generator-hughes/generators/app/mixins/project/prompting.js
'use strict';

var yeoman = require('yeoman-generator'),
    chalk = require('chalk'),
    yosay = require('yosay'),
		printTitle  = require('./../../helpers/printTitle'),
    fs = require('graceful-fs');

module.exports = {
	project: function() {
		
		this.log(printTitle('Project Information'))
		
		var done = this.async();
		this.prompt([{
				// Project Name
				name: 'projectName',
				message: 'What is the name of this project?',
				default: 'My super cool project'
			}, {
				// Project version
				name: 'projectVersion',
				message: 'What is the project version?',
				default: '0.0.0'
			}, {
				// Project description
				name: 'projectDescription',
				message: 'A little bit about the project',
				default: 'This project is wicked awesome, bud'
			}, {
				// Author
				name: 'projectAuthor',
				message: 'Who is the project author?',
				default: 'John Doe'
			}], 

		function(props){
			this.props = props;
			// To access props later use this.props.someAnswer;
			
			this.projectName = this.props.projectName;
			this.projectNameJson = this.projectName.replace(/\s/g,'-').toLowerCase();
			
			this.projectVersion = this.props.projectVersion;
			this.projectDescription = this.props.projectDescription;
			this.projectAuthor = this.props.projectAuthor;
			done();
		}.bind(this))
	}};
