// formidable/lib/json_parser_jalangi_.js
J$.iids = {"8":[3,5,3,18],"9":[1,20,1,27],"10":[16,29,16,62],"16":[16,9,16,62],"17":[1,28,1,32],"18":[16,9,16,62],"25":[1,20,1,33],"33":[1,20,1,33],"41":[1,20,1,33],"49":[2,26,2,38],"57":[2,26,2,45],"65":[2,26,2,45],"73":[2,26,2,45],"81":[3,5,3,11],"89":[3,5,3,18],"97":[4,15,4,21],"105":[4,29,4,36],"113":[4,15,4,37],"115":[4,15,4,28],"121":[4,15,4,37],"129":[4,5,4,38],"137":[5,14,5,21],"145":[5,22,5,30],"153":[5,14,5,31],"161":[5,14,5,38],"169":[5,14,5,38],"177":[5,14,5,38],"185":[7,5,7,9],"193":[7,21,7,27],"201":[7,28,7,30],"209":[7,17,7,31],"217":[7,5,7,31],"225":[7,5,7,32],"233":[8,5,8,9],"241":[8,25,8,26],"249":[8,5,8,26],"257":[8,5,8,27],"265":[6,1,9,2],"273":[6,1,9,2],"281":[11,1,11,8],"289":[11,22,11,32],"297":[11,1,11,32],"305":[11,1,11,33],"313":[12,1,12,11],"321":[12,1,12,21],"329":[13,5,13,9],"337":[13,21,13,27],"345":[13,28,13,34],"353":[13,17,13,35],"361":[13,5,13,35],"369":[13,5,13,36],"377":[12,39,14,2],"385":[12,39,14,2],"393":[12,39,14,2],"401":[12,39,14,2],"409":[12,1,14,2],"417":[12,1,14,3],"425":[15,1,15,11],"433":[15,1,15,21],"441":[16,9,16,13],"449":[16,9,16,18],"457":[16,9,16,25],"465":[16,29,16,33],"473":[16,29,16,46],"481":[16,49,16,55],"489":[16,49,16,62],"497":[17,9,17,15],"505":[17,21,17,25],"513":[17,21,17,30],"521":[17,32,17,36],"529":[17,32,17,49],"537":[17,9,17,50],"539":[17,9,17,20],"545":[17,9,17,51],"553":[19,9,19,13],"561":[19,21,19,27],"569":[20,13,20,17],"577":[20,13,20,22],"585":[21,13,21,19],"593":[19,35,22,10],"601":[19,21,22,11],"603":[19,21,19,34],"609":[19,9,22,11],"617":[19,9,22,12],"625":[24,5,24,9],"633":[24,26,24,32],"641":[24,26,24,39],"649":[24,5,24,39],"651":[24,5,24,39],"657":[24,5,24,40],"665":[25,12,25,18],"673":[25,12,25,25],"681":[25,12,25,25],"689":[25,5,25,26],"697":[15,30,26,2],"705":[15,30,26,2],"713":[15,30,26,2],"721":[15,30,26,2],"729":[15,1,26,2],"737":[15,1,26,3],"745":[27,1,27,11],"753":[27,1,27,21],"761":[29,22,29,26],"769":[29,33,29,37],"777":[29,33,29,42],"785":[29,52,29,58],"793":[29,33,29,59],"795":[29,33,29,51],"801":[29,22,29,60],"803":[29,22,29,32],"809":[29,22,29,60],"817":[29,22,29,60],"825":[30,27,30,33],"833":[31,13,31,17],"841":[31,26,31,31],"849":[31,33,31,39],"857":[31,40,31,45],"865":[31,33,31,46],"873":[31,13,31,47],"875":[31,13,31,25],"881":[31,13,31,48],"889":[30,27,30,33],"897":[30,9,32,10],"905":[30,9,32,10],"913":[33,7,34,6],"921":[35,5,35,9],"929":[35,17,35,21],"937":[35,5,35,21],"945":[35,5,35,22],"953":[36,5,36,9],"961":[36,5,36,17],"963":[36,5,36,15],"969":[36,5,36,18],"977":[27,28,37,2],"985":[27,28,37,2],"993":[27,28,37,2],"1001":[27,28,37,2],"1009":[27,28,37,2],"1017":[27,1,37,2],"1025":[27,1,37,3],"1033":[1,1,37,3],"1041":[1,1,37,3],"1049":[1,1,37,3],"1057":[1,1,37,3],"1065":[6,1,9,2],"1073":[1,1,37,3],"1081":[3,1,4,38],"1089":[6,1,9,2],"1097":[6,1,9,2],"1105":[12,39,14,2],"1113":[12,39,14,2],"1121":[16,5,23,6],"1129":[15,30,26,2],"1137":[15,30,26,2],"1145":[27,28,37,2],"1153":[27,28,37,2],"1161":[1,1,37,3],"1169":[1,1,37,3],"nBranches":4,"originalCodeFileName":"/home/blake/ExpoSE/Targets/mvvm/node_modules/formidable/lib/json_parser.js","instrumentedCodeFileName":"/home/blake/ExpoSE/Targets/mvvm/node_modules/formidable/lib/json_parser_jalangi_.js","code":"if (global.GENTLY) require = GENTLY.hijack(require);\n\nvar Buffer = require('buffer').Buffer\n\nfunction JSONParser() {\n  this.data = new Buffer('');\n  this.bytesWritten = 0;\n};\nexports.JSONParser = JSONParser;\n\nJSONParser.prototype.initWithLength = function(length) {\n  this.data = new Buffer(length);\n}\n\nJSONParser.prototype.write = function(buffer) {\n  if (this.data.length >= this.bytesWritten + buffer.length) {\n    buffer.copy(this.data, this.bytesWritten);\n  } else {\n    this.data = Buffer.concat([this.data, buffer]);\n  }\n  this.bytesWritten += buffer.length;\n  return buffer.length;\n}\n\nJSONParser.prototype.end = function() {\n  try {\n    var fields = JSON.parse(this.data.toString('utf8'))\n    for (var field in fields) {\n      this.onField(field, fields[field]);\n    }\n  } catch (e) {}\n  this.data = null;\n\n  this.onEnd();\n}"};
jalangiLabel262:
    while (true) {
        try {
            J$.Se(1033, '/home/blake/ExpoSE/Targets/mvvm/node_modules/formidable/lib/json_parser_jalangi_.js', '/home/blake/ExpoSE/Targets/mvvm/node_modules/formidable/lib/json_parser.js');
            function JSONParser() {
                jalangiLabel258:
                    while (true) {
                        try {
                            J$.Fe(265, arguments.callee, this, arguments);
                            arguments = J$.N(273, 'arguments', arguments, 4);
                            J$.X1(225, J$.P(217, J$.R(185, 'this', this, 0), 'data', J$.F(209, J$.R(193, 'Buffer', Buffer, 1), 1)(J$.T(201, '', 21, false)), 0));
                            J$.X1(257, J$.P(249, J$.R(233, 'this', this, 0), 'bytesWritten', J$.T(241, 0, 22, false), 0));
                        } catch (J$e) {
                            J$.Ex(1089, J$e);
                        } finally {
                            if (J$.Fr(1097))
                                continue jalangiLabel258;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(1041, '__secret__S$', __secret__S$, 0);
            J$.N(1049, '__secret__traits__', __secret__traits__, 0);
            J$.N(1057, 'Buffer', Buffer, 0);
            JSONParser = J$.N(1073, 'JSONParser', J$.T(1065, JSONParser, 12, false, 265), 0);
            var __secret__S$ = J$.X1(41, J$.W(33, '__secret__S$', J$.F(25, J$.R(9, 'require', require, 2), 0)(J$.T(17, 'S$', 21, false)), __secret__S$, 3));
            var __secret__traits__ = J$.X1(73, J$.W(65, '__secret__traits__', J$.G(57, J$.R(49, '__secret__S$', __secret__S$, 1), 'Traits', 0), __secret__traits__, 3));
            if (J$.X1(1081, J$.C(8, J$.G(89, J$.R(81, 'global', global, 2), 'GENTLY', 0))))
                J$.X1(129, require = J$.W(121, 'require', J$.M(113, J$.R(97, 'GENTLY', GENTLY, 2), 'hijack', 0)(J$.R(105, 'require', require, 2)), J$.I(typeof require === 'undefined' ? undefined : require), 4));
            var Buffer = J$.X1(177, J$.W(169, 'Buffer', J$.G(161, J$.F(153, J$.R(137, 'require', require, 2), 0)(J$.T(145, 'buffer', 21, false)), 'Buffer', 0), Buffer, 3));
            ;
            J$.X1(305, J$.P(297, J$.R(281, 'exports', exports, 2), 'JSONParser', J$.R(289, 'JSONParser', JSONParser, 1), 0));
            J$.X1(417, J$.P(409, J$.G(321, J$.R(313, 'JSONParser', JSONParser, 1), 'prototype', 0), 'initWithLength', J$.T(401, function (length) {
                jalangiLabel259:
                    while (true) {
                        try {
                            J$.Fe(377, arguments.callee, this, arguments);
                            arguments = J$.N(385, 'arguments', arguments, 4);
                            length = J$.N(393, 'length', length, 4);
                            J$.X1(369, J$.P(361, J$.R(329, 'this', this, 0), 'data', J$.F(353, J$.R(337, 'Buffer', Buffer, 1), 1)(J$.R(345, 'length', length, 0)), 0));
                        } catch (J$e) {
                            J$.Ex(1105, J$e);
                        } finally {
                            if (J$.Fr(1113))
                                continue jalangiLabel259;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 377), 0));
            J$.X1(737, J$.P(729, J$.G(433, J$.R(425, 'JSONParser', JSONParser, 1), 'prototype', 0), 'write', J$.T(721, function (buffer) {
                jalangiLabel260:
                    while (true) {
                        try {
                            J$.Fe(697, arguments.callee, this, arguments);
                            arguments = J$.N(705, 'arguments', arguments, 4);
                            buffer = J$.N(713, 'buffer', buffer, 4);
                            if (J$.X1(1121, J$.C(16, J$.B(18, '>=', J$.G(457, J$.G(449, J$.R(441, 'this', this, 0), 'data', 0), 'length', 0), J$.B(10, '+', J$.G(473, J$.R(465, 'this', this, 0), 'bytesWritten', 0), J$.G(489, J$.R(481, 'buffer', buffer, 0), 'length', 0), 0), 0)))) {
                                J$.X1(545, J$.M(537, J$.R(497, 'buffer', buffer, 0), 'copy', 0)(J$.G(513, J$.R(505, 'this', this, 0), 'data', 0), J$.G(529, J$.R(521, 'this', this, 0), 'bytesWritten', 0)));
                            } else {
                                J$.X1(617, J$.P(609, J$.R(553, 'this', this, 0), 'data', J$.M(601, J$.R(561, 'Buffer', Buffer, 1), 'concat', 0)(J$.T(593, [
                                    J$.G(577, J$.R(569, 'this', this, 0), 'data', 0),
                                    J$.R(585, 'buffer', buffer, 0)
                                ], 10, false)), 0));
                            }
                            J$.X1(657, J$.A(649, J$.R(625, 'this', this, 0), 'bytesWritten', '+', 0)(J$.G(641, J$.R(633, 'buffer', buffer, 0), 'length', 0)));
                            return J$.X1(689, J$.Rt(681, J$.G(673, J$.R(665, 'buffer', buffer, 0), 'length', 0)));
                        } catch (J$e) {
                            J$.Ex(1129, J$e);
                        } finally {
                            if (J$.Fr(1137))
                                continue jalangiLabel260;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 697), 0));
            J$.X1(1025, J$.P(1017, J$.G(753, J$.R(745, 'JSONParser', JSONParser, 1), 'prototype', 0), 'end', J$.T(1009, function () {
                jalangiLabel261:
                    while (true) {
                        try {
                            J$.Fe(977, arguments.callee, this, arguments);
                            arguments = J$.N(985, 'arguments', arguments, 4);
                            J$.N(993, 'fields', fields, 0);
                            J$.N(1001, 'field', field, 0);
                            try {
                                var fields = J$.X1(817, J$.W(809, 'fields', J$.M(801, J$.R(761, 'JSON', JSON, 2), 'parse', 0)(J$.M(793, J$.G(777, J$.R(769, 'this', this, 0), 'data', 0), 'toString', 0)(J$.T(785, 'utf8', 21, false))), fields, 1));
                                for (J$._tm_p in J$.H(889, J$.R(825, 'fields', fields, 0))) {
                                    var field = J$.X1(905, J$.W(897, 'field', J$._tm_p, field, 1));
                                    {
                                        {
                                            J$.X1(881, J$.M(873, J$.R(833, 'this', this, 0), 'onField', 0)(J$.R(841, 'field', field, 0), J$.G(865, J$.R(849, 'fields', fields, 0), J$.R(857, 'field', field, 0), 4)));
                                        }
                                    }
                                }
                            } catch (e) {
                                e = J$.N(913, 'e', e, 1);
                            }
                            J$.X1(945, J$.P(937, J$.R(921, 'this', this, 0), 'data', J$.T(929, null, 25, false), 0));
                            J$.X1(969, J$.M(961, J$.R(953, 'this', this, 0), 'onEnd', 0)());
                        } catch (J$e) {
                            J$.Ex(1145, J$e);
                        } finally {
                            if (J$.Fr(1153))
                                continue jalangiLabel261;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 977), 0));
        } catch (J$e) {
            J$.Ex(1161, J$e);
        } finally {
            if (J$.Sr(1169)) {
                J$.L();
                continue jalangiLabel262;
            } else {
                J$.L();
                break jalangiLabel262;
            }
        }
    }
// JALANGI DO NOT INSTRUMENT
