// less/lib/less/tree/node_jalangi_.js
J$.iids = {"8":[27,13,27,15],"9":[1,20,1,27],"10":[12,20,12,37],"16":[28,10,28,13],"17":[1,28,1,32],"18":[29,16,29,21],"24":[30,10,30,13],"25":[1,20,1,33],"26":[31,16,31,21],"32":[32,10,32,13],"33":[1,20,1,33],"34":[33,16,33,21],"40":[34,10,34,13],"41":[1,20,1,33],"42":[35,16,35,21],"48":[39,21,39,52],"49":[2,26,2,38],"50":[40,12,40,29],"56":[40,12,40,29],"57":[2,26,2,45],"58":[40,48,40,61],"64":[43,24,43,69],"65":[2,26,2,45],"66":[43,24,43,43],"72":[43,9,43,70],"73":[2,26,2,45],"74":[43,47,43,69],"80":[47,16,47,33],"81":[3,12,4,2],"82":[43,22,43,70],"88":[45,16,45,25],"89":[3,12,4,2],"90":[46,16,46,29],"96":[43,9,43,70],"97":[3,12,4,2],"98":[47,16,47,33],"104":[53,16,53,23],"105":[3,12,4,2],"106":[52,9,52,26],"112":[52,9,52,26],"113":[3,12,4,2],"114":[53,16,53,23],"120":[55,9,55,30],"121":[5,1,5,5],"122":[55,9,55,30],"128":[59,13,59,43],"129":[5,1,5,15],"130":[58,21,58,33],"136":[58,21,58,33],"137":[6,16,6,18],"138":[58,35,58,38],"144":[66,39,66,44],"145":[6,16,6,18],"152":[66,25,66,32],"153":[6,16,6,18],"154":[58,35,58,38],"160":[66,12,66,17],"161":[7,5,7,9],"162":[59,13,59,43],"168":[69,9,69,38],"169":[7,17,7,24],"170":[66,12,66,17],"176":[75,9,75,38],"177":[9,13,9,17],"178":[66,20,66,22],"184":[81,9,81,38],"185":[9,23,9,28],"186":[66,25,66,32],"192":[102,9,102,14],"193":[9,13,9,29],"194":[66,39,66,44],"195":[9,13,9,22],"201":[9,13,9,30],"202":[69,9,69,38],"209":[8,14,10,10],"210":[72,12,72,39],"217":[8,14,10,10],"218":[75,9,75,38],"225":[8,14,10,10],"226":[78,29,78,54],"233":[8,14,10,10],"234":[81,9,81,38],"241":[8,14,10,10],"242":[84,29,84,54],"249":[8,14,10,10],"250":[102,9,102,14],"257":[12,20,12,24],"265":[12,20,12,31],"273":[12,36,12,37],"281":[12,20,12,37],"289":[12,13,12,38],"297":[11,18,13,10],"305":[11,18,13,10],"313":[11,18,13,10],"321":[7,26,14,6],"329":[7,5,14,7],"331":[7,5,7,16],"337":[7,5,14,8],"345":[15,12,15,16],"353":[15,22,15,24],"361":[15,12,15,25],"363":[15,12,15,21],"369":[15,12,15,25],"377":[15,5,15,26],"385":[5,24,16,2],"393":[5,24,16,2],"401":[5,24,16,2],"409":[5,24,16,2],"417":[5,24,16,2],"425":[5,1,16,2],"433":[5,1,16,3],"441":[17,1,17,5],"449":[17,1,17,15],"457":[18,5,18,11],"465":[18,16,18,20],"473":[18,16,18,26],"481":[18,5,18,27],"483":[18,5,18,15],"489":[18,5,18,28],"497":[17,25,19,2],"505":[17,25,19,2],"513":[17,25,19,2],"521":[17,25,19,2],"529":[17,25,19,2],"537":[17,1,19,2],"545":[17,1,19,3],"553":[20,1,20,5],"561":[20,1,20,15],"569":[21,5,21,9],"577":[21,18,21,25],"585":[21,32,21,36],"593":[21,32,21,42],"601":[21,18,21,43],"603":[21,18,21,31],"609":[21,5,21,43],"617":[21,5,21,44],"625":[20,25,22,2],"633":[20,25,22,2],"641":[20,25,22,2],"649":[20,25,22,2],"657":[20,1,22,2],"665":[20,1,22,3],"673":[23,1,23,5],"681":[23,1,23,15],"689":[24,12,24,16],"697":[24,12,24,16],"705":[24,5,24,17],"713":[23,23,25,2],"721":[23,23,25,2],"729":[23,23,25,2],"737":[23,1,25,2],"745":[23,1,25,3],"753":[26,1,26,5],"761":[26,1,26,15],"769":[27,13,27,15],"777":[29,16,29,17],"785":[29,20,29,21],"793":[29,16,29,21],"801":[29,9,29,22],"809":[28,10,28,13],"817":[31,16,31,17],"825":[31,20,31,21],"833":[31,16,31,21],"841":[31,9,31,22],"849":[30,10,30,13],"857":[33,16,33,17],"865":[33,20,33,21],"873":[33,16,33,21],"881":[33,9,33,22],"889":[32,10,32,13],"897":[35,16,35,17],"905":[35,20,35,21],"913":[35,16,35,21],"921":[35,9,35,22],"929":[34,10,34,13],"937":[26,27,37,2],"945":[26,27,37,2],"953":[26,27,37,2],"961":[26,27,37,2],"969":[26,27,37,2],"977":[26,27,37,2],"985":[26,27,37,2],"993":[26,1,37,2],"1001":[26,1,37,3],"1009":[38,1,38,5],"1017":[38,1,38,15],"1025":[39,21,39,28],"1033":[39,32,39,39],"1041":[39,32,39,52],"1049":[39,21,39,52],"1057":[39,21,39,52],"1065":[40,12,40,21],"1073":[40,25,40,29],"1081":[40,32,40,37],"1089":[40,40,40,46],"1097":[40,48,40,53],"1105":[40,56,40,61],"1113":[40,71,40,80],"1121":[40,47,40,81],"1123":[40,47,40,70],"1129":[40,40,40,82],"1137":[40,12,40,82],"1145":[40,5,40,83],"1153":[38,25,41,2],"1161":[38,25,41,2],"1169":[38,25,41,2],"1177":[38,25,41,2],"1185":[38,25,41,2],"1193":[38,25,41,2],"1201":[38,1,41,2],"1209":[38,1,41,3],"1217":[42,1,42,5],"1225":[43,9,43,10],"1233":[43,9,43,18],"1241":[43,24,43,25],"1249":[43,24,43,30],"1257":[43,35,43,43],"1265":[43,47,43,48],"1273":[43,47,43,53],"1281":[43,58,43,69],"1289":[44,16,44,17],"1297":[44,26,44,27],"1305":[44,16,44,28],"1307":[44,16,44,25],"1313":[44,16,44,28],"1321":[44,9,44,29],"1329":[45,16,45,17],"1337":[45,16,45,25],"1345":[46,17,46,18],"1353":[46,27,46,28],"1361":[46,17,46,29],"1363":[46,17,46,26],"1369":[46,16,46,29],"1377":[46,9,46,30],"1385":[47,16,47,17],"1393":[47,16,47,22],"1401":[47,27,47,28],"1409":[47,27,47,33],"1417":[48,16,48,25],"1425":[48,16,48,25],"1433":[48,9,48,26],"1441":[50,9,50,10],"1449":[50,9,50,16],"1457":[50,9,50,16],"1465":[50,5,50,17],"1473":[51,9,51,10],"1481":[51,9,51,16],"1489":[51,9,51,16],"1497":[51,5,51,17],"1505":[52,10,52,15],"1513":[52,24,52,25],"1521":[52,10,52,26],"1523":[52,10,52,23],"1529":[53,16,53,17],"1537":[53,22,53,23],"1545":[53,26,53,27],"1553":[53,30,53,39],"1561":[53,16,53,39],"1569":[53,9,53,40],"1577":[55,9,55,10],"1585":[55,9,55,17],"1593":[55,22,55,23],"1601":[55,22,55,30],"1609":[56,16,56,25],"1617":[56,16,56,25],"1625":[56,9,56,26],"1633":[58,18,58,19],"1641":[58,18,58,19],"1649":[58,18,58,19],"1657":[58,21,58,22],"1665":[58,25,58,26],"1673":[58,25,58,33],"1689":[58,35,58,36],"1697":[58,35,58,38],"1713":[59,13,59,17],"1721":[59,26,59,27],"1729":[59,28,59,29],"1737":[59,26,59,30],"1745":[59,32,59,33],"1753":[59,34,59,35],"1761":[59,32,59,36],"1769":[59,13,59,37],"1771":[59,13,59,25],"1777":[59,42,59,43],"1785":[60,20,60,29],"1793":[60,20,60,29],"1801":[60,13,60,30],"1809":[63,12,63,13],"1817":[63,12,63,13],"1825":[63,5,63,14],"1833":[42,16,64,2],"1841":[42,16,64,2],"1849":[42,16,64,2],"1857":[42,16,64,2],"1865":[42,16,64,2],"1873":[42,16,64,2],"1881":[42,1,64,2],"1889":[42,1,64,3],"1897":[65,1,65,5],"1905":[66,12,66,13],"1913":[66,16,66,17],"1921":[66,21,66,22],"1929":[66,25,66,26],"1937":[66,31,66,32],"1945":[66,35,66,36],"1953":[66,39,66,40],"1961":[66,43,66,44],"1969":[66,47,66,48],"1977":[66,51,66,60],"1985":[66,12,66,60],"1993":[66,5,66,61],"2001":[65,23,67,2],"2009":[65,23,67,2],"2017":[65,23,67,2],"2025":[65,23,67,2],"2033":[65,23,67,2],"2041":[65,1,67,2],"2049":[65,1,67,3],"2057":[68,1,68,5],"2065":[68,1,68,15],"2073":[69,9,69,13],"2081":[69,9,69,30],"2089":[69,34,69,38],"2097":[70,9,70,13],"2105":[70,33,70,34],"2113":[70,9,70,34],"2121":[70,9,70,35],"2129":[72,12,72,16],"2137":[72,12,72,33],"2145":[72,38,72,39],"2153":[72,12,72,39],"2161":[72,5,72,40],"2169":[68,35,73,2],"2177":[68,35,73,2],"2185":[68,35,73,2],"2193":[68,1,73,2],"2201":[68,1,73,3],"2209":[74,1,74,5],"2217":[74,1,74,15],"2225":[75,9,75,13],"2233":[75,9,75,30],"2241":[75,34,75,38],"2249":[76,9,76,13],"2257":[76,33,76,34],"2265":[76,9,76,34],"2273":[76,9,76,35],"2281":[78,5,78,9],"2289":[78,29,78,33],"2297":[78,29,78,50],"2305":[78,53,78,54],"2313":[78,5,78,54],"2321":[78,5,78,55],"2329":[74,37,79,2],"2337":[74,37,79,2],"2345":[74,37,79,2],"2353":[74,1,79,2],"2361":[74,1,79,3],"2369":[80,1,80,5],"2377":[80,1,80,15],"2385":[81,9,81,13],"2393":[81,9,81,30],"2401":[81,34,81,38],"2409":[82,9,82,13],"2417":[82,33,82,34],"2425":[82,9,82,34],"2433":[82,9,82,35],"2441":[84,5,84,9],"2449":[84,29,84,33],"2457":[84,29,84,50],"2465":[84,53,84,54],"2473":[84,5,84,54],"2481":[84,5,84,55],"2489":[80,40,85,2],"2497":[80,40,85,2],"2505":[80,40,85,2],"2513":[80,1,85,2],"2521":[80,1,85,3],"2529":[86,1,86,5],"2537":[86,1,86,15],"2545":[87,5,87,9],"2553":[87,24,87,28],"2561":[87,5,87,28],"2569":[87,5,87,29],"2577":[86,35,88,2],"2585":[86,35,88,2],"2593":[86,35,88,2],"2601":[86,1,88,2],"2609":[86,1,88,3],"2617":[89,1,89,5],"2625":[89,1,89,15],"2633":[90,5,90,9],"2641":[90,24,90,29],"2649":[90,5,90,29],"2657":[90,5,90,30],"2665":[89,37,91,2],"2673":[89,37,91,2],"2681":[89,37,91,2],"2689":[89,1,91,2],"2697":[89,1,91,3],"2705":[92,1,92,5],"2713":[92,1,92,15],"2721":[93,12,93,16],"2729":[93,12,93,28],"2737":[93,12,93,28],"2745":[93,5,93,29],"2753":[92,28,94,2],"2761":[92,28,94,2],"2769":[92,28,94,2],"2777":[92,1,94,2],"2785":[92,1,94,3],"2793":[95,1,95,5],"2801":[95,1,95,15],"2809":[97,27,97,31],"2817":[97,27,97,48],"2825":[98,22,98,26],"2833":[98,22,98,38],"2841":[96,12,99,6],"2849":[96,12,99,6],"2857":[96,5,99,7],"2865":[95,33,100,2],"2873":[95,33,100,2],"2881":[95,33,100,2],"2889":[95,1,100,2],"2897":[95,1,100,3],"2905":[101,1,101,5],"2913":[101,1,101,15],"2921":[102,10,102,14],"2929":[103,9,103,16],"2937":[103,9,103,16],"2945":[105,5,105,9],"2953":[105,29,105,33],"2961":[105,29,105,50],"2969":[105,5,105,50],"2977":[105,5,105,51],"2985":[106,5,106,9],"2993":[106,24,106,28],"3001":[106,24,106,40],"3009":[106,5,106,40],"3017":[106,5,106,41],"3025":[101,37,107,2],"3033":[101,37,107,2],"3041":[101,37,107,2],"3049":[101,37,107,2],"3057":[101,1,107,2],"3065":[101,1,107,3],"3073":[108,1,108,7],"3081":[108,18,108,22],"3089":[108,1,108,22],"3097":[108,1,108,23],"3105":[1,1,108,23],"3113":[1,1,108,23],"3121":[1,1,108,23],"3129":[1,1,108,23],"3137":[3,12,4,2],"3145":[3,12,4,2],"3153":[8,14,10,10],"3161":[8,14,10,10],"3169":[11,18,13,10],"3177":[11,18,13,10],"3185":[5,24,16,2],"3193":[5,24,16,2],"3201":[17,25,19,2],"3209":[17,25,19,2],"3217":[20,25,22,2],"3225":[20,25,22,2],"3233":[23,23,25,2],"3241":[23,23,25,2],"3249":[27,13,27,15],"3257":[28,10,28,13],"3265":[30,10,30,13],"3273":[32,10,32,13],"3281":[34,10,34,13],"3289":[26,27,37,2],"3297":[26,27,37,2],"3305":[40,12,40,82],"3313":[38,25,41,2],"3321":[38,25,41,2],"3329":[47,12,49,6],"3337":[45,12,49,6],"3345":[43,5,49,6],"3353":[53,16,53,39],"3361":[52,5,54,6],"3369":[55,5,57,6],"3377":[59,9,61,10],"3385":[58,5,62,6],"3393":[58,5,62,6],"3401":[42,16,64,2],"3409":[42,16,64,2],"3417":[66,39,66,60],"3425":[66,25,66,60],"3433":[66,12,66,60],"3441":[65,23,67,2],"3449":[65,23,67,2],"3457":[69,5,71,6],"3465":[68,35,73,2],"3473":[68,35,73,2],"3481":[75,5,77,6],"3489":[74,37,79,2],"3497":[74,37,79,2],"3505":[81,5,83,6],"3513":[80,40,85,2],"3521":[80,40,85,2],"3529":[86,35,88,2],"3537":[86,35,88,2],"3545":[89,37,91,2],"3553":[89,37,91,2],"3561":[92,28,94,2],"3569":[92,28,94,2],"3577":[95,33,100,2],"3585":[95,33,100,2],"3593":[102,5,104,6],"3601":[101,37,107,2],"3609":[101,37,107,2],"3617":[1,1,108,23],"3625":[1,1,108,23],"nBranches":48,"originalCodeFileName":"/home/blake/ExpoSE/Targets/less-file/node_modules/less/lib/less/tree/node.js","instrumentedCodeFileName":"/home/blake/ExpoSE/Targets/less-file/node_modules/less/lib/less/tree/node_jalangi_.js","code":"var Node = function() {\n};\nNode.prototype.toCSS = function (context) {\n    var strs = [];\n    this.genCSS(context, {\n        add: function(chunk, fileInfo, index) {\n            strs.push(chunk);\n        },\n        isEmpty: function () {\n            return strs.length === 0;\n        }\n    });\n    return strs.join('');\n};\nNode.prototype.genCSS = function (context, output) {\n    output.add(this.value);\n};\nNode.prototype.accept = function (visitor) {\n    this.value = visitor.visit(this.value);\n};\nNode.prototype.eval = function () { return this; };\nNode.prototype._operate = function (context, op, a, b) {\n    switch (op) {\n        case '+': return a + b;\n        case '-': return a - b;\n        case '*': return a * b;\n        case '/': return a / b;\n    }\n};\nNode.prototype.fround = function(context, value) {\n    var precision = context && context.numPrecision;\n    //add \"epsilon\" to ensure numbers like 1.000000005 (represented as 1.000000004999....) are properly rounded...\n    return (precision == null) ? value : Number((value + 2e-16).toFixed(precision));\n};\nNode.compare = function (a, b) {\n    /* returns:\n     -1: a < b\n     0: a = b\n     1: a > b\n     and *any* other value for a != b (e.g. undefined, NaN, -2 etc.) */\n\n    if ((a.compare) &&\n        // for \"symmetric results\" force toCSS-based comparison\n        // of Quoted or Anonymous if either value is one of those\n        !(b.type === \"Quoted\" || b.type === \"Anonymous\")) {\n        return a.compare(b);\n    } else if (b.compare) {\n        return -b.compare(a);\n    } else if (a.type !== b.type) {\n        return undefined;\n    }\n\n    a = a.value;\n    b = b.value;\n    if (!Array.isArray(a)) {\n        return a === b ? 0 : undefined;\n    }\n    if (a.length !== b.length) {\n        return undefined;\n    }\n    for (var i = 0; i < a.length; i++) {\n        if (Node.compare(a[i], b[i]) !== 0) {\n            return undefined;\n        }\n    }\n    return 0;\n};\n\nNode.numericCompare = function (a, b) {\n    return a  <  b ? -1\n        : a === b ?  0\n        : a  >  b ?  1 : undefined;\n};\n// Returns true if this node represents root of ast imported by reference\nNode.prototype.blocksVisibility = function () {\n    if (this.visibilityBlocks == null) {\n        this.visibilityBlocks = 0;\n    }\n    return this.visibilityBlocks !== 0;\n};\nNode.prototype.addVisibilityBlock = function () {\n    if (this.visibilityBlocks == null) {\n        this.visibilityBlocks = 0;\n    }\n    this.visibilityBlocks = this.visibilityBlocks + 1;\n};\nNode.prototype.removeVisibilityBlock = function () {\n    if (this.visibilityBlocks == null) {\n        this.visibilityBlocks = 0;\n    }\n    this.visibilityBlocks = this.visibilityBlocks - 1;\n};\n//Turns on node visibility - if called node will be shown in output regardless\n//of whether it comes from import by reference or not\nNode.prototype.ensureVisibility = function () {\n    this.nodeVisible = true;\n};\n//Turns off node visibility - if called node will NOT be shown in output regardless\n//of whether it comes from import by reference or not\nNode.prototype.ensureInvisibility = function () {\n    this.nodeVisible = false;\n};\n// return values:\n// false - the node must not be visible\n// true - the node must be visible\n// undefined or null - the node has the same visibility as its parent\nNode.prototype.isVisible = function () {\n    return this.nodeVisible;\n};\nNode.prototype.visibilityInfo = function() {\n    return {\n        visibilityBlocks: this.visibilityBlocks,\n        nodeVisible: this.nodeVisible\n    };\n};\nNode.prototype.copyVisibilityInfo = function(info) {\n    if (!info) {\n        return;\n    }\n    this.visibilityBlocks = info.visibilityBlocks;\n    this.nodeVisible = info.nodeVisible;\n};\nmodule.exports = Node;\n"};
jalangiLabel206:
    while (true) {
        try {
            J$.Se(3105, '/home/blake/ExpoSE/Targets/less-file/node_modules/less/lib/less/tree/node_jalangi_.js', '/home/blake/ExpoSE/Targets/less-file/node_modules/less/lib/less/tree/node.js');
            J$.N(3113, '__secret__S$', __secret__S$, 0);
            J$.N(3121, '__secret__traits__', __secret__traits__, 0);
            J$.N(3129, 'Node', Node, 0);
            var __secret__S$ = J$.X1(41, J$.W(33, '__secret__S$', J$.F(25, J$.R(9, 'require', require, 2), 0)(J$.T(17, 'S$', 21, false)), __secret__S$, 3));
            var __secret__traits__ = J$.X1(73, J$.W(65, '__secret__traits__', J$.G(57, J$.R(49, '__secret__S$', __secret__S$, 1), 'Traits', 0), __secret__traits__, 3));
            var Node = J$.X1(113, J$.W(105, 'Node', J$.T(97, function () {
                jalangiLabel187:
                    while (true) {
                        try {
                            J$.Fe(81, arguments.callee, this, arguments);
                            arguments = J$.N(89, 'arguments', arguments, 4);
                        } catch (J$e) {
                            J$.Ex(3137, J$e);
                        } finally {
                            if (J$.Fr(3145))
                                continue jalangiLabel187;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 81), Node, 3));
            J$.X1(433, J$.P(425, J$.G(129, J$.R(121, 'Node', Node, 1), 'prototype', 0), 'toCSS', J$.T(417, function (context) {
                jalangiLabel190:
                    while (true) {
                        try {
                            J$.Fe(385, arguments.callee, this, arguments);
                            arguments = J$.N(393, 'arguments', arguments, 4);
                            context = J$.N(401, 'context', context, 4);
                            J$.N(409, 'strs', strs, 0);
                            var strs = J$.X1(153, J$.W(145, 'strs', J$.T(137, [], 10, false), strs, 1));
                            J$.X1(337, J$.M(329, J$.R(161, 'this', this, 0), 'genCSS', 0)(J$.R(169, 'context', context, 0), J$.T(321, {
                                add: J$.T(249, function (chunk, fileInfo, index) {
                                    jalangiLabel188:
                                        while (true) {
                                            try {
                                                J$.Fe(209, arguments.callee, this, arguments);
                                                arguments = J$.N(217, 'arguments', arguments, 4);
                                                chunk = J$.N(225, 'chunk', chunk, 4);
                                                fileInfo = J$.N(233, 'fileInfo', fileInfo, 4);
                                                index = J$.N(241, 'index', index, 4);
                                                J$.X1(201, J$.M(193, J$.R(177, 'strs', strs, 0), 'push', 0)(J$.R(185, 'chunk', chunk, 0)));
                                            } catch (J$e) {
                                                J$.Ex(3153, J$e);
                                            } finally {
                                                if (J$.Fr(3161))
                                                    continue jalangiLabel188;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false, 209),
                                isEmpty: J$.T(313, function () {
                                    jalangiLabel189:
                                        while (true) {
                                            try {
                                                J$.Fe(297, arguments.callee, this, arguments);
                                                arguments = J$.N(305, 'arguments', arguments, 4);
                                                return J$.X1(289, J$.Rt(281, J$.B(10, '===', J$.G(265, J$.R(257, 'strs', strs, 0), 'length', 0), J$.T(273, 0, 22, false), 0)));
                                            } catch (J$e) {
                                                J$.Ex(3169, J$e);
                                            } finally {
                                                if (J$.Fr(3177))
                                                    continue jalangiLabel189;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false, 297)
                            }, 11, false)));
                            return J$.X1(377, J$.Rt(369, J$.M(361, J$.R(345, 'strs', strs, 0), 'join', 0)(J$.T(353, '', 21, false))));
                        } catch (J$e) {
                            J$.Ex(3185, J$e);
                        } finally {
                            if (J$.Fr(3193))
                                continue jalangiLabel190;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 385), 0));
            J$.X1(545, J$.P(537, J$.G(449, J$.R(441, 'Node', Node, 1), 'prototype', 0), 'genCSS', J$.T(529, function (context, output) {
                jalangiLabel191:
                    while (true) {
                        try {
                            J$.Fe(497, arguments.callee, this, arguments);
                            arguments = J$.N(505, 'arguments', arguments, 4);
                            context = J$.N(513, 'context', context, 4);
                            output = J$.N(521, 'output', output, 4);
                            J$.X1(489, J$.M(481, J$.R(457, 'output', output, 0), 'add', 0)(J$.G(473, J$.R(465, 'this', this, 0), 'value', 0)));
                        } catch (J$e) {
                            J$.Ex(3201, J$e);
                        } finally {
                            if (J$.Fr(3209))
                                continue jalangiLabel191;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 497), 0));
            J$.X1(665, J$.P(657, J$.G(561, J$.R(553, 'Node', Node, 1), 'prototype', 0), 'accept', J$.T(649, function (visitor) {
                jalangiLabel192:
                    while (true) {
                        try {
                            J$.Fe(625, arguments.callee, this, arguments);
                            arguments = J$.N(633, 'arguments', arguments, 4);
                            visitor = J$.N(641, 'visitor', visitor, 4);
                            J$.X1(617, J$.P(609, J$.R(569, 'this', this, 0), 'value', J$.M(601, J$.R(577, 'visitor', visitor, 0), 'visit', 0)(J$.G(593, J$.R(585, 'this', this, 0), 'value', 0)), 0));
                        } catch (J$e) {
                            J$.Ex(3217, J$e);
                        } finally {
                            if (J$.Fr(3225))
                                continue jalangiLabel192;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 625), 0));
            J$.X1(745, J$.P(737, J$.G(681, J$.R(673, 'Node', Node, 1), 'prototype', 0), 'eval', J$.T(729, function () {
                jalangiLabel193:
                    while (true) {
                        try {
                            J$.Fe(713, arguments.callee, this, arguments);
                            arguments = J$.N(721, 'arguments', arguments, 4);
                            return J$.X1(705, J$.Rt(697, J$.R(689, 'this', this, 0)));
                        } catch (J$e) {
                            J$.Ex(3233, J$e);
                        } finally {
                            if (J$.Fr(3241))
                                continue jalangiLabel193;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 713), 0));
            J$.X1(1001, J$.P(993, J$.G(761, J$.R(753, 'Node', Node, 1), 'prototype', 0), '_operate', J$.T(985, function (context, op, a, b) {
                jalangiLabel194:
                    while (true) {
                        try {
                            J$.Fe(937, arguments.callee, this, arguments);
                            arguments = J$.N(945, 'arguments', arguments, 4);
                            context = J$.N(953, 'context', context, 4);
                            op = J$.N(961, 'op', op, 4);
                            a = J$.N(969, 'a', a, 4);
                            b = J$.N(977, 'b', b, 4);
                            switch (J$.X1(3249, J$.C1(8, J$.R(769, 'op', op, 0)))) {
                            case J$.X1(3257, J$.C2(16, J$.T(809, '+', 21, false))):
                                return J$.X1(801, J$.Rt(793, J$.B(18, '+', J$.R(777, 'a', a, 0), J$.R(785, 'b', b, 0), 0)));
                            case J$.X1(3265, J$.C2(24, J$.T(849, '-', 21, false))):
                                return J$.X1(841, J$.Rt(833, J$.B(26, '-', J$.R(817, 'a', a, 0), J$.R(825, 'b', b, 0), 0)));
                            case J$.X1(3273, J$.C2(32, J$.T(889, '*', 21, false))):
                                return J$.X1(881, J$.Rt(873, J$.B(34, '*', J$.R(857, 'a', a, 0), J$.R(865, 'b', b, 0), 0)));
                            case J$.X1(3281, J$.C2(40, J$.T(929, '/', 21, false))):
                                return J$.X1(921, J$.Rt(913, J$.B(42, '/', J$.R(897, 'a', a, 0), J$.R(905, 'b', b, 0), 0)));
                            }
                        } catch (J$e) {
                            J$.Ex(3289, J$e);
                        } finally {
                            if (J$.Fr(3297))
                                continue jalangiLabel194;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 937), 0));
            J$.X1(1209, J$.P(1201, J$.G(1017, J$.R(1009, 'Node', Node, 1), 'prototype', 0), 'fround', J$.T(1193, function (context, value) {
                jalangiLabel195:
                    while (true) {
                        try {
                            J$.Fe(1153, arguments.callee, this, arguments);
                            arguments = J$.N(1161, 'arguments', arguments, 4);
                            context = J$.N(1169, 'context', context, 4);
                            value = J$.N(1177, 'value', value, 4);
                            J$.N(1185, 'precision', precision, 0);
                            var precision = J$.X1(1057, J$.W(1049, 'precision', J$.C(48, J$.R(1025, 'context', context, 0)) ? J$.G(1041, J$.R(1033, 'context', context, 0), 'numPrecision', 0) : J$._(), precision, 1));
                            return J$.X1(1145, J$.Rt(1137, J$.X1(3305, J$.C(56, J$.B(50, '==', J$.R(1065, 'precision', precision, 0), J$.T(1073, null, 25, false), 0))) ? J$.R(1081, 'value', value, 0) : J$.F(1129, J$.R(1089, 'Number', Number, 2), 0)(J$.M(1121, J$.B(58, '+', J$.R(1097, 'value', value, 0), J$.T(1105, 2e-16, 22, false), 0), 'toFixed', 0)(J$.R(1113, 'precision', precision, 0)))));
                        } catch (J$e) {
                            J$.Ex(3313, J$e);
                        } finally {
                            if (J$.Fr(3321))
                                continue jalangiLabel195;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 1153), 0));
            J$.X1(1889, J$.P(1881, J$.R(1217, 'Node', Node, 1), 'compare', J$.T(1873, function (a, b) {
                jalangiLabel196:
                    while (true) {
                        try {
                            J$.Fe(1833, arguments.callee, this, arguments);
                            arguments = J$.N(1841, 'arguments', arguments, 4);
                            a = J$.N(1849, 'a', a, 4);
                            b = J$.N(1857, 'b', b, 4);
                            J$.N(1865, 'i', i, 0);
                            if (J$.X1(3345, J$.C(96, J$.C(72, J$.G(1233, J$.R(1225, 'a', a, 0), 'compare', 0)) ? J$.U(82, '!', J$.C(64, J$.B(66, '===', J$.G(1249, J$.R(1241, 'b', b, 0), 'type', 0), J$.T(1257, 'Quoted', 21, false), 0)) ? J$._() : J$.B(74, '===', J$.G(1273, J$.R(1265, 'b', b, 0), 'type', 0), J$.T(1281, 'Anonymous', 21, false), 0)) : J$._()))) {
                                return J$.X1(1321, J$.Rt(1313, J$.M(1305, J$.R(1289, 'a', a, 0), 'compare', 0)(J$.R(1297, 'b', b, 0))));
                            } else if (J$.X1(3337, J$.C(88, J$.G(1337, J$.R(1329, 'b', b, 0), 'compare', 0)))) {
                                return J$.X1(1377, J$.Rt(1369, J$.U(90, '-', J$.M(1361, J$.R(1345, 'b', b, 0), 'compare', 0)(J$.R(1353, 'a', a, 0)))));
                            } else if (J$.X1(3329, J$.C(80, J$.B(98, '!==', J$.G(1393, J$.R(1385, 'a', a, 0), 'type', 0), J$.G(1409, J$.R(1401, 'b', b, 0), 'type', 0), 0)))) {
                                return J$.X1(1433, J$.Rt(1425, J$.T(1417, undefined, 24, false)));
                            }
                            J$.X1(1465, a = J$.W(1457, 'a', J$.G(1449, J$.R(1441, 'a', a, 0), 'value', 0), a, 0));
                            J$.X1(1497, b = J$.W(1489, 'b', J$.G(1481, J$.R(1473, 'b', b, 0), 'value', 0), b, 0));
                            if (J$.X1(3361, J$.C(112, J$.U(106, '!', J$.M(1521, J$.R(1505, 'Array', Array, 2), 'isArray', 0)(J$.R(1513, 'a', a, 0)))))) {
                                return J$.X1(1569, J$.Rt(1561, J$.X1(3353, J$.C(104, J$.B(114, '===', J$.R(1529, 'a', a, 0), J$.R(1537, 'b', b, 0), 0))) ? J$.T(1545, 0, 22, false) : J$.T(1553, undefined, 24, false)));
                            }
                            if (J$.X1(3369, J$.C(120, J$.B(122, '!==', J$.G(1585, J$.R(1577, 'a', a, 0), 'length', 0), J$.G(1601, J$.R(1593, 'b', b, 0), 'length', 0), 0)))) {
                                return J$.X1(1625, J$.Rt(1617, J$.T(1609, undefined, 24, false)));
                            }
                            for (var i = J$.X1(1649, J$.W(1641, 'i', J$.T(1633, 0, 22, false), i, 1)); J$.X1(3385, J$.C(136, J$.B(130, '<', J$.R(1657, 'i', i, 0), J$.G(1673, J$.R(1665, 'a', a, 0), 'length', 0), 0))); J$.X1(3393, J$.B(154, '-', i = J$.W(1697, 'i', J$.B(146, '+', J$.U(138, '+', J$.R(1689, 'i', i, 0)), J$.T(1681, 1, 22, false), 0), i, 0), J$.T(1705, 1, 22, false), 0))) {
                                if (J$.X1(3377, J$.C(128, J$.B(162, '!==', J$.M(1769, J$.R(1713, 'Node', Node, 1), 'compare', 0)(J$.G(1737, J$.R(1721, 'a', a, 0), J$.R(1729, 'i', i, 0), 4), J$.G(1761, J$.R(1745, 'b', b, 0), J$.R(1753, 'i', i, 0), 4)), J$.T(1777, 0, 22, false), 0)))) {
                                    return J$.X1(1801, J$.Rt(1793, J$.T(1785, undefined, 24, false)));
                                }
                            }
                            return J$.X1(1825, J$.Rt(1817, J$.T(1809, 0, 22, false)));
                        } catch (J$e) {
                            J$.Ex(3401, J$e);
                        } finally {
                            if (J$.Fr(3409))
                                continue jalangiLabel196;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 1833), 0));
            J$.X1(2049, J$.P(2041, J$.R(1897, 'Node', Node, 1), 'numericCompare', J$.T(2033, function (a, b) {
                jalangiLabel197:
                    while (true) {
                        try {
                            J$.Fe(2001, arguments.callee, this, arguments);
                            arguments = J$.N(2009, 'arguments', arguments, 4);
                            a = J$.N(2017, 'a', a, 4);
                            b = J$.N(2025, 'b', b, 4);
                            return J$.X1(1993, J$.Rt(1985, J$.X1(3433, J$.C(160, J$.B(170, '<', J$.R(1905, 'a', a, 0), J$.R(1913, 'b', b, 0), 0))) ? J$.U(178, '-', J$.T(1921, 1, 22, false)) : J$.X1(3425, J$.C(152, J$.B(186, '===', J$.R(1929, 'a', a, 0), J$.R(1937, 'b', b, 0), 0))) ? J$.T(1945, 0, 22, false) : J$.X1(3417, J$.C(144, J$.B(194, '>', J$.R(1953, 'a', a, 0), J$.R(1961, 'b', b, 0), 0))) ? J$.T(1969, 1, 22, false) : J$.T(1977, undefined, 24, false)));
                        } catch (J$e) {
                            J$.Ex(3441, J$e);
                        } finally {
                            if (J$.Fr(3449))
                                continue jalangiLabel197;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 2001), 0));
            J$.X1(2201, J$.P(2193, J$.G(2065, J$.R(2057, 'Node', Node, 1), 'prototype', 0), 'blocksVisibility', J$.T(2185, function () {
                jalangiLabel198:
                    while (true) {
                        try {
                            J$.Fe(2169, arguments.callee, this, arguments);
                            arguments = J$.N(2177, 'arguments', arguments, 4);
                            if (J$.X1(3457, J$.C(168, J$.B(202, '==', J$.G(2081, J$.R(2073, 'this', this, 0), 'visibilityBlocks', 0), J$.T(2089, null, 25, false), 0)))) {
                                J$.X1(2121, J$.P(2113, J$.R(2097, 'this', this, 0), 'visibilityBlocks', J$.T(2105, 0, 22, false), 0));
                            }
                            return J$.X1(2161, J$.Rt(2153, J$.B(210, '!==', J$.G(2137, J$.R(2129, 'this', this, 0), 'visibilityBlocks', 0), J$.T(2145, 0, 22, false), 0)));
                        } catch (J$e) {
                            J$.Ex(3465, J$e);
                        } finally {
                            if (J$.Fr(3473))
                                continue jalangiLabel198;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 2169), 0));
            J$.X1(2361, J$.P(2353, J$.G(2217, J$.R(2209, 'Node', Node, 1), 'prototype', 0), 'addVisibilityBlock', J$.T(2345, function () {
                jalangiLabel199:
                    while (true) {
                        try {
                            J$.Fe(2329, arguments.callee, this, arguments);
                            arguments = J$.N(2337, 'arguments', arguments, 4);
                            if (J$.X1(3481, J$.C(176, J$.B(218, '==', J$.G(2233, J$.R(2225, 'this', this, 0), 'visibilityBlocks', 0), J$.T(2241, null, 25, false), 0)))) {
                                J$.X1(2273, J$.P(2265, J$.R(2249, 'this', this, 0), 'visibilityBlocks', J$.T(2257, 0, 22, false), 0));
                            }
                            J$.X1(2321, J$.P(2313, J$.R(2281, 'this', this, 0), 'visibilityBlocks', J$.B(226, '+', J$.G(2297, J$.R(2289, 'this', this, 0), 'visibilityBlocks', 0), J$.T(2305, 1, 22, false), 0), 0));
                        } catch (J$e) {
                            J$.Ex(3489, J$e);
                        } finally {
                            if (J$.Fr(3497))
                                continue jalangiLabel199;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 2329), 0));
            J$.X1(2521, J$.P(2513, J$.G(2377, J$.R(2369, 'Node', Node, 1), 'prototype', 0), 'removeVisibilityBlock', J$.T(2505, function () {
                jalangiLabel200:
                    while (true) {
                        try {
                            J$.Fe(2489, arguments.callee, this, arguments);
                            arguments = J$.N(2497, 'arguments', arguments, 4);
                            if (J$.X1(3505, J$.C(184, J$.B(234, '==', J$.G(2393, J$.R(2385, 'this', this, 0), 'visibilityBlocks', 0), J$.T(2401, null, 25, false), 0)))) {
                                J$.X1(2433, J$.P(2425, J$.R(2409, 'this', this, 0), 'visibilityBlocks', J$.T(2417, 0, 22, false), 0));
                            }
                            J$.X1(2481, J$.P(2473, J$.R(2441, 'this', this, 0), 'visibilityBlocks', J$.B(242, '-', J$.G(2457, J$.R(2449, 'this', this, 0), 'visibilityBlocks', 0), J$.T(2465, 1, 22, false), 0), 0));
                        } catch (J$e) {
                            J$.Ex(3513, J$e);
                        } finally {
                            if (J$.Fr(3521))
                                continue jalangiLabel200;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 2489), 0));
            J$.X1(2609, J$.P(2601, J$.G(2537, J$.R(2529, 'Node', Node, 1), 'prototype', 0), 'ensureVisibility', J$.T(2593, function () {
                jalangiLabel201:
                    while (true) {
                        try {
                            J$.Fe(2577, arguments.callee, this, arguments);
                            arguments = J$.N(2585, 'arguments', arguments, 4);
                            J$.X1(2569, J$.P(2561, J$.R(2545, 'this', this, 0), 'nodeVisible', J$.T(2553, true, 23, false), 0));
                        } catch (J$e) {
                            J$.Ex(3529, J$e);
                        } finally {
                            if (J$.Fr(3537))
                                continue jalangiLabel201;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 2577), 0));
            J$.X1(2697, J$.P(2689, J$.G(2625, J$.R(2617, 'Node', Node, 1), 'prototype', 0), 'ensureInvisibility', J$.T(2681, function () {
                jalangiLabel202:
                    while (true) {
                        try {
                            J$.Fe(2665, arguments.callee, this, arguments);
                            arguments = J$.N(2673, 'arguments', arguments, 4);
                            J$.X1(2657, J$.P(2649, J$.R(2633, 'this', this, 0), 'nodeVisible', J$.T(2641, false, 23, false), 0));
                        } catch (J$e) {
                            J$.Ex(3545, J$e);
                        } finally {
                            if (J$.Fr(3553))
                                continue jalangiLabel202;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 2665), 0));
            J$.X1(2785, J$.P(2777, J$.G(2713, J$.R(2705, 'Node', Node, 1), 'prototype', 0), 'isVisible', J$.T(2769, function () {
                jalangiLabel203:
                    while (true) {
                        try {
                            J$.Fe(2753, arguments.callee, this, arguments);
                            arguments = J$.N(2761, 'arguments', arguments, 4);
                            return J$.X1(2745, J$.Rt(2737, J$.G(2729, J$.R(2721, 'this', this, 0), 'nodeVisible', 0)));
                        } catch (J$e) {
                            J$.Ex(3561, J$e);
                        } finally {
                            if (J$.Fr(3569))
                                continue jalangiLabel203;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 2753), 0));
            J$.X1(2897, J$.P(2889, J$.G(2801, J$.R(2793, 'Node', Node, 1), 'prototype', 0), 'visibilityInfo', J$.T(2881, function () {
                jalangiLabel204:
                    while (true) {
                        try {
                            J$.Fe(2865, arguments.callee, this, arguments);
                            arguments = J$.N(2873, 'arguments', arguments, 4);
                            return J$.X1(2857, J$.Rt(2849, J$.T(2841, {
                                visibilityBlocks: J$.G(2817, J$.R(2809, 'this', this, 0), 'visibilityBlocks', 0),
                                nodeVisible: J$.G(2833, J$.R(2825, 'this', this, 0), 'nodeVisible', 0)
                            }, 11, false)));
                        } catch (J$e) {
                            J$.Ex(3577, J$e);
                        } finally {
                            if (J$.Fr(3585))
                                continue jalangiLabel204;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 2865), 0));
            J$.X1(3065, J$.P(3057, J$.G(2913, J$.R(2905, 'Node', Node, 1), 'prototype', 0), 'copyVisibilityInfo', J$.T(3049, function (info) {
                jalangiLabel205:
                    while (true) {
                        try {
                            J$.Fe(3025, arguments.callee, this, arguments);
                            arguments = J$.N(3033, 'arguments', arguments, 4);
                            info = J$.N(3041, 'info', info, 4);
                            if (J$.X1(3593, J$.C(192, J$.U(250, '!', J$.R(2921, 'info', info, 0))))) {
                                return J$.X1(2937, J$.Rt(2929, undefined));
                            }
                            J$.X1(2977, J$.P(2969, J$.R(2945, 'this', this, 0), 'visibilityBlocks', J$.G(2961, J$.R(2953, 'info', info, 0), 'visibilityBlocks', 0), 0));
                            J$.X1(3017, J$.P(3009, J$.R(2985, 'this', this, 0), 'nodeVisible', J$.G(3001, J$.R(2993, 'info', info, 0), 'nodeVisible', 0), 0));
                        } catch (J$e) {
                            J$.Ex(3601, J$e);
                        } finally {
                            if (J$.Fr(3609))
                                continue jalangiLabel205;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 3025), 0));
            J$.X1(3097, J$.P(3089, J$.R(3073, 'module', module, 2), 'exports', J$.R(3081, 'Node', Node, 1), 0));
        } catch (J$e) {
            J$.Ex(3617, J$e);
        } finally {
            if (J$.Sr(3625)) {
                J$.L();
                continue jalangiLabel206;
            } else {
                J$.L();
                break jalangiLabel206;
            }
        }
    }
// JALANGI DO NOT INSTRUMENT
