// less/lib/less/logger_jalangi_.js
J$.iids = {"8":[21,17,21,48],"9":[1,20,1,27],"10":[20,25,20,51],"16":[20,25,20,51],"17":[1,28,1,32],"18":[20,53,20,56],"24":[30,17,30,28],"25":[1,20,1,33],"32":[28,25,28,51],"33":[1,20,1,33],"34":[20,53,20,56],"41":[1,20,1,33],"42":[21,17,21,48],"49":[2,26,2,38],"50":[28,25,28,51],"57":[2,26,2,45],"58":[28,53,28,56],"65":[2,26,2,45],"73":[2,26,2,45],"74":[28,53,28,56],"81":[3,1,3,7],"89":[5,9,5,13],"97":[5,25,5,32],"105":[5,34,5,37],"113":[5,9,5,38],"115":[5,9,5,24],"121":[5,9,5,39],"129":[4,12,6,6],"137":[4,12,6,6],"145":[4,12,6,6],"153":[4,12,6,6],"161":[8,9,8,13],"169":[8,25,8,31],"177":[8,33,8,36],"185":[8,9,8,37],"187":[8,9,8,24],"193":[8,9,8,38],"201":[7,11,9,6],"209":[7,11,9,6],"217":[7,11,9,6],"225":[7,11,9,6],"233":[11,9,11,13],"241":[11,25,11,31],"249":[11,33,11,36],"257":[11,9,11,37],"259":[11,9,11,24],"265":[11,9,11,38],"273":[10,11,12,6],"281":[10,11,12,6],"289":[10,11,12,6],"297":[10,11,12,6],"305":[14,9,14,13],"313":[14,25,14,32],"321":[14,34,14,37],"329":[14,9,14,38],"331":[14,9,14,24],"337":[14,9,14,39],"345":[13,12,15,6],"353":[13,12,15,6],"361":[13,12,15,6],"369":[13,12,15,6],"377":[17,9,17,13],"385":[17,9,17,24],"393":[17,30,17,38],"401":[17,9,17,39],"403":[17,9,17,29],"409":[17,9,17,40],"417":[16,18,18,6],"425":[16,18,18,6],"433":[16,18,18,6],"441":[16,18,18,6],"449":[20,22,20,23],"457":[20,22,20,23],"465":[20,22,20,23],"473":[20,25,20,26],"481":[20,29,20,33],"489":[20,29,20,44],"497":[20,29,20,51],"513":[20,53,20,54],"521":[20,53,20,56],"537":[21,17,21,21],"545":[21,17,21,32],"553":[21,33,21,34],"561":[21,17,21,35],"569":[21,40,21,48],"577":[22,17,22,21],"585":[22,17,22,32],"593":[22,40,22,41],"601":[22,43,22,44],"609":[22,17,22,45],"611":[22,17,22,39],"617":[22,17,22,46],"625":[23,17,23,24],"633":[23,17,23,24],"641":[19,21,26,6],"649":[19,21,26,6],"657":[19,21,26,6],"665":[19,21,26,6],"673":[19,21,26,6],"681":[28,22,28,23],"689":[28,22,28,23],"697":[28,22,28,23],"705":[28,25,28,26],"713":[28,29,28,33],"721":[28,29,28,44],"729":[28,29,28,51],"745":[28,53,28,54],"753":[28,53,28,56],"769":[29,31,29,35],"777":[29,31,29,46],"785":[29,47,29,48],"793":[29,31,29,49],"801":[29,50,29,54],"809":[29,31,29,55],"817":[29,31,29,55],"825":[29,31,29,55],"833":[30,17,30,28],"841":[31,17,31,28],"849":[31,29,31,32],"857":[31,17,31,33],"865":[31,17,31,34],"873":[27,17,34,6],"881":[27,17,34,6],"889":[27,17,34,6],"897":[27,17,34,6],"905":[27,17,34,6],"913":[27,17,34,6],"921":[27,17,34,6],"929":[35,17,35,19],"937":[3,18,36,2],"945":[3,1,36,2],"953":[3,1,36,3],"961":[1,1,36,3],"969":[1,1,36,3],"977":[1,1,36,3],"985":[4,12,6,6],"993":[4,12,6,6],"1001":[7,11,9,6],"1009":[7,11,9,6],"1017":[10,11,12,6],"1025":[10,11,12,6],"1033":[13,12,15,6],"1041":[13,12,15,6],"1049":[16,18,18,6],"1057":[16,18,18,6],"1065":[21,13,24,14],"1073":[20,9,25,10],"1081":[20,9,25,10],"1089":[19,21,26,6],"1097":[19,21,26,6],"1105":[30,13,32,14],"1113":[28,9,33,10],"1121":[28,9,33,10],"1129":[27,17,34,6],"1137":[27,17,34,6],"1145":[1,1,36,3],"1153":[1,1,36,3],"nBranches":8,"originalCodeFileName":"/home/blake/ExpoSE/Targets/less-file/node_modules/less/lib/less/logger.js","instrumentedCodeFileName":"/home/blake/ExpoSE/Targets/less-file/node_modules/less/lib/less/logger_jalangi_.js","code":"module.exports = {\n    error: function(msg) {\n        this._fireEvent(\"error\", msg);\n    },\n    warn: function(msg) {\n        this._fireEvent(\"warn\", msg);\n    },\n    info: function(msg) {\n        this._fireEvent(\"info\", msg);\n    },\n    debug: function(msg) {\n        this._fireEvent(\"debug\", msg);\n    },\n    addListener: function(listener) {\n        this._listeners.push(listener);\n    },\n    removeListener: function(listener) {\n        for (var i = 0; i < this._listeners.length; i++) {\n            if (this._listeners[i] === listener) {\n                this._listeners.splice(i, 1);\n                return;\n            }\n        }\n    },\n    _fireEvent: function(type, msg) {\n        for (var i = 0; i < this._listeners.length; i++) {\n            var logFunction = this._listeners[i][type];\n            if (logFunction) {\n                logFunction(msg);\n            }\n        }\n    },\n    _listeners: []\n};\n"};
jalangiLabel180:
    while (true) {
        try {
            J$.Se(961, '/home/blake/ExpoSE/Targets/less-file/node_modules/less/lib/less/logger_jalangi_.js', '/home/blake/ExpoSE/Targets/less-file/node_modules/less/lib/less/logger.js');
            J$.N(969, '__secret__S$', __secret__S$, 0);
            J$.N(977, '__secret__traits__', __secret__traits__, 0);
            var __secret__S$ = J$.X1(41, J$.W(33, '__secret__S$', J$.F(25, J$.R(9, 'require', require, 2), 0)(J$.T(17, 'S$', 21, false)), __secret__S$, 3));
            var __secret__traits__ = J$.X1(73, J$.W(65, '__secret__traits__', J$.G(57, J$.R(49, '__secret__S$', __secret__S$, 1), 'Traits', 0), __secret__traits__, 3));
            J$.X1(953, J$.P(945, J$.R(81, 'module', module, 2), 'exports', J$.T(937, {
                error: J$.T(153, function (msg) {
                    jalangiLabel173:
                        while (true) {
                            try {
                                J$.Fe(129, arguments.callee, this, arguments);
                                arguments = J$.N(137, 'arguments', arguments, 4);
                                msg = J$.N(145, 'msg', msg, 4);
                                J$.X1(121, J$.M(113, J$.R(89, 'this', this, 0), '_fireEvent', 0)(J$.T(97, 'error', 21, false), J$.R(105, 'msg', msg, 0)));
                            } catch (J$e) {
                                J$.Ex(985, J$e);
                            } finally {
                                if (J$.Fr(993))
                                    continue jalangiLabel173;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false, 129),
                warn: J$.T(225, function (msg) {
                    jalangiLabel174:
                        while (true) {
                            try {
                                J$.Fe(201, arguments.callee, this, arguments);
                                arguments = J$.N(209, 'arguments', arguments, 4);
                                msg = J$.N(217, 'msg', msg, 4);
                                J$.X1(193, J$.M(185, J$.R(161, 'this', this, 0), '_fireEvent', 0)(J$.T(169, 'warn', 21, false), J$.R(177, 'msg', msg, 0)));
                            } catch (J$e) {
                                J$.Ex(1001, J$e);
                            } finally {
                                if (J$.Fr(1009))
                                    continue jalangiLabel174;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false, 201),
                info: J$.T(297, function (msg) {
                    jalangiLabel175:
                        while (true) {
                            try {
                                J$.Fe(273, arguments.callee, this, arguments);
                                arguments = J$.N(281, 'arguments', arguments, 4);
                                msg = J$.N(289, 'msg', msg, 4);
                                J$.X1(265, J$.M(257, J$.R(233, 'this', this, 0), '_fireEvent', 0)(J$.T(241, 'info', 21, false), J$.R(249, 'msg', msg, 0)));
                            } catch (J$e) {
                                J$.Ex(1017, J$e);
                            } finally {
                                if (J$.Fr(1025))
                                    continue jalangiLabel175;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false, 273),
                debug: J$.T(369, function (msg) {
                    jalangiLabel176:
                        while (true) {
                            try {
                                J$.Fe(345, arguments.callee, this, arguments);
                                arguments = J$.N(353, 'arguments', arguments, 4);
                                msg = J$.N(361, 'msg', msg, 4);
                                J$.X1(337, J$.M(329, J$.R(305, 'this', this, 0), '_fireEvent', 0)(J$.T(313, 'debug', 21, false), J$.R(321, 'msg', msg, 0)));
                            } catch (J$e) {
                                J$.Ex(1033, J$e);
                            } finally {
                                if (J$.Fr(1041))
                                    continue jalangiLabel176;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false, 345),
                addListener: J$.T(441, function (listener) {
                    jalangiLabel177:
                        while (true) {
                            try {
                                J$.Fe(417, arguments.callee, this, arguments);
                                arguments = J$.N(425, 'arguments', arguments, 4);
                                listener = J$.N(433, 'listener', listener, 4);
                                J$.X1(409, J$.M(401, J$.G(385, J$.R(377, 'this', this, 0), '_listeners', 0), 'push', 0)(J$.R(393, 'listener', listener, 0)));
                            } catch (J$e) {
                                J$.Ex(1049, J$e);
                            } finally {
                                if (J$.Fr(1057))
                                    continue jalangiLabel177;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false, 417),
                removeListener: J$.T(673, function (listener) {
                    jalangiLabel178:
                        while (true) {
                            try {
                                J$.Fe(641, arguments.callee, this, arguments);
                                arguments = J$.N(649, 'arguments', arguments, 4);
                                listener = J$.N(657, 'listener', listener, 4);
                                J$.N(665, 'i', i, 0);
                                for (var i = J$.X1(465, J$.W(457, 'i', J$.T(449, 0, 22, false), i, 1)); J$.X1(1073, J$.C(16, J$.B(10, '<', J$.R(473, 'i', i, 0), J$.G(497, J$.G(489, J$.R(481, 'this', this, 0), '_listeners', 0), 'length', 0), 0))); J$.X1(1081, J$.B(34, '-', i = J$.W(521, 'i', J$.B(26, '+', J$.U(18, '+', J$.R(513, 'i', i, 0)), J$.T(505, 1, 22, false), 0), i, 0), J$.T(529, 1, 22, false), 0))) {
                                    if (J$.X1(1065, J$.C(8, J$.B(42, '===', J$.G(561, J$.G(545, J$.R(537, 'this', this, 0), '_listeners', 0), J$.R(553, 'i', i, 0), 4), J$.R(569, 'listener', listener, 0), 0)))) {
                                        J$.X1(617, J$.M(609, J$.G(585, J$.R(577, 'this', this, 0), '_listeners', 0), 'splice', 0)(J$.R(593, 'i', i, 0), J$.T(601, 1, 22, false)));
                                        return J$.X1(633, J$.Rt(625, undefined));
                                    }
                                }
                            } catch (J$e) {
                                J$.Ex(1089, J$e);
                            } finally {
                                if (J$.Fr(1097))
                                    continue jalangiLabel178;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false, 641),
                _fireEvent: J$.T(921, function (type, msg) {
                    jalangiLabel179:
                        while (true) {
                            try {
                                J$.Fe(873, arguments.callee, this, arguments);
                                arguments = J$.N(881, 'arguments', arguments, 4);
                                type = J$.N(889, 'type', type, 4);
                                msg = J$.N(897, 'msg', msg, 4);
                                J$.N(905, 'i', i, 0);
                                J$.N(913, 'logFunction', logFunction, 0);
                                for (var i = J$.X1(697, J$.W(689, 'i', J$.T(681, 0, 22, false), i, 1)); J$.X1(1113, J$.C(32, J$.B(50, '<', J$.R(705, 'i', i, 0), J$.G(729, J$.G(721, J$.R(713, 'this', this, 0), '_listeners', 0), 'length', 0), 0))); J$.X1(1121, J$.B(74, '-', i = J$.W(753, 'i', J$.B(66, '+', J$.U(58, '+', J$.R(745, 'i', i, 0)), J$.T(737, 1, 22, false), 0), i, 0), J$.T(761, 1, 22, false), 0))) {
                                    var logFunction = J$.X1(825, J$.W(817, 'logFunction', J$.G(809, J$.G(793, J$.G(777, J$.R(769, 'this', this, 0), '_listeners', 0), J$.R(785, 'i', i, 0), 4), J$.R(801, 'type', type, 0), 4), logFunction, 1));
                                    if (J$.X1(1105, J$.C(24, J$.R(833, 'logFunction', logFunction, 0)))) {
                                        J$.X1(865, J$.F(857, J$.R(841, 'logFunction', logFunction, 0), 0)(J$.R(849, 'msg', msg, 0)));
                                    }
                                }
                            } catch (J$e) {
                                J$.Ex(1129, J$e);
                            } finally {
                                if (J$.Fr(1137))
                                    continue jalangiLabel179;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false, 873),
                _listeners: J$.T(929, [], 10, false)
            }, 11, false), 0));
        } catch (J$e) {
            J$.Ex(1145, J$e);
        } finally {
            if (J$.Sr(1153)) {
                J$.L();
                continue jalangiLabel180;
            } else {
                J$.L();
                break jalangiLabel180;
            }
        }
    }
// JALANGI DO NOT INSTRUMENT
