// fs-tree-diff/lib/entry_jalangi_.js
J$.iids = {"8":[9,17,9,28],"9":[1,20,1,27],"10":[7,43,7,66],"17":[1,28,1,32],"18":[7,23,7,75],"25":[1,20,1,33],"26":[14,13,14,30],"33":[1,20,1,33],"34":[14,12,14,41],"41":[1,20,1,33],"49":[2,26,2,38],"57":[2,26,2,45],"65":[2,26,2,45],"73":[2,26,2,45],"81":[3,1,3,13],"89":[3,1,3,14],"97":[4,22,4,27],"105":[4,22,4,27],"113":[4,22,4,27],"121":[5,1,5,7],"129":[5,18,5,23],"137":[5,1,5,23],"145":[5,1,5,24],"153":[7,23,7,35],"161":[7,43,7,55],"169":[7,43,7,62],"177":[7,65,7,66],"185":[7,23,7,67],"187":[7,23,7,42],"193":[7,72,7,75],"201":[7,23,7,75],"209":[7,23,7,75],"217":[8,5,8,9],"225":[8,25,8,37],"233":[8,5,8,37],"241":[8,5,8,38],"249":[9,5,9,9],"257":[9,17,9,28],"265":[9,31,9,45],"273":[9,48,9,49],"281":[9,5,9,49],"289":[9,5,9,50],"297":[10,5,10,9],"305":[10,17,10,21],"313":[10,5,10,21],"321":[10,5,10,22],"329":[11,5,11,9],"337":[11,18,11,23],"345":[11,5,11,23],"353":[11,5,11,24],"361":[6,1,12,2],"369":[6,1,12,2],"377":[6,1,12,2],"385":[6,1,12,2],"393":[6,1,12,2],"401":[6,1,12,2],"409":[13,1,13,6],"417":[13,1,13,16],"425":[14,13,14,17],"433":[14,13,14,22],"441":[14,25,14,30],"449":[14,36,14,41],"457":[14,12,14,41],"465":[14,5,14,42],"473":[13,31,15,2],"481":[13,31,15,2],"489":[13,31,15,2],"497":[13,1,15,2],"505":[13,1,15,3],"513":[1,1,15,3],"521":[1,1,15,3],"529":[1,1,15,3],"537":[1,1,15,3],"545":[6,1,12,2],"553":[1,1,15,3],"561":[9,17,9,49],"569":[6,1,12,2],"577":[6,1,12,2],"585":[13,31,15,2],"593":[13,31,15,2],"601":[1,1,15,3],"609":[1,1,15,3],"nBranches":2,"originalCodeFileName":"/home/blake/ExpoSE/Targets/ember-component-css-withbp/node_modules/fs-tree-diff/lib/entry.js","instrumentedCodeFileName":"/home/blake/ExpoSE/Targets/ember-component-css-withbp/node_modules/fs-tree-diff/lib/entry_jalangi_.js","code":"'use strict';\n\nvar DIRECTORY_MODE = 16877;\n\nmodule.exports = Entry;\nfunction Entry(relativePath, size, mtime) {\n  var isDirectory = relativePath.charAt(relativePath.length - 1) === '/';\n  this.relativePath = relativePath;\n  this.mode = isDirectory ? DIRECTORY_MODE : 0;\n  this.size = size;\n  this.mtime = mtime;\n}\n\nEntry.prototype.isDirectory = function() {\n  return (this.mode & 61440) === 16384;\n};\n\n"};
jalangiLabel323:
    while (true) {
        try {
            J$.Se(513, '/home/blake/ExpoSE/Targets/ember-component-css-withbp/node_modules/fs-tree-diff/lib/entry_jalangi_.js', '/home/blake/ExpoSE/Targets/ember-component-css-withbp/node_modules/fs-tree-diff/lib/entry.js');
            function Entry(relativePath, size, mtime) {
                jalangiLabel321:
                    while (true) {
                        try {
                            J$.Fe(361, arguments.callee, this, arguments);
                            arguments = J$.N(369, 'arguments', arguments, 4);
                            relativePath = J$.N(377, 'relativePath', relativePath, 4);
                            size = J$.N(385, 'size', size, 4);
                            mtime = J$.N(393, 'mtime', mtime, 4);
                            J$.N(401, 'isDirectory', isDirectory, 0);
                            var isDirectory = J$.X1(209, J$.W(201, 'isDirectory', J$.B(18, '===', J$.M(185, J$.R(153, 'relativePath', relativePath, 0), 'charAt', 0)(J$.B(10, '-', J$.G(169, J$.R(161, 'relativePath', relativePath, 0), 'length', 0), J$.T(177, 1, 22, false), 0)), J$.T(193, '/', 21, false), 0), isDirectory, 1));
                            J$.X1(241, J$.P(233, J$.R(217, 'this', this, 0), 'relativePath', J$.R(225, 'relativePath', relativePath, 0), 0));
                            J$.X1(289, J$.P(281, J$.R(249, 'this', this, 0), 'mode', J$.X1(561, J$.C(8, J$.R(257, 'isDirectory', isDirectory, 0))) ? J$.R(265, 'DIRECTORY_MODE', DIRECTORY_MODE, 1) : J$.T(273, 0, 22, false), 0));
                            J$.X1(321, J$.P(313, J$.R(297, 'this', this, 0), 'size', J$.R(305, 'size', size, 0), 0));
                            J$.X1(353, J$.P(345, J$.R(329, 'this', this, 0), 'mtime', J$.R(337, 'mtime', mtime, 0), 0));
                        } catch (J$e) {
                            J$.Ex(569, J$e);
                        } finally {
                            if (J$.Fr(577))
                                continue jalangiLabel321;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(521, '__secret__S$', __secret__S$, 0);
            J$.N(529, '__secret__traits__', __secret__traits__, 0);
            J$.N(537, 'DIRECTORY_MODE', DIRECTORY_MODE, 0);
            Entry = J$.N(553, 'Entry', J$.T(545, Entry, 12, false, 361), 0);
            var __secret__S$ = J$.X1(41, J$.W(33, '__secret__S$', J$.F(25, J$.R(9, 'require', require, 2), 0)(J$.T(17, 'S$', 21, false)), __secret__S$, 3));
            var __secret__traits__ = J$.X1(73, J$.W(65, '__secret__traits__', J$.G(57, J$.R(49, '__secret__S$', __secret__S$, 1), 'Traits', 0), __secret__traits__, 3));
            J$.X1(89, J$.T(81, 'use strict', 21, false));
            var DIRECTORY_MODE = J$.X1(113, J$.W(105, 'DIRECTORY_MODE', J$.T(97, 16877, 22, false), DIRECTORY_MODE, 3));
            J$.X1(145, J$.P(137, J$.R(121, 'module', module, 2), 'exports', J$.R(129, 'Entry', Entry, 1), 0));
            J$.X1(505, J$.P(497, J$.G(417, J$.R(409, 'Entry', Entry, 1), 'prototype', 0), 'isDirectory', J$.T(489, function () {
                jalangiLabel322:
                    while (true) {
                        try {
                            J$.Fe(473, arguments.callee, this, arguments);
                            arguments = J$.N(481, 'arguments', arguments, 4);
                            return J$.X1(465, J$.Rt(457, J$.B(34, '===', J$.B(26, '&', J$.G(433, J$.R(425, 'this', this, 0), 'mode', 0), J$.T(441, 61440, 22, false), 0), J$.T(449, 16384, 22, false), 0)));
                        } catch (J$e) {
                            J$.Ex(585, J$e);
                        } finally {
                            if (J$.Fr(593))
                                continue jalangiLabel322;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 473), 0));
        } catch (J$e) {
            J$.Ex(601, J$e);
        } finally {
            if (J$.Sr(609)) {
                J$.L();
                continue jalangiLabel323;
            } else {
                J$.L();
                break jalangiLabel323;
            }
        }
    }
// JALANGI DO NOT INSTRUMENT
