// generator-hughes/generators/app/mixins/wordpress/prompting.js
'use strict';

var yeoman = require('yeoman-generator'),
    chalk = require('chalk'),
    yosay = require('yosay'),
		printTitle  = require('./../../helpers/printTitle'),
    fs = require('graceful-fs');

module.exports = {
	wordpress: function() {
		
		this.log(printTitle('Configure Wordpress'))
		
		var done = this.async();
		this.prompt([{
      name: 'wpDir',
      message: 'Directory for Wordpress core files',
      default: 'wordpress'
    }, {
      name: 'wpContentDir',
      message: 'Directory for Wordpress content',
      default: 'wp-content'
    },  {
      // Theme Name
      name: 'themeName',
      message: 'What is the name of your theme?',
      default: 'Wordpress Theme'
    }, {
      // Wordpress Site URL
      name: 'wpUrl',
      message: 'The address of the new site',
      default: 'example.com'
    },  {
      // Wordpress Admin name
      name: 'wpAdmin',
      message: 'Wordpress admin user',
      default: 'admin'
    }, {
      // Wordpress Admin password
      name: 'wpAdminPass',
      message: 'Wordpress admin password',
      default: 'password'
    }, {
      // Wordpress Admin email
      name: 'wpAdminEmail',
      message: 'Wordpress admin email',
      default: 'kmh3785@gmail.com'
    }], 

		function(props){
			this.props = props;
			// To access props later use this.props.someAnswer;
			// Wordpress options
      this.wpDir = this.props.wpDir;
      this.wpContentDir = this.props.wpContentDir;
      this.themeName = this.props.themeName;
      this.themeDirName = this.themeName.replace(/\s/g,'-').toLowerCase();
      // Wordpress admin info
      this.wpUrl = this.props.wpUrl;
      this.wpTitle = this.props.projectName;
      this.wpAdmin = this.props.wpAdmin;
      this.wpAdminPass = this.props.wpAdminPass;
      this.wpAdminEmail = this.props.wpAdminEmail;
      
			done();
		}.bind(this))
	}};
