// generator-hughes/generators/app/mixins/wordpress/writing.js
'use strict';

var yeoman = require('yeoman-generator'),
    chalk = require('chalk'),
    yosay = require('yosay'),
    fs = require('fs.extra'),
	mysql = require('mysql'),
	exec = require('child_process').execSync;

module.exports = {
	wordpress: function() {
		
		var wordpressContext = {
				// Wordpress dirs
				wpDir: this.wpDir,
				wpContentDir: this.wpContentDir,

				// Build Directories
				srcDir: this.srcDir,
				buildDir: this.buildDir,
				themeDirName: this.themeDirName,

				// Database info
				dbName: this.dbName,
				dbUser: this.dbUser,
				dbPass: this.dbPass,
			}
		
		var done = this.async();
		var self = this;
		
	
		// Set Wordpres vars
		 var wpDir = this.wpDir,
			 buildDir = this.buildDir,

			 wpContentDir = this.wpContentDir,
			 wpBuildDir = buildDir+'/'+wpDir,

			 // Install vars
			 url = this.wpUrl,
			 title = this.wpTitle,
			 admin = this.wpAdmin,
			 adminPass = this.wpAdminPass,
			 adminEmail = this.wpAdminEmail,
			
			// WP CLI Install
			wpcliInstall = "curl -O https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli.phar && chmod +x wp-cli.phar && sudo mv wp-cli.phar /usr/local/bin/wp",
			// WP Install command
			wpInstall = 'wp core install --skip-email --url=' + url + ' --title=' + title + ' --admin_user=' + admin + ' --admin_password=' + adminPass + ' --admin_email=' + adminEmail + ' --path=' + buildDir;
        
        
		// Move WP files
		// into build dir
		this.fs.copyTpl (
			this.templatePath('wordpress-files'),
			this.destinationPath(buildDir), wordpressContext
		);
        
		// Create WP dir
		fs.mkdirs(wpBuildDir, function() {
			
			exec(wpcliInstall, {
				cwd: wpBuildDir 
			}),
			
			exec('wp core download', {
				cwd: wpBuildDir 
			}),
				
			exec('cp -R ' + wpDir + '/wp-content ' + wpContentDir, {
				cwd: buildDir
			})
		});
		
	 // Move starter kit
		this.fs.copy(
			this.templatePath('wordpress-starter-kit'),
			this.destinationPath(this.srcDir)
			
		), function() {
			// Once done,
			// Install Wordpress
			// exec(wpInstall);
		}
		done();
		
		
	
}}
