// inflect/lib/inflect/string_extensions_jalangi_.js
J$.iids = {"8":[13,13,13,46],"9":[1,20,1,27],"10":[13,13,13,46],"16":[34,13,34,24],"17":[1,28,1,32],"18":[34,13,34,24],"25":[1,20,1,33],"33":[1,20,1,33],"41":[1,20,1,33],"49":[2,26,2,38],"57":[2,26,2,45],"65":[2,26,2,45],"73":[2,26,2,45],"81":[4,11,4,18],"89":[4,19,4,31],"97":[4,11,4,32],"105":[4,11,4,32],"113":[4,1,4,33],"121":[6,5,6,11],"129":[6,5,6,21],"137":[7,16,7,23],"145":[7,34,7,38],"153":[7,16,7,39],"155":[7,16,7,33],"161":[7,16,7,39],"169":[7,9,7,40],"177":[6,34,8,6],"185":[6,34,8,6],"193":[6,34,8,6],"201":[6,5,8,6],"209":[6,5,8,7],"217":[9,5,9,11],"225":[9,5,9,21],"233":[10,16,10,23],"241":[10,36,10,40],"249":[10,16,10,41],"251":[10,16,10,35],"257":[10,16,10,41],"265":[10,9,10,42],"273":[9,36,11,6],"281":[9,36,11,6],"289":[9,36,11,6],"297":[9,5,11,6],"305":[9,5,11,7],"313":[12,5,12,11],"321":[12,5,12,21],"329":[13,13,13,38],"337":[13,42,13,46],"345":[14,41,14,45],"353":[14,41,14,45],"361":[14,13,14,46],"369":[16,16,16,23],"377":[16,33,16,37],"385":[16,39,16,64],"393":[16,16,16,65],"395":[16,16,16,32],"401":[16,16,16,65],"409":[16,9,16,66],"417":[12,33,17,6],"425":[12,33,17,6],"433":[12,33,17,6],"441":[12,33,17,6],"449":[12,5,17,6],"457":[12,5,17,7],"465":[18,5,18,11],"473":[18,5,18,21],"481":[19,16,19,23],"489":[19,35,19,39],"497":[19,16,19,40],"499":[19,16,19,34],"505":[19,16,19,40],"513":[19,9,19,41],"521":[18,35,20,6],"529":[18,35,20,6],"537":[18,35,20,6],"545":[18,5,20,6],"553":[18,5,20,7],"561":[21,5,21,11],"569":[21,5,21,21],"577":[22,16,22,23],"585":[22,37,22,41],"593":[22,16,22,42],"595":[22,16,22,36],"601":[22,16,22,42],"609":[22,9,22,43],"617":[21,37,23,6],"625":[21,37,23,6],"633":[21,37,23,6],"641":[21,5,23,6],"649":[21,5,23,7],"657":[24,5,24,11],"665":[24,5,24,21],"673":[25,16,25,23],"681":[25,33,25,37],"689":[25,16,25,38],"691":[25,16,25,32],"697":[25,16,25,38],"705":[25,9,25,39],"713":[24,33,26,6],"721":[24,33,26,6],"729":[24,33,26,6],"737":[24,5,26,6],"745":[24,5,26,7],"753":[27,5,27,11],"761":[27,5,27,21],"769":[28,16,28,23],"777":[28,35,28,39],"785":[28,16,28,40],"787":[28,16,28,34],"793":[28,16,28,40],"801":[28,9,28,41],"809":[27,35,29,6],"817":[27,35,29,6],"825":[27,35,29,6],"833":[27,5,29,6],"841":[27,5,29,7],"849":[30,5,30,11],"857":[30,5,30,21],"865":[31,16,31,23],"873":[31,34,31,38],"881":[31,16,31,39],"883":[31,16,31,33],"889":[31,16,31,39],"897":[31,9,31,40],"905":[30,34,32,6],"913":[30,34,32,6],"921":[30,34,32,6],"929":[30,5,32,6],"937":[30,5,32,7],"945":[33,5,33,11],"953":[33,5,33,21],"961":[34,13,34,16],"969":[34,20,34,24],"977":[35,19,35,22],"985":[35,19,35,22],"993":[35,13,35,23],"1001":[37,16,37,23],"1009":[37,37,37,41],"1017":[37,43,37,46],"1025":[37,16,37,47],"1027":[37,16,37,36],"1033":[37,16,37,47],"1041":[37,9,37,48],"1049":[33,37,38,6],"1057":[33,37,38,6],"1065":[33,37,38,6],"1073":[33,37,38,6],"1081":[33,5,38,6],"1089":[33,5,38,7],"1097":[39,12,39,18],"1105":[39,12,39,28],"1113":[40,16,40,23],"1121":[40,33,40,37],"1129":[40,16,40,38],"1131":[40,16,40,32],"1137":[40,16,40,38],"1145":[40,9,40,39],"1153":[39,40,41,6],"1161":[39,40,41,6],"1169":[39,40,41,6],"1177":[39,12,41,6],"1185":[39,12,41,6],"1193":[39,5,41,7],"1201":[5,26,42,2],"1209":[5,26,42,2],"1217":[5,26,42,2],"1225":[5,26,42,2],"1233":[5,1,42,3],"1241":[43,1,43,8],"1249":[43,34,43,56],"1257":[43,1,43,56],"1265":[43,1,43,57],"1273":[1,1,43,57],"1281":[1,1,43,57],"1289":[1,1,43,57],"1297":[1,1,43,57],"1305":[1,1,43,57],"1313":[6,34,8,6],"1321":[6,34,8,6],"1329":[9,36,11,6],"1337":[9,36,11,6],"1345":[13,9,15,10],"1353":[12,33,17,6],"1361":[12,33,17,6],"1369":[18,35,20,6],"1377":[18,35,20,6],"1385":[21,37,23,6],"1393":[21,37,23,6],"1401":[24,33,26,6],"1409":[24,33,26,6],"1417":[27,35,29,6],"1425":[27,35,29,6],"1433":[30,34,32,6],"1441":[30,34,32,6],"1449":[34,9,36,10],"1457":[33,37,38,6],"1465":[33,37,38,6],"1473":[39,40,41,6],"1481":[39,40,41,6],"1489":[5,26,42,2],"1497":[5,26,42,2],"1505":[1,1,43,57],"1513":[1,1,43,57],"nBranches":4,"originalCodeFileName":"/home/blake/ExpoSE/Targets/xero-extension/node_modules/inflect/lib/inflect/string_extensions.js","instrumentedCodeFileName":"/home/blake/ExpoSE/Targets/xero-extension/node_modules/inflect/lib/inflect/string_extensions_jalangi_.js","code":"// Generated by CoffeeScript 1.3.3\nvar enableStringExtensions, inflect;\n\ninflect = require('../inflect');\n\nenableStringExtensions = function() {\n  String.prototype.pluralize = function() {\n    return inflect.pluralize(this);\n  };\n  String.prototype.singularize = function() {\n    return inflect.singularize(this);\n  };\n  String.prototype.camelize = function(first_letter_in_uppercase) {\n    if (first_letter_in_uppercase == null) {\n      first_letter_in_uppercase = true;\n    }\n    return inflect.camelize(this, first_letter_in_uppercase);\n  };\n  String.prototype.capitalize = function() {\n    return inflect.capitalize(this);\n  };\n  String.prototype.decapitalize = function() {\n    return inflect.decapitalize(this);\n  };\n  String.prototype.titleize = function() {\n    return inflect.titleize(this);\n  };\n  String.prototype.underscore = function() {\n    return inflect.underscore(this);\n  };\n  String.prototype.dasherize = function() {\n    return inflect.dasherize(this);\n  };\n  String.prototype.parameterize = function(sep) {\n    if (sep == null) {\n      sep = '-';\n    }\n    return inflect.parameterize(this, sep);\n  };\n  return String.prototype.humanize = function() {\n    return inflect.humanize(this);\n  };\n};\n\nexports.enableStringExtensions = enableStringExtensions;\n"};
jalangiLabel396:
    while (true) {
        try {
            J$.Se(1273, '/home/blake/ExpoSE/Targets/xero-extension/node_modules/inflect/lib/inflect/string_extensions_jalangi_.js', '/home/blake/ExpoSE/Targets/xero-extension/node_modules/inflect/lib/inflect/string_extensions.js');
            J$.N(1281, '__secret__S$', __secret__S$, 0);
            J$.N(1289, '__secret__traits__', __secret__traits__, 0);
            J$.N(1297, 'enableStringExtensions', enableStringExtensions, 0);
            J$.N(1305, 'inflect', inflect, 0);
            var __secret__S$ = J$.X1(41, J$.W(33, '__secret__S$', J$.F(25, J$.R(9, 'require', require, 2), 0)(J$.T(17, 'S$', 21, false)), __secret__S$, 3));
            var __secret__traits__ = J$.X1(73, J$.W(65, '__secret__traits__', J$.G(57, J$.R(49, '__secret__S$', __secret__S$, 1), 'Traits', 0), __secret__traits__, 3));
            var enableStringExtensions, inflect;
            J$.X1(113, inflect = J$.W(105, 'inflect', J$.F(97, J$.R(81, 'require', require, 2), 0)(J$.T(89, '../inflect', 21, false)), inflect, 2));
            J$.X1(1233, enableStringExtensions = J$.W(1225, 'enableStringExtensions', J$.T(1217, function () {
                jalangiLabel395:
                    while (true) {
                        try {
                            J$.Fe(1201, arguments.callee, this, arguments);
                            arguments = J$.N(1209, 'arguments', arguments, 4);
                            J$.X1(209, J$.P(201, J$.G(129, J$.R(121, 'String', String, 2), 'prototype', 0), 'pluralize', J$.T(193, function () {
                                jalangiLabel385:
                                    while (true) {
                                        try {
                                            J$.Fe(177, arguments.callee, this, arguments);
                                            arguments = J$.N(185, 'arguments', arguments, 4);
                                            return J$.X1(169, J$.Rt(161, J$.M(153, J$.R(137, 'inflect', inflect, 1), 'pluralize', 0)(J$.R(145, 'this', this, 0))));
                                        } catch (J$e) {
                                            J$.Ex(1313, J$e);
                                        } finally {
                                            if (J$.Fr(1321))
                                                continue jalangiLabel385;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false, 177), 0));
                            J$.X1(305, J$.P(297, J$.G(225, J$.R(217, 'String', String, 2), 'prototype', 0), 'singularize', J$.T(289, function () {
                                jalangiLabel386:
                                    while (true) {
                                        try {
                                            J$.Fe(273, arguments.callee, this, arguments);
                                            arguments = J$.N(281, 'arguments', arguments, 4);
                                            return J$.X1(265, J$.Rt(257, J$.M(249, J$.R(233, 'inflect', inflect, 1), 'singularize', 0)(J$.R(241, 'this', this, 0))));
                                        } catch (J$e) {
                                            J$.Ex(1329, J$e);
                                        } finally {
                                            if (J$.Fr(1337))
                                                continue jalangiLabel386;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false, 273), 0));
                            J$.X1(457, J$.P(449, J$.G(321, J$.R(313, 'String', String, 2), 'prototype', 0), 'camelize', J$.T(441, function (first_letter_in_uppercase) {
                                jalangiLabel387:
                                    while (true) {
                                        try {
                                            J$.Fe(417, arguments.callee, this, arguments);
                                            arguments = J$.N(425, 'arguments', arguments, 4);
                                            first_letter_in_uppercase = J$.N(433, 'first_letter_in_uppercase', first_letter_in_uppercase, 4);
                                            if (J$.X1(1345, J$.C(8, J$.B(10, '==', J$.R(329, 'first_letter_in_uppercase', first_letter_in_uppercase, 0), J$.T(337, null, 25, false), 0)))) {
                                                J$.X1(361, first_letter_in_uppercase = J$.W(353, 'first_letter_in_uppercase', J$.T(345, true, 23, false), first_letter_in_uppercase, 0));
                                            }
                                            return J$.X1(409, J$.Rt(401, J$.M(393, J$.R(369, 'inflect', inflect, 1), 'camelize', 0)(J$.R(377, 'this', this, 0), J$.R(385, 'first_letter_in_uppercase', first_letter_in_uppercase, 0))));
                                        } catch (J$e) {
                                            J$.Ex(1353, J$e);
                                        } finally {
                                            if (J$.Fr(1361))
                                                continue jalangiLabel387;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false, 417), 0));
                            J$.X1(553, J$.P(545, J$.G(473, J$.R(465, 'String', String, 2), 'prototype', 0), 'capitalize', J$.T(537, function () {
                                jalangiLabel388:
                                    while (true) {
                                        try {
                                            J$.Fe(521, arguments.callee, this, arguments);
                                            arguments = J$.N(529, 'arguments', arguments, 4);
                                            return J$.X1(513, J$.Rt(505, J$.M(497, J$.R(481, 'inflect', inflect, 1), 'capitalize', 0)(J$.R(489, 'this', this, 0))));
                                        } catch (J$e) {
                                            J$.Ex(1369, J$e);
                                        } finally {
                                            if (J$.Fr(1377))
                                                continue jalangiLabel388;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false, 521), 0));
                            J$.X1(649, J$.P(641, J$.G(569, J$.R(561, 'String', String, 2), 'prototype', 0), 'decapitalize', J$.T(633, function () {
                                jalangiLabel389:
                                    while (true) {
                                        try {
                                            J$.Fe(617, arguments.callee, this, arguments);
                                            arguments = J$.N(625, 'arguments', arguments, 4);
                                            return J$.X1(609, J$.Rt(601, J$.M(593, J$.R(577, 'inflect', inflect, 1), 'decapitalize', 0)(J$.R(585, 'this', this, 0))));
                                        } catch (J$e) {
                                            J$.Ex(1385, J$e);
                                        } finally {
                                            if (J$.Fr(1393))
                                                continue jalangiLabel389;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false, 617), 0));
                            J$.X1(745, J$.P(737, J$.G(665, J$.R(657, 'String', String, 2), 'prototype', 0), 'titleize', J$.T(729, function () {
                                jalangiLabel390:
                                    while (true) {
                                        try {
                                            J$.Fe(713, arguments.callee, this, arguments);
                                            arguments = J$.N(721, 'arguments', arguments, 4);
                                            return J$.X1(705, J$.Rt(697, J$.M(689, J$.R(673, 'inflect', inflect, 1), 'titleize', 0)(J$.R(681, 'this', this, 0))));
                                        } catch (J$e) {
                                            J$.Ex(1401, J$e);
                                        } finally {
                                            if (J$.Fr(1409))
                                                continue jalangiLabel390;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false, 713), 0));
                            J$.X1(841, J$.P(833, J$.G(761, J$.R(753, 'String', String, 2), 'prototype', 0), 'underscore', J$.T(825, function () {
                                jalangiLabel391:
                                    while (true) {
                                        try {
                                            J$.Fe(809, arguments.callee, this, arguments);
                                            arguments = J$.N(817, 'arguments', arguments, 4);
                                            return J$.X1(801, J$.Rt(793, J$.M(785, J$.R(769, 'inflect', inflect, 1), 'underscore', 0)(J$.R(777, 'this', this, 0))));
                                        } catch (J$e) {
                                            J$.Ex(1417, J$e);
                                        } finally {
                                            if (J$.Fr(1425))
                                                continue jalangiLabel391;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false, 809), 0));
                            J$.X1(937, J$.P(929, J$.G(857, J$.R(849, 'String', String, 2), 'prototype', 0), 'dasherize', J$.T(921, function () {
                                jalangiLabel392:
                                    while (true) {
                                        try {
                                            J$.Fe(905, arguments.callee, this, arguments);
                                            arguments = J$.N(913, 'arguments', arguments, 4);
                                            return J$.X1(897, J$.Rt(889, J$.M(881, J$.R(865, 'inflect', inflect, 1), 'dasherize', 0)(J$.R(873, 'this', this, 0))));
                                        } catch (J$e) {
                                            J$.Ex(1433, J$e);
                                        } finally {
                                            if (J$.Fr(1441))
                                                continue jalangiLabel392;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false, 905), 0));
                            J$.X1(1089, J$.P(1081, J$.G(953, J$.R(945, 'String', String, 2), 'prototype', 0), 'parameterize', J$.T(1073, function (sep) {
                                jalangiLabel393:
                                    while (true) {
                                        try {
                                            J$.Fe(1049, arguments.callee, this, arguments);
                                            arguments = J$.N(1057, 'arguments', arguments, 4);
                                            sep = J$.N(1065, 'sep', sep, 4);
                                            if (J$.X1(1449, J$.C(16, J$.B(18, '==', J$.R(961, 'sep', sep, 0), J$.T(969, null, 25, false), 0)))) {
                                                J$.X1(993, sep = J$.W(985, 'sep', J$.T(977, '-', 21, false), sep, 0));
                                            }
                                            return J$.X1(1041, J$.Rt(1033, J$.M(1025, J$.R(1001, 'inflect', inflect, 1), 'parameterize', 0)(J$.R(1009, 'this', this, 0), J$.R(1017, 'sep', sep, 0))));
                                        } catch (J$e) {
                                            J$.Ex(1457, J$e);
                                        } finally {
                                            if (J$.Fr(1465))
                                                continue jalangiLabel393;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false, 1049), 0));
                            return J$.X1(1193, J$.Rt(1185, J$.P(1177, J$.G(1105, J$.R(1097, 'String', String, 2), 'prototype', 0), 'humanize', J$.T(1169, function () {
                                jalangiLabel394:
                                    while (true) {
                                        try {
                                            J$.Fe(1153, arguments.callee, this, arguments);
                                            arguments = J$.N(1161, 'arguments', arguments, 4);
                                            return J$.X1(1145, J$.Rt(1137, J$.M(1129, J$.R(1113, 'inflect', inflect, 1), 'humanize', 0)(J$.R(1121, 'this', this, 0))));
                                        } catch (J$e) {
                                            J$.Ex(1473, J$e);
                                        } finally {
                                            if (J$.Fr(1481))
                                                continue jalangiLabel394;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false, 1153), 0)));
                        } catch (J$e) {
                            J$.Ex(1489, J$e);
                        } finally {
                            if (J$.Fr(1497))
                                continue jalangiLabel395;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 1201), enableStringExtensions, 2));
            J$.X1(1265, J$.P(1257, J$.R(1241, 'exports', exports, 2), 'enableStringExtensions', J$.R(1249, 'enableStringExtensions', enableStringExtensions, 1), 0));
        } catch (J$e) {
            J$.Ex(1505, J$e);
        } finally {
            if (J$.Sr(1513)) {
                J$.L();
                continue jalangiLabel396;
            } else {
                J$.L();
                break jalangiLabel396;
            }
        }
    }
// JALANGI DO NOT INSTRUMENT
