// request/lib/cookies_jalangi_.js
J$.iids = {"8":[7,9,7,23],"9":[1,20,1,27],"10":[10,9,10,19],"16":[7,9,7,23],"17":[1,28,1,32],"18":[10,9,10,32],"24":[10,9,10,32],"25":[1,20,1,33],"32":[21,54,21,67],"33":[1,20,1,33],"41":[1,20,1,33],"49":[2,26,2,38],"57":[2,26,2,45],"65":[2,26,2,45],"73":[2,26,2,45],"81":[3,1,3,13],"89":[3,1,3,14],"97":[4,13,4,20],"105":[4,21,4,35],"113":[4,13,4,36],"121":[4,13,4,36],"129":[4,13,4,36],"137":[5,14,5,19],"145":[5,14,5,26],"153":[5,40,5,45],"161":[5,40,5,55],"169":[5,14,5,26],"177":[5,14,5,26],"185":[5,40,5,55],"193":[5,40,5,55],"201":[6,1,6,8],"209":[7,9,7,12],"217":[7,16,7,19],"225":[7,16,7,23],"233":[8,15,8,18],"241":[8,15,8,22],"249":[8,15,8,22],"257":[8,9,8,23],"265":[10,16,10,19],"273":[10,24,10,32],"281":[11,19,11,24],"289":[11,25,11,75],"297":[11,15,11,76],"305":[11,15,11,76],"313":[11,9,11,77],"321":[13,12,13,18],"329":[13,25,13,28],"337":[13,39,13,43],"345":[13,30,13,45],"353":[13,12,13,46],"355":[13,12,13,24],"361":[13,12,13,46],"369":[13,5,13,47],"377":[6,17,14,2],"385":[6,17,14,2],"393":[6,17,14,2],"401":[6,17,14,2],"409":[6,1,14,2],"417":[6,1,14,3],"425":[16,16,16,20],"433":[16,16,16,20],"441":[16,16,16,20],"449":[17,5,17,9],"457":[17,21,17,30],"465":[17,31,17,36],"473":[17,51,17,55],"481":[17,38,17,57],"489":[17,17,17,58],"497":[17,5,17,58],"505":[17,5,17,59],"513":[15,1,18,2],"521":[15,1,18,2],"529":[15,1,18,2],"537":[15,1,18,2],"545":[19,1,19,11],"553":[19,1,19,21],"561":[20,16,20,20],"569":[20,16,20,20],"577":[20,16,20,20],"585":[21,12,21,16],"593":[21,12,21,21],"601":[21,36,21,47],"609":[21,49,21,52],"617":[21,54,21,61],"625":[21,65,21,67],"633":[21,12,21,68],"635":[21,12,21,35],"641":[21,12,21,68],"649":[21,5,21,69],"657":[19,34,22,2],"665":[19,34,22,2],"673":[19,34,22,2],"681":[19,34,22,2],"689":[19,34,22,2],"697":[19,34,22,2],"705":[19,34,22,2],"713":[19,1,22,2],"721":[19,1,22,3],"729":[23,1,23,11],"737":[23,1,23,21],"745":[24,16,24,20],"753":[24,16,24,20],"761":[24,16,24,20],"769":[25,12,25,16],"777":[25,12,25,21],"785":[25,42,25,45],"793":[25,12,25,46],"795":[25,12,25,41],"801":[25,12,25,46],"809":[25,5,25,47],"817":[23,40,26,2],"825":[23,40,26,2],"833":[23,40,26,2],"841":[23,40,26,2],"849":[23,40,26,2],"857":[23,1,26,2],"865":[23,1,26,3],"873":[27,1,27,11],"881":[27,1,27,21],"889":[28,16,28,20],"897":[28,16,28,20],"905":[28,16,28,20],"913":[29,12,29,16],"921":[29,12,29,21],"929":[29,37,29,40],"937":[29,12,29,41],"939":[29,12,29,36],"945":[29,12,29,41],"953":[29,5,29,42],"961":[27,35,30,2],"969":[27,35,30,2],"977":[27,35,30,2],"985":[27,35,30,2],"993":[27,35,30,2],"1001":[27,1,30,2],"1009":[27,1,30,3],"1017":[31,1,31,8],"1025":[32,16,32,26],"1033":[32,27,32,32],"1041":[32,12,32,33],"1049":[32,12,32,33],"1057":[32,5,32,34],"1065":[31,15,33,2],"1073":[31,15,33,2],"1081":[31,15,33,2],"1089":[31,15,33,2],"1097":[31,1,33,2],"1105":[31,1,33,3],"1113":[1,1,33,3],"1121":[1,1,33,3],"1129":[1,1,33,3],"1137":[1,1,33,3],"1145":[1,1,33,3],"1153":[1,1,33,3],"1161":[15,1,18,2],"1169":[1,1,33,3],"1177":[7,5,9,6],"1185":[10,5,12,6],"1193":[6,17,14,2],"1201":[6,17,14,2],"1209":[15,1,18,2],"1217":[15,1,18,2],"1225":[19,34,22,2],"1233":[19,34,22,2],"1241":[23,40,26,2],"1249":[23,40,26,2],"1257":[27,35,30,2],"1265":[27,35,30,2],"1273":[31,15,33,2],"1281":[31,15,33,2],"1289":[1,1,33,3],"1297":[1,1,33,3],"nBranches":8,"originalCodeFileName":"/home/blake/ExpoSE/Targets/mapjam-geo-providers/node_modules/request/lib/cookies.js","instrumentedCodeFileName":"/home/blake/ExpoSE/Targets/mapjam-geo-providers/node_modules/request/lib/cookies_jalangi_.js","code":"'use strict'\n\nvar tough = require('tough-cookie')\n\nvar Cookie = tough.Cookie\n  , CookieJar = tough.CookieJar\n\n\nexports.parse = function(str) {\n  if (str && str.uri) {\n    str = str.uri\n  }\n  if (typeof str !== 'string') {\n    throw new Error('The cookie function only accepts STRING as param')\n  }\n  return Cookie.parse(str, {loose: true})\n}\n\n// Adapt the sometimes-Async api of tough.CookieJar to our requirements\nfunction RequestJar(store) {\n  var self = this\n  self._jar = new CookieJar(store, {looseMode: true})\n}\nRequestJar.prototype.setCookie = function(cookieOrStr, uri, options) {\n  var self = this\n  return self._jar.setCookieSync(cookieOrStr, uri, options || {})\n}\nRequestJar.prototype.getCookieString = function(uri) {\n  var self = this\n  return self._jar.getCookieStringSync(uri)\n}\nRequestJar.prototype.getCookies = function(uri) {\n  var self = this\n  return self._jar.getCookiesSync(uri)\n}\n\nexports.jar = function(store) {\n  return new RequestJar(store)\n}\n"};
jalangiLabel91:
    while (true) {
        try {
            J$.Se(1113, '/home/blake/ExpoSE/Targets/mapjam-geo-providers/node_modules/request/lib/cookies_jalangi_.js', '/home/blake/ExpoSE/Targets/mapjam-geo-providers/node_modules/request/lib/cookies.js');
            function RequestJar(store) {
                jalangiLabel86:
                    while (true) {
                        try {
                            J$.Fe(513, arguments.callee, this, arguments);
                            arguments = J$.N(521, 'arguments', arguments, 4);
                            store = J$.N(529, 'store', store, 4);
                            J$.N(537, 'self', self, 0);
                            var self = J$.X1(441, J$.W(433, 'self', J$.R(425, 'this', this, 0), self, 1));
                            J$.X1(505, J$.P(497, J$.R(449, 'self', self, 0), '_jar', J$.F(489, J$.R(457, 'CookieJar', CookieJar, 1), 1)(J$.R(465, 'store', store, 0), J$.T(481, {
                                looseMode: J$.T(473, true, 23, false)
                            }, 11, false)), 0));
                        } catch (J$e) {
                            J$.Ex(1209, J$e);
                        } finally {
                            if (J$.Fr(1217))
                                continue jalangiLabel86;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(1121, '__secret__S$', __secret__S$, 0);
            J$.N(1129, '__secret__traits__', __secret__traits__, 0);
            J$.N(1137, 'tough', tough, 0);
            J$.N(1145, 'Cookie', Cookie, 0);
            J$.N(1153, 'CookieJar', CookieJar, 0);
            RequestJar = J$.N(1169, 'RequestJar', J$.T(1161, RequestJar, 12, false, 513), 0);
            var __secret__S$ = J$.X1(41, J$.W(33, '__secret__S$', J$.F(25, J$.R(9, 'require', require, 2), 0)(J$.T(17, 'S$', 21, false)), __secret__S$, 3));
            var __secret__traits__ = J$.X1(73, J$.W(65, '__secret__traits__', J$.G(57, J$.R(49, '__secret__S$', __secret__S$, 1), 'Traits', 0), __secret__traits__, 3));
            J$.X1(89, J$.T(81, 'use strict', 21, false));
            var tough = J$.X1(129, J$.W(121, 'tough', J$.F(113, J$.R(97, 'require', require, 2), 0)(J$.T(105, 'tough-cookie', 21, false)), tough, 3));
            var Cookie = J$.X1(177, J$.W(169, 'Cookie', J$.G(145, J$.R(137, 'tough', tough, 1), 'Cookie', 0), Cookie, 3)), CookieJar = J$.X1(193, J$.W(185, 'CookieJar', J$.G(161, J$.R(153, 'tough', tough, 1), 'CookieJar', 0), CookieJar, 3));
            J$.X1(417, J$.P(409, J$.R(201, 'exports', exports, 2), 'parse', J$.T(401, function (str) {
                jalangiLabel85:
                    while (true) {
                        try {
                            J$.Fe(377, arguments.callee, this, arguments);
                            arguments = J$.N(385, 'arguments', arguments, 4);
                            str = J$.N(393, 'str', str, 4);
                            if (J$.X1(1177, J$.C(16, J$.C(8, J$.R(209, 'str', str, 0)) ? J$.G(225, J$.R(217, 'str', str, 0), 'uri', 0) : J$._()))) {
                                J$.X1(257, str = J$.W(249, 'str', J$.G(241, J$.R(233, 'str', str, 0), 'uri', 0), str, 0));
                            }
                            if (J$.X1(1185, J$.C(24, J$.B(18, '!==', J$.U(10, 'typeof', J$.R(265, 'str', str, 0)), J$.T(273, 'string', 21, false), 0)))) {
                                throw J$.X1(313, J$.Th(305, J$.F(297, J$.R(281, 'Error', Error, 2), 1)(J$.T(289, 'The cookie function only accepts STRING as param', 21, false))));
                            }
                            return J$.X1(369, J$.Rt(361, J$.M(353, J$.R(321, 'Cookie', Cookie, 1), 'parse', 0)(J$.R(329, 'str', str, 0), J$.T(345, {
                                loose: J$.T(337, true, 23, false)
                            }, 11, false))));
                        } catch (J$e) {
                            J$.Ex(1193, J$e);
                        } finally {
                            if (J$.Fr(1201))
                                continue jalangiLabel85;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 377), 0));
            J$.X1(721, J$.P(713, J$.G(553, J$.R(545, 'RequestJar', RequestJar, 1), 'prototype', 0), 'setCookie', J$.T(705, function (cookieOrStr, uri, options) {
                jalangiLabel87:
                    while (true) {
                        try {
                            J$.Fe(657, arguments.callee, this, arguments);
                            arguments = J$.N(665, 'arguments', arguments, 4);
                            cookieOrStr = J$.N(673, 'cookieOrStr', cookieOrStr, 4);
                            uri = J$.N(681, 'uri', uri, 4);
                            options = J$.N(689, 'options', options, 4);
                            J$.N(697, 'self', self, 0);
                            var self = J$.X1(577, J$.W(569, 'self', J$.R(561, 'this', this, 0), self, 1));
                            return J$.X1(649, J$.Rt(641, J$.M(633, J$.G(593, J$.R(585, 'self', self, 0), '_jar', 0), 'setCookieSync', 0)(J$.R(601, 'cookieOrStr', cookieOrStr, 0), J$.R(609, 'uri', uri, 0), J$.C(32, J$.R(617, 'options', options, 0)) ? J$._() : J$.T(625, {}, 11, false))));
                        } catch (J$e) {
                            J$.Ex(1225, J$e);
                        } finally {
                            if (J$.Fr(1233))
                                continue jalangiLabel87;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 657), 0));
            J$.X1(865, J$.P(857, J$.G(737, J$.R(729, 'RequestJar', RequestJar, 1), 'prototype', 0), 'getCookieString', J$.T(849, function (uri) {
                jalangiLabel88:
                    while (true) {
                        try {
                            J$.Fe(817, arguments.callee, this, arguments);
                            arguments = J$.N(825, 'arguments', arguments, 4);
                            uri = J$.N(833, 'uri', uri, 4);
                            J$.N(841, 'self', self, 0);
                            var self = J$.X1(761, J$.W(753, 'self', J$.R(745, 'this', this, 0), self, 1));
                            return J$.X1(809, J$.Rt(801, J$.M(793, J$.G(777, J$.R(769, 'self', self, 0), '_jar', 0), 'getCookieStringSync', 0)(J$.R(785, 'uri', uri, 0))));
                        } catch (J$e) {
                            J$.Ex(1241, J$e);
                        } finally {
                            if (J$.Fr(1249))
                                continue jalangiLabel88;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 817), 0));
            J$.X1(1009, J$.P(1001, J$.G(881, J$.R(873, 'RequestJar', RequestJar, 1), 'prototype', 0), 'getCookies', J$.T(993, function (uri) {
                jalangiLabel89:
                    while (true) {
                        try {
                            J$.Fe(961, arguments.callee, this, arguments);
                            arguments = J$.N(969, 'arguments', arguments, 4);
                            uri = J$.N(977, 'uri', uri, 4);
                            J$.N(985, 'self', self, 0);
                            var self = J$.X1(905, J$.W(897, 'self', J$.R(889, 'this', this, 0), self, 1));
                            return J$.X1(953, J$.Rt(945, J$.M(937, J$.G(921, J$.R(913, 'self', self, 0), '_jar', 0), 'getCookiesSync', 0)(J$.R(929, 'uri', uri, 0))));
                        } catch (J$e) {
                            J$.Ex(1257, J$e);
                        } finally {
                            if (J$.Fr(1265))
                                continue jalangiLabel89;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 961), 0));
            J$.X1(1105, J$.P(1097, J$.R(1017, 'exports', exports, 2), 'jar', J$.T(1089, function (store) {
                jalangiLabel90:
                    while (true) {
                        try {
                            J$.Fe(1065, arguments.callee, this, arguments);
                            arguments = J$.N(1073, 'arguments', arguments, 4);
                            store = J$.N(1081, 'store', store, 4);
                            return J$.X1(1057, J$.Rt(1049, J$.F(1041, J$.R(1025, 'RequestJar', RequestJar, 1), 1)(J$.R(1033, 'store', store, 0))));
                        } catch (J$e) {
                            J$.Ex(1273, J$e);
                        } finally {
                            if (J$.Fr(1281))
                                continue jalangiLabel90;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 1065), 0));
        } catch (J$e) {
            J$.Ex(1289, J$e);
        } finally {
            if (J$.Sr(1297)) {
                J$.L();
                continue jalangiLabel91;
            } else {
                J$.L();
                break jalangiLabel91;
            }
        }
    }
// JALANGI DO NOT INSTRUMENT
