// request/lib/helpers_jalangi_.js
J$.iids = {"8":[5,13,5,48],"9":[1,20,1,27],"10":[5,13,5,32],"16":[7,12,7,51],"17":[1,28,1,32],"18":[5,13,5,48],"24":[7,55,7,102],"25":[1,20,1,33],"26":[7,70,7,88],"32":[7,12,7,102],"33":[1,20,1,33],"34":[7,70,7,102],"40":[7,12,7,122],"41":[1,20,1,33],"48":[22,12,22,34],"49":[2,26,2,38],"56":[22,12,22,45],"57":[2,26,2,45],"64":[25,24,25,33],"65":[2,26,2,45],"73":[2,26,2,45],"81":[3,1,3,13],"89":[3,1,3,14],"97":[4,25,4,32],"105":[4,33,4,54],"113":[4,25,4,55],"121":[4,66,4,73],"129":[4,74,4,82],"137":[4,66,4,83],"145":[4,94,4,101],"153":[4,102,4,115],"161":[4,94,4,116],"169":[4,94,4,123],"177":[4,25,4,55],"185":[4,25,4,55],"193":[4,66,4,83],"201":[4,66,4,83],"209":[4,94,4,123],"217":[4,94,4,123],"225":[5,20,5,32],"233":[5,37,5,48],"241":[5,51,5,58],"249":[5,51,5,67],"257":[5,70,5,82],"265":[5,13,5,82],"273":[5,13,5,82],"281":[7,12,7,18],"289":[7,12,7,23],"297":[7,27,7,33],"305":[7,27,7,51],"313":[7,55,7,61],"321":[7,55,7,66],"329":[7,77,7,83],"337":[7,77,7,88],"345":[7,93,7,102],"353":[7,106,7,112],"361":[7,106,7,122],"369":[7,12,7,122],"377":[7,5,7,123],"385":[6,1,8,2],"393":[6,1,8,2],"401":[6,1,8,2],"409":[12,15,12,19],"417":[12,30,12,33],"425":[12,35,12,43],"433":[12,15,12,44],"435":[12,15,12,29],"441":[12,15,12,44],"449":[12,9,12,45],"457":[14,15,14,32],"465":[14,33,14,36],"473":[14,38,14,46],"481":[14,15,14,47],"489":[14,15,14,47],"497":[14,9,14,48],"505":[13,7,15,6],"513":[16,12,16,15],"521":[16,12,16,15],"529":[16,5,16,16],"537":[9,1,17,2],"545":[9,1,17,2],"553":[9,1,17,2],"561":[9,1,17,2],"569":[9,1,17,2],"577":[19,12,19,18],"585":[19,30,19,35],"593":[19,12,19,36],"595":[19,12,19,29],"601":[19,44,19,47],"609":[19,12,19,48],"611":[19,12,19,43],"617":[19,56,19,61],"625":[19,12,19,62],"627":[19,12,19,55],"633":[19,12,19,62],"641":[19,5,19,63],"649":[18,1,20,2],"657":[18,1,20,2],"665":[18,1,20,2],"673":[22,12,22,14],"681":[22,12,22,23],"689":[22,27,22,29],"697":[22,27,22,34],"705":[22,38,22,40],"713":[22,38,22,45],"721":[22,12,22,45],"729":[22,5,22,46],"737":[21,1,23,2],"745":[21,1,23,2],"753":[21,1,23,2],"761":[25,12,25,18],"769":[25,24,25,27],"777":[25,31,25,33],"785":[25,35,25,41],"793":[25,12,25,42],"795":[25,12,25,23],"801":[25,52,25,60],"809":[25,12,25,61],"811":[25,12,25,51],"817":[25,12,25,61],"825":[25,5,25,62],"833":[24,1,26,2],"841":[24,1,26,2],"849":[24,1,26,2],"857":[28,13,28,15],"865":[28,13,28,15],"873":[28,13,28,15],"881":[29,5,29,11],"889":[29,17,29,20],"897":[29,5,29,21],"899":[29,5,29,16],"905":[30,9,30,10],"913":[30,11,30,12],"921":[30,16,30,19],"929":[30,20,30,21],"937":[30,16,30,22],"945":[30,9,30,22],"953":[30,9,30,23],"961":[29,30,31,6],"969":[29,30,31,6],"977":[29,30,31,6],"985":[29,30,31,6],"993":[29,5,31,7],"995":[29,5,29,29],"1001":[29,5,31,8],"1009":[32,12,32,13],"1017":[32,12,32,13],"1025":[32,5,32,14],"1033":[27,1,33,2],"1041":[27,1,33,2],"1049":[27,1,33,2],"1057":[27,1,33,2],"1065":[35,19,35,26],"1073":[35,19,35,34],"1081":[35,43,35,46],"1089":[35,48,35,50],"1097":[35,19,35,51],"1099":[35,19,35,42],"1105":[35,58,35,61],"1113":[35,19,35,62],"1115":[35,19,35,57],"1121":[35,19,35,62],"1129":[35,19,35,62],"1137":[37,16,37,24],"1145":[37,25,37,32],"1153":[37,33,37,34],"1161":[37,25,37,35],"1169":[37,37,37,39],"1177":[37,16,37,40],"1185":[38,16,38,24],"1193":[38,25,38,32],"1201":[38,33,38,34],"1209":[38,25,38,35],"1217":[38,37,38,39],"1225":[38,16,38,40],"1233":[39,16,39,24],"1241":[39,25,39,32],"1249":[39,33,39,34],"1257":[39,25,39,35],"1265":[39,37,39,39],"1273":[39,16,39,40],"1281":[36,12,40,6],"1289":[36,12,40,6],"1297":[36,5,40,7],"1305":[34,1,41,2],"1313":[34,1,41,2],"1321":[34,1,41,2],"1329":[42,1,42,8],"1337":[42,33,42,54],"1345":[42,1,42,54],"1353":[42,1,42,55],"1361":[43,1,43,8],"1369":[43,25,43,38],"1377":[43,1,43,38],"1385":[43,1,43,39],"1393":[44,1,44,8],"1401":[44,15,44,18],"1409":[44,1,44,18],"1417":[44,1,44,19],"1425":[45,1,45,8],"1433":[45,24,45,36],"1441":[45,1,45,36],"1449":[45,1,45,37],"1457":[46,1,46,8],"1465":[46,20,46,28],"1473":[46,1,46,28],"1481":[46,1,46,29],"1489":[47,1,47,8],"1497":[47,16,47,20],"1505":[47,1,47,20],"1513":[47,1,47,21],"1521":[48,1,48,8],"1529":[48,19,48,26],"1537":[48,1,48,26],"1545":[48,1,48,27],"1553":[49,1,49,8],"1561":[49,17,49,22],"1569":[49,1,49,22],"1577":[49,1,49,23],"1585":[1,1,49,23],"1593":[1,1,49,23],"1601":[1,1,49,23],"1609":[1,1,49,23],"1617":[1,1,49,23],"1625":[1,1,49,23],"1633":[1,1,49,23],"1641":[6,1,8,2],"1649":[1,1,49,23],"1657":[9,1,17,2],"1665":[1,1,49,23],"1673":[18,1,20,2],"1681":[1,1,49,23],"1689":[21,1,23,2],"1697":[1,1,49,23],"1705":[24,1,26,2],"1713":[1,1,49,23],"1721":[27,1,33,2],"1729":[1,1,49,23],"1737":[34,1,41,2],"1745":[1,1,49,23],"1753":[5,13,5,82],"1761":[6,1,8,2],"1769":[6,1,8,2],"1777":[9,1,17,2],"1785":[9,1,17,2],"1793":[18,1,20,2],"1801":[18,1,20,2],"1809":[21,1,23,2],"1817":[21,1,23,2],"1825":[24,1,26,2],"1833":[24,1,26,2],"1841":[29,30,31,6],"1849":[29,30,31,6],"1857":[27,1,33,2],"1865":[27,1,33,2],"1873":[34,1,41,2],"1881":[34,1,41,2],"1889":[1,1,49,23],"1897":[1,1,49,23],"nBranches":16,"originalCodeFileName":"/home/blake/ExpoSE/Targets/mapjam-geo-providers/node_modules/request/lib/helpers.js","instrumentedCodeFileName":"/home/blake/ExpoSE/Targets/mapjam-geo-providers/node_modules/request/lib/helpers_jalangi_.js","code":"'use strict'\n\nvar jsonSafeStringify = require('json-stringify-safe')\n  , crypto = require('crypto')\n  , Buffer = require('safe-buffer').Buffer\n\nvar defer = typeof setImmediate === 'undefined'\n  ? process.nextTick\n  : setImmediate\n\nfunction paramsHaveRequestBody(params) {\n  return (\n    params.body ||\n    params.requestBodyStream ||\n    (params.json && typeof params.json !== 'boolean') ||\n    params.multipart\n  )\n}\n\nfunction safeStringify (obj, replacer) {\n  var ret\n  try {\n    ret = JSON.stringify(obj, replacer)\n  } catch (e) {\n    ret = jsonSafeStringify(obj, replacer)\n  }\n  return ret\n}\n\nfunction md5 (str) {\n  return crypto.createHash('md5').update(str).digest('hex')\n}\n\nfunction isReadStream (rs) {\n  return rs.readable && rs.path && rs.mode\n}\n\nfunction toBase64 (str) {\n  return Buffer.from(str || '', 'utf8').toString('base64')\n}\n\nfunction copy (obj) {\n  var o = {}\n  Object.keys(obj).forEach(function (i) {\n    o[i] = obj[i]\n  })\n  return o\n}\n\nfunction version () {\n  var numbers = process.version.replace('v', '').split('.')\n  return {\n    major: parseInt(numbers[0], 10),\n    minor: parseInt(numbers[1], 10),\n    patch: parseInt(numbers[2], 10)\n  }\n}\n\nexports.paramsHaveRequestBody = paramsHaveRequestBody\nexports.safeStringify         = safeStringify\nexports.md5                   = md5\nexports.isReadStream          = isReadStream\nexports.toBase64              = toBase64\nexports.copy                  = copy\nexports.version               = version\nexports.defer                 = defer\n"};
jalangiLabel193:
    while (true) {
        try {
            J$.Se(1585, '/home/blake/ExpoSE/Targets/mapjam-geo-providers/node_modules/request/lib/helpers_jalangi_.js', '/home/blake/ExpoSE/Targets/mapjam-geo-providers/node_modules/request/lib/helpers.js');
            function paramsHaveRequestBody(params) {
                jalangiLabel185:
                    while (true) {
                        try {
                            J$.Fe(385, arguments.callee, this, arguments);
                            arguments = J$.N(393, 'arguments', arguments, 4);
                            params = J$.N(401, 'params', params, 4);
                            return J$.X1(377, J$.Rt(369, J$.C(40, J$.C(32, J$.C(16, J$.G(289, J$.R(281, 'params', params, 0), 'body', 0)) ? J$._() : J$.G(305, J$.R(297, 'params', params, 0), 'requestBodyStream', 0)) ? J$._() : J$.C(24, J$.G(321, J$.R(313, 'params', params, 0), 'json', 0)) ? J$.B(34, '!==', J$.U(26, 'typeof', J$.G(337, J$.R(329, 'params', params, 0), 'json', 0)), J$.T(345, 'boolean', 21, false), 0) : J$._()) ? J$._() : J$.G(361, J$.R(353, 'params', params, 0), 'multipart', 0)));
                        } catch (J$e) {
                            J$.Ex(1761, J$e);
                        } finally {
                            if (J$.Fr(1769))
                                continue jalangiLabel185;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function safeStringify(obj, replacer) {
                jalangiLabel186:
                    while (true) {
                        try {
                            J$.Fe(537, arguments.callee, this, arguments);
                            arguments = J$.N(545, 'arguments', arguments, 4);
                            obj = J$.N(553, 'obj', obj, 4);
                            replacer = J$.N(561, 'replacer', replacer, 4);
                            J$.N(569, 'ret', ret, 0);
                            var ret;
                            try {
                                J$.X1(449, ret = J$.W(441, 'ret', J$.M(433, J$.R(409, 'JSON', JSON, 2), 'stringify', 0)(J$.R(417, 'obj', obj, 0), J$.R(425, 'replacer', replacer, 0)), ret, 0));
                            } catch (e) {
                                e = J$.N(505, 'e', e, 1);
                                J$.X1(497, ret = J$.W(489, 'ret', J$.F(481, J$.R(457, 'jsonSafeStringify', jsonSafeStringify, 1), 0)(J$.R(465, 'obj', obj, 0), J$.R(473, 'replacer', replacer, 0)), ret, 0));
                            }
                            return J$.X1(529, J$.Rt(521, J$.R(513, 'ret', ret, 0)));
                        } catch (J$e) {
                            J$.Ex(1777, J$e);
                        } finally {
                            if (J$.Fr(1785))
                                continue jalangiLabel186;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function md5(str) {
                jalangiLabel187:
                    while (true) {
                        try {
                            J$.Fe(649, arguments.callee, this, arguments);
                            arguments = J$.N(657, 'arguments', arguments, 4);
                            str = J$.N(665, 'str', str, 4);
                            return J$.X1(641, J$.Rt(633, J$.M(625, J$.M(609, J$.M(593, J$.R(577, 'crypto', crypto, 1), 'createHash', 0)(J$.T(585, 'md5', 21, false)), 'update', 0)(J$.R(601, 'str', str, 0)), 'digest', 0)(J$.T(617, 'hex', 21, false))));
                        } catch (J$e) {
                            J$.Ex(1793, J$e);
                        } finally {
                            if (J$.Fr(1801))
                                continue jalangiLabel187;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isReadStream(rs) {
                jalangiLabel188:
                    while (true) {
                        try {
                            J$.Fe(737, arguments.callee, this, arguments);
                            arguments = J$.N(745, 'arguments', arguments, 4);
                            rs = J$.N(753, 'rs', rs, 4);
                            return J$.X1(729, J$.Rt(721, J$.C(56, J$.C(48, J$.G(681, J$.R(673, 'rs', rs, 0), 'readable', 0)) ? J$.G(697, J$.R(689, 'rs', rs, 0), 'path', 0) : J$._()) ? J$.G(713, J$.R(705, 'rs', rs, 0), 'mode', 0) : J$._()));
                        } catch (J$e) {
                            J$.Ex(1809, J$e);
                        } finally {
                            if (J$.Fr(1817))
                                continue jalangiLabel188;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function toBase64(str) {
                jalangiLabel189:
                    while (true) {
                        try {
                            J$.Fe(833, arguments.callee, this, arguments);
                            arguments = J$.N(841, 'arguments', arguments, 4);
                            str = J$.N(849, 'str', str, 4);
                            return J$.X1(825, J$.Rt(817, J$.M(809, J$.M(793, J$.R(761, 'Buffer', Buffer, 1), 'from', 0)(J$.C(64, J$.R(769, 'str', str, 0)) ? J$._() : J$.T(777, '', 21, false), J$.T(785, 'utf8', 21, false)), 'toString', 0)(J$.T(801, 'base64', 21, false))));
                        } catch (J$e) {
                            J$.Ex(1825, J$e);
                        } finally {
                            if (J$.Fr(1833))
                                continue jalangiLabel189;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function copy(obj) {
                jalangiLabel191:
                    while (true) {
                        try {
                            J$.Fe(1033, arguments.callee, this, arguments);
                            arguments = J$.N(1041, 'arguments', arguments, 4);
                            obj = J$.N(1049, 'obj', obj, 4);
                            J$.N(1057, 'o', o, 0);
                            var o = J$.X1(873, J$.W(865, 'o', J$.T(857, {}, 11, false), o, 1));
                            J$.X1(1001, J$.M(993, J$.M(897, J$.R(881, 'Object', Object, 2), 'keys', 0)(J$.R(889, 'obj', obj, 0)), 'forEach', 0)(J$.T(985, function (i) {
                                jalangiLabel190:
                                    while (true) {
                                        try {
                                            J$.Fe(961, arguments.callee, this, arguments);
                                            arguments = J$.N(969, 'arguments', arguments, 4);
                                            i = J$.N(977, 'i', i, 4);
                                            J$.X1(953, J$.P(945, J$.R(905, 'o', o, 0), J$.R(913, 'i', i, 0), J$.G(937, J$.R(921, 'obj', obj, 0), J$.R(929, 'i', i, 0), 4), 2));
                                        } catch (J$e) {
                                            J$.Ex(1841, J$e);
                                        } finally {
                                            if (J$.Fr(1849))
                                                continue jalangiLabel190;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false, 961)));
                            return J$.X1(1025, J$.Rt(1017, J$.R(1009, 'o', o, 0)));
                        } catch (J$e) {
                            J$.Ex(1857, J$e);
                        } finally {
                            if (J$.Fr(1865))
                                continue jalangiLabel191;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function version() {
                jalangiLabel192:
                    while (true) {
                        try {
                            J$.Fe(1305, arguments.callee, this, arguments);
                            arguments = J$.N(1313, 'arguments', arguments, 4);
                            J$.N(1321, 'numbers', numbers, 0);
                            var numbers = J$.X1(1129, J$.W(1121, 'numbers', J$.M(1113, J$.M(1097, J$.G(1073, J$.R(1065, 'process', process, 2), 'version', 0), 'replace', 0)(J$.T(1081, 'v', 21, false), J$.T(1089, '', 21, false)), 'split', 0)(J$.T(1105, '.', 21, false)), numbers, 1));
                            return J$.X1(1297, J$.Rt(1289, J$.T(1281, {
                                major: J$.F(1177, J$.R(1137, 'parseInt', parseInt, 2), 0)(J$.G(1161, J$.R(1145, 'numbers', numbers, 0), J$.T(1153, 0, 22, false), 4), J$.T(1169, 10, 22, false)),
                                minor: J$.F(1225, J$.R(1185, 'parseInt', parseInt, 2), 0)(J$.G(1209, J$.R(1193, 'numbers', numbers, 0), J$.T(1201, 1, 22, false), 4), J$.T(1217, 10, 22, false)),
                                patch: J$.F(1273, J$.R(1233, 'parseInt', parseInt, 2), 0)(J$.G(1257, J$.R(1241, 'numbers', numbers, 0), J$.T(1249, 2, 22, false), 4), J$.T(1265, 10, 22, false))
                            }, 11, false)));
                        } catch (J$e) {
                            J$.Ex(1873, J$e);
                        } finally {
                            if (J$.Fr(1881))
                                continue jalangiLabel192;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(1593, '__secret__S$', __secret__S$, 0);
            J$.N(1601, '__secret__traits__', __secret__traits__, 0);
            J$.N(1609, 'jsonSafeStringify', jsonSafeStringify, 0);
            J$.N(1617, 'crypto', crypto, 0);
            J$.N(1625, 'Buffer', Buffer, 0);
            J$.N(1633, 'defer', defer, 0);
            paramsHaveRequestBody = J$.N(1649, 'paramsHaveRequestBody', J$.T(1641, paramsHaveRequestBody, 12, false, 385), 0);
            safeStringify = J$.N(1665, 'safeStringify', J$.T(1657, safeStringify, 12, false, 537), 0);
            md5 = J$.N(1681, 'md5', J$.T(1673, md5, 12, false, 649), 0);
            isReadStream = J$.N(1697, 'isReadStream', J$.T(1689, isReadStream, 12, false, 737), 0);
            toBase64 = J$.N(1713, 'toBase64', J$.T(1705, toBase64, 12, false, 833), 0);
            copy = J$.N(1729, 'copy', J$.T(1721, copy, 12, false, 1033), 0);
            version = J$.N(1745, 'version', J$.T(1737, version, 12, false, 1305), 0);
            var __secret__S$ = J$.X1(41, J$.W(33, '__secret__S$', J$.F(25, J$.R(9, 'require', require, 2), 0)(J$.T(17, 'S$', 21, false)), __secret__S$, 3));
            var __secret__traits__ = J$.X1(73, J$.W(65, '__secret__traits__', J$.G(57, J$.R(49, '__secret__S$', __secret__S$, 1), 'Traits', 0), __secret__traits__, 3));
            J$.X1(89, J$.T(81, 'use strict', 21, false));
            var jsonSafeStringify = J$.X1(185, J$.W(177, 'jsonSafeStringify', J$.F(113, J$.R(97, 'require', require, 2), 0)(J$.T(105, 'json-stringify-safe', 21, false)), jsonSafeStringify, 3)), crypto = J$.X1(201, J$.W(193, 'crypto', J$.F(137, J$.R(121, 'require', require, 2), 0)(J$.T(129, 'crypto', 21, false)), crypto, 3)), Buffer = J$.X1(217, J$.W(209, 'Buffer', J$.G(169, J$.F(161, J$.R(145, 'require', require, 2), 0)(J$.T(153, 'safe-buffer', 21, false)), 'Buffer', 0), Buffer, 3));
            var defer = J$.X1(273, J$.W(265, 'defer', J$.X1(1753, J$.C(8, J$.B(18, '===', J$.U(10, 'typeof', J$.I(typeof setImmediate === 'undefined' ? J$.R(225, 'setImmediate', undefined, 2) : J$.R(225, 'setImmediate', setImmediate, 2))), J$.T(233, 'undefined', 21, false), 0))) ? J$.G(249, J$.R(241, 'process', process, 2), 'nextTick', 0) : J$.R(257, 'setImmediate', setImmediate, 2), defer, 3));
            J$.X1(1353, J$.P(1345, J$.R(1329, 'exports', exports, 2), 'paramsHaveRequestBody', J$.R(1337, 'paramsHaveRequestBody', paramsHaveRequestBody, 1), 0));
            J$.X1(1385, J$.P(1377, J$.R(1361, 'exports', exports, 2), 'safeStringify', J$.R(1369, 'safeStringify', safeStringify, 1), 0));
            J$.X1(1417, J$.P(1409, J$.R(1393, 'exports', exports, 2), 'md5', J$.R(1401, 'md5', md5, 1), 0));
            J$.X1(1449, J$.P(1441, J$.R(1425, 'exports', exports, 2), 'isReadStream', J$.R(1433, 'isReadStream', isReadStream, 1), 0));
            J$.X1(1481, J$.P(1473, J$.R(1457, 'exports', exports, 2), 'toBase64', J$.R(1465, 'toBase64', toBase64, 1), 0));
            J$.X1(1513, J$.P(1505, J$.R(1489, 'exports', exports, 2), 'copy', J$.R(1497, 'copy', copy, 1), 0));
            J$.X1(1545, J$.P(1537, J$.R(1521, 'exports', exports, 2), 'version', J$.R(1529, 'version', version, 1), 0));
            J$.X1(1577, J$.P(1569, J$.R(1553, 'exports', exports, 2), 'defer', J$.R(1561, 'defer', defer, 1), 0));
        } catch (J$e) {
            J$.Ex(1889, J$e);
        } finally {
            if (J$.Sr(1897)) {
                J$.L();
                continue jalangiLabel193;
            } else {
                J$.L();
                break jalangiLabel193;
            }
        }
    }
// JALANGI DO NOT INSTRUMENT
