// numbro/languages/sl_jalangi_.js
J$.iids = {"8":[23,9,23,56],"9":[1,20,1,27],"10":[23,9,23,22],"16":[23,9,23,56],"17":[1,28,1,32],"18":[23,9,23,38],"24":[26,9,26,55],"25":[1,20,1,33],"26":[26,9,26,22],"32":[26,9,26,80],"33":[1,20,1,33],"34":[26,9,26,38],"40":[26,9,26,80],"41":[1,20,1,33],"49":[2,26,2,38],"57":[2,26,2,45],"65":[2,26,2,45],"73":[2,26,2,45],"81":[4,5,4,17],"89":[4,5,4,18],"97":[6,25,6,29],"105":[7,22,7,26],"113":[9,24,9,27],"121":[10,22,10,25],"129":[8,21,11,10],"137":[13,23,13,29],"145":[14,22,14,28],"153":[15,22,15,25],"161":[16,23,16,26],"169":[12,24,17,10],"177":[19,20,19,23],"185":[19,20,19,23],"193":[19,13,19,24],"201":[18,18,20,10],"209":[18,18,20,10],"217":[18,18,20,10],"225":[21,29,21,37],"233":[21,19,21,39],"241":[5,20,22,6],"249":[5,20,22,6],"257":[5,20,22,6],"265":[23,16,23,22],"273":[23,27,23,38],"281":[23,42,23,48],"289":[23,42,23,56],"297":[24,9,24,15],"305":[24,26,24,34],"313":[24,9,24,34],"321":[24,9,24,35],"329":[26,16,26,22],"337":[26,27,26,38],"345":[26,42,26,48],"353":[26,42,26,55],"361":[26,59,26,65],"369":[26,59,26,72],"377":[26,59,26,80],"385":[27,9,27,15],"393":[27,9,27,22],"401":[27,31,27,35],"409":[27,37,27,45],"417":[27,9,27,46],"419":[27,9,27,30],"425":[27,9,27,47],"433":[3,2,29,2],"441":[3,2,29,2],"449":[3,2,29,2],"457":[3,2,29,2],"465":[3,2,29,4],"473":[3,1,29,6],"481":[1,1,29,6],"489":[1,1,29,6],"497":[1,1,29,6],"505":[18,18,20,10],"513":[18,18,20,10],"521":[23,5,25,6],"529":[26,5,28,6],"537":[3,2,29,2],"545":[3,2,29,2],"553":[1,1,29,6],"561":[1,1,29,6],"nBranches":10,"originalCodeFileName":"/home/blake/ExpoSE/Targets/numbro/node_modules/numbro/languages/sl.js","instrumentedCodeFileName":"/home/blake/ExpoSE/Targets/numbro/node_modules/numbro/languages/sl_jalangi_.js","code":"/*!\n * numbro.js language configuration\n * language : Slovene\n * locale: Slovenia\n * author : Tim McIntosh (StayinFront NZ)\n */\n(function () {\n    'use strict';\n\n    var language = {\n        langLocaleCode: 'sl',\n        cultureCode: 'sl',\n        delimiters: {\n            thousands: '.',\n            decimal: ','\n        },\n        abbreviations: {\n            thousand: 'tis.',\n            million: 'mil.',\n            billion: 'b',\n            trillion: 't'\n        },\n        ordinal: function () {\n            return '.';\n        },\n        currency: {\n            symbol: '€'\n        }\n    };\n\n    // Node\n    if (typeof module !== 'undefined' && module.exports) {\n        module.exports = language;\n    }\n    // Browser\n    if (typeof window !== 'undefined' && window.numbro && window.numbro.culture) {\n        window.numbro.culture('sl', language);\n    }\n}());\n"};
jalangiLabel257:
    while (true) {
        try {
            J$.Se(481, '/home/blake/ExpoSE/Targets/numbro/node_modules/numbro/languages/sl_jalangi_.js', '/home/blake/ExpoSE/Targets/numbro/node_modules/numbro/languages/sl.js');
            J$.N(489, '__secret__S$', __secret__S$, 0);
            J$.N(497, '__secret__traits__', __secret__traits__, 0);
            var __secret__S$ = J$.X1(41, J$.W(33, '__secret__S$', J$.F(25, J$.R(9, 'require', require, 2), 0)(J$.T(17, 'S$', 21, false)), __secret__S$, 3));
            var __secret__traits__ = J$.X1(73, J$.W(65, '__secret__traits__', J$.G(57, J$.R(49, '__secret__S$', __secret__S$, 1), 'Traits', 0), __secret__traits__, 3));
            J$.X1(473, J$.F(465, J$.T(457, function () {
                jalangiLabel256:
                    while (true) {
                        try {
                            J$.Fe(433, arguments.callee, this, arguments);
                            arguments = J$.N(441, 'arguments', arguments, 4);
                            J$.N(449, 'language', language, 0);
                            J$.X1(89, J$.T(81, 'use strict', 21, false));
                            var language = J$.X1(257, J$.W(249, 'language', J$.T(241, {
                                langLocaleCode: J$.T(97, 'sl', 21, false),
                                cultureCode: J$.T(105, 'sl', 21, false),
                                delimiters: J$.T(129, {
                                    thousands: J$.T(113, '.', 21, false),
                                    decimal: J$.T(121, ',', 21, false)
                                }, 11, false),
                                abbreviations: J$.T(169, {
                                    thousand: J$.T(137, 'tis.', 21, false),
                                    million: J$.T(145, 'mil.', 21, false),
                                    billion: J$.T(153, 'b', 21, false),
                                    trillion: J$.T(161, 't', 21, false)
                                }, 11, false),
                                ordinal: J$.T(217, function () {
                                    jalangiLabel255:
                                        while (true) {
                                            try {
                                                J$.Fe(201, arguments.callee, this, arguments);
                                                arguments = J$.N(209, 'arguments', arguments, 4);
                                                return J$.X1(193, J$.Rt(185, J$.T(177, '.', 21, false)));
                                            } catch (J$e) {
                                                J$.Ex(505, J$e);
                                            } finally {
                                                if (J$.Fr(513))
                                                    continue jalangiLabel255;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false, 201),
                                currency: J$.T(233, {
                                    symbol: J$.T(225, '\u20AC', 21, false)
                                }, 11, false)
                            }, 11, false), language, 1));
                            if (J$.X1(521, J$.C(16, J$.C(8, J$.B(18, '!==', J$.U(10, 'typeof', J$.I(typeof module === 'undefined' ? J$.R(265, 'module', undefined, 2) : J$.R(265, 'module', module, 2))), J$.T(273, 'undefined', 21, false), 0)) ? J$.G(289, J$.R(281, 'module', module, 2), 'exports', 0) : J$._()))) {
                                J$.X1(321, J$.P(313, J$.R(297, 'module', module, 2), 'exports', J$.R(305, 'language', language, 0), 0));
                            }
                            if (J$.X1(529, J$.C(40, J$.C(32, J$.C(24, J$.B(34, '!==', J$.U(26, 'typeof', J$.I(typeof window === 'undefined' ? J$.R(329, 'window', undefined, 2) : J$.R(329, 'window', window, 2))), J$.T(337, 'undefined', 21, false), 0)) ? J$.G(353, J$.R(345, 'window', window, 2), 'numbro', 0) : J$._()) ? J$.G(377, J$.G(369, J$.R(361, 'window', window, 2), 'numbro', 0), 'culture', 0) : J$._()))) {
                                J$.X1(425, J$.M(417, J$.G(393, J$.R(385, 'window', window, 2), 'numbro', 0), 'culture', 0)(J$.T(401, 'sl', 21, false), J$.R(409, 'language', language, 0)));
                            }
                        } catch (J$e) {
                            J$.Ex(537, J$e);
                        } finally {
                            if (J$.Fr(545))
                                continue jalangiLabel256;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 433), 0)());
        } catch (J$e) {
            J$.Ex(553, J$e);
        } finally {
            if (J$.Sr(561)) {
                J$.L();
                continue jalangiLabel257;
            } else {
                J$.L();
                break jalangiLabel257;
            }
        }
    }
// JALANGI DO NOT INSTRUMENT
