// generator-hughes/generators/app/mixins/database/writing.js
'use strict';

var yeoman = require('yeoman-generator'),
    chalk = require('chalk'),
    yosay = require('yosay'),
    fs = require('fs.extra'),
		mysql = require('mysql'),
		exec = require('child_process').execSync;

module.exports = {
	database: function() {
		
		var dbContext = {
				// Database info
				dbName: this.dbName,
				dbUser: this.dbUser,
				dbPass: this.dbPass
			}
		
		var done = this.async();
		var self = this;
		
		// Set database info
		var connection  = mysql.createConnection({
			host     : 'localhost',
			user     : 'root',
			password : ''
		});
        
		// Create database
		connection.connect(function (err) {
			if (err) {
				self.log.error('Error connecting to MySQL, please create the database manually');
				done();
			} else {
				connection.query('CREATE DATABASE ' + self.dbName, function (err, result) {
					self.log.create(self.dbName);
					connection.end(function () {
						done();
					});
				});
			}
          
           
			// Create user
			if (err) {
				self.log.error('Error connecting to MySQL, please create the user manually');
				done();
			} else {
				connection.query('CREATE USER ' + self.dbUser+'@localhost', function (err, result) {
					self.log.create(self.dbUser);
					connection.end(function () {
						done();
					});
				});
			}
          
			// Grant permissions
			if (err) {
				self.log.error('Error connecting to MySQL, please create the user manually');
				done();
			} else {
				connection.query('GRANT ALL PRIVILEGES ON ' + self.dbName + '. * TO ' + self.dbUser + '@localhost', function (err, result) {
					connection.end(function () {
						done();
					});
				});
			}
		});
	
}}
