// bless/lib/index_jalangi_.js
J$.iids = {"8":[12,12,12,33],"9":[1,20,1,27],"16":[12,12,12,33],"17":[1,28,1,32],"24":[16,9,16,46],"25":[1,20,1,33],"33":[1,20,1,33],"41":[1,20,1,33],"49":[2,26,2,38],"57":[2,26,2,45],"65":[2,26,2,45],"73":[2,26,2,45],"81":[3,1,3,13],"89":[3,1,3,14],"97":[4,1,4,7],"105":[4,23,4,30],"113":[4,32,4,44],"121":[4,55,4,59],"129":[4,46,4,61],"137":[4,1,4,62],"139":[4,1,4,22],"145":[4,1,4,63],"153":[5,18,5,25],"161":[5,26,5,38],"169":[5,18,5,39],"177":[5,18,5,39],"185":[5,18,5,39],"193":[6,19,6,41],"201":[6,42,6,52],"209":[6,19,6,53],"217":[6,19,6,53],"225":[6,19,6,53],"233":[7,14,7,21],"241":[7,22,7,31],"249":[7,14,7,32],"257":[7,14,7,32],"265":[7,14,7,32],"273":[8,15,8,37],"281":[8,38,8,44],"289":[8,15,8,45],"297":[8,15,8,45],"305":[8,15,8,45],"313":[9,15,9,22],"321":[9,23,9,31],"329":[9,15,9,32],"337":[9,15,9,32],"345":[9,15,9,32],"353":[10,16,10,38],"361":[10,39,10,46],"369":[10,16,10,47],"377":[10,16,10,47],"385":[10,16,10,47],"393":[12,12,12,15],"401":[12,19,12,22],"409":[12,19,12,33],"417":[12,36,12,39],"425":[12,53,12,56],"433":[12,42,12,58],"441":[12,12,12,58],"449":[12,5,12,59],"457":[11,1,13,2],"465":[11,1,13,2],"473":[11,1,13,2],"481":[15,41,15,46],"489":[15,27,15,48],"497":[15,27,15,48],"505":[15,27,15,48],"513":[16,9,16,17],"521":[16,9,16,25],"529":[16,38,16,45],"537":[16,9,16,46],"539":[16,9,16,37],"545":[17,19,17,21],"553":[17,19,17,21],"561":[17,9,17,22],"569":[19,12,19,20],"577":[19,12,19,28],"585":[19,38,19,45],"593":[19,47,19,62],"601":[19,12,19,63],"603":[19,12,19,37],"609":[19,12,19,63],"617":[19,5,19,64],"625":[14,1,20,2],"633":[14,1,20,2],"641":[14,1,20,2],"649":[14,1,20,2],"657":[22,13,22,14],"665":[22,16,22,23],"673":[22,16,22,31],"681":[22,13,22,14],"689":[22,33,22,37],"697":[22,39,22,53],"705":[22,54,22,61],"713":[22,39,22,62],"721":[22,12,22,63],"729":[22,12,22,63],"737":[22,5,22,64],"745":[21,1,23,2],"753":[21,1,23,2],"761":[21,1,23,2],"769":[21,1,23,2],"777":[25,12,25,23],"785":[25,12,25,31],"793":[25,41,25,49],"801":[25,63,25,69],"809":[25,51,25,71],"817":[25,12,25,72],"819":[25,12,25,40],"825":[26,28,26,36],"833":[26,28,26,44],"841":[26,54,26,61],"849":[26,73,26,81],"857":[26,63,26,83],"865":[26,28,26,84],"867":[26,28,26,53],"873":[26,28,26,84],"881":[26,28,26,84],"889":[27,16,27,21],"897":[27,22,27,26],"905":[27,28,27,40],"913":[27,16,27,41],"921":[27,16,27,41],"929":[27,9,27,42],"937":[25,78,28,6],"945":[25,78,28,6],"953":[25,78,28,6],"961":[25,78,28,6],"969":[25,78,28,6],"977":[25,12,28,7],"979":[25,12,25,77],"985":[25,12,28,7],"993":[25,5,28,8],"1001":[24,1,29,2],"1009":[24,1,29,2],"1017":[24,1,29,2],"1025":[24,1,29,2],"1033":[30,1,30,8],"1041":[31,12,31,17],"1049":[32,16,32,25],"1057":[30,19,33,2],"1065":[30,1,33,2],"1073":[30,1,33,3],"1081":[34,1,34,7],"1089":[34,18,34,25],"1097":[34,26,34,35],"1105":[34,18,34,36],"1113":[34,1,34,36],"1121":[34,1,34,37],"1129":[1,1,34,37],"1137":[1,1,34,37],"1145":[1,1,34,37],"1153":[1,1,34,37],"1161":[1,1,34,37],"1169":[1,1,34,37],"1177":[1,1,34,37],"1185":[1,1,34,37],"1193":[1,1,34,37],"1201":[11,1,13,2],"1209":[1,1,34,37],"1217":[14,1,20,2],"1225":[1,1,34,37],"1233":[21,1,23,2],"1241":[1,1,34,37],"1249":[24,1,29,2],"1257":[1,1,34,37],"1265":[12,12,12,58],"1273":[11,1,13,2],"1281":[11,1,13,2],"1289":[16,5,18,6],"1297":[14,1,20,2],"1305":[14,1,20,2],"1313":[21,1,23,2],"1321":[21,1,23,2],"1329":[25,78,28,6],"1337":[25,78,28,6],"1345":[24,1,29,2],"1353":[24,1,29,2],"1361":[1,1,34,37],"1369":[1,1,34,37],"nBranches":6,"originalCodeFileName":"/home/blake/ExpoSE/Targets/bless/node_modules/bless/lib/index.js","instrumentedCodeFileName":"/home/blake/ExpoSE/Targets/bless/node_modules/bless/lib/index_jalangi_.js","code":"'use strict';\n\nObject.defineProperty(exports, \"__esModule\", {\n  value: true\n});\n\nvar _fsPromise = require('fs-promise');\n\nvar _fsPromise2 = _interopRequireDefault(_fsPromise);\n\nvar _chunk = require('./chunk');\n\nvar _chunk2 = _interopRequireDefault(_chunk);\n\nvar _lodash = require('lodash');\n\nvar _lodash2 = _interopRequireDefault(_lodash);\n\nfunction _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }\n\nfunction defaultOptions(options) {\n  var _defaultOptions = {\n    sourcemaps: false\n  };\n\n  if (_lodash2.default.isUndefined(options)) {\n    options = {};\n  }\n\n  return _lodash2.default.defaults(options, _defaultOptions);\n}\n\nfunction chunk(code, options) {\n  return (0, _chunk2.default)(code, defaultOptions(options));\n}\n\nfunction chunkFile(filepath, options) {\n  return _fsPromise2.default.readFile(filepath, { encoding: 'utf8' }).then(function (code) {\n    var chunkOptions = _lodash2.default.defaults(options, { source: filepath });\n    return chunk(code, chunkOptions);\n  });\n}\n\nexports.default = {\n  chunk: chunk,\n  chunkFile: chunkFile\n};\nmodule.exports = exports['default'];"};
jalangiLabel48:
    while (true) {
        try {
            J$.Se(1129, '/home/blake/ExpoSE/Targets/bless/node_modules/bless/lib/index_jalangi_.js', '/home/blake/ExpoSE/Targets/bless/node_modules/bless/lib/index.js');
            function _interopRequireDefault(obj) {
                jalangiLabel43:
                    while (true) {
                        try {
                            J$.Fe(457, arguments.callee, this, arguments);
                            arguments = J$.N(465, 'arguments', arguments, 4);
                            obj = J$.N(473, 'obj', obj, 4);
                            return J$.X1(449, J$.Rt(441, J$.X1(1265, J$.C(16, J$.C(8, J$.R(393, 'obj', obj, 0)) ? J$.G(409, J$.R(401, 'obj', obj, 0), '__esModule', 0) : J$._())) ? J$.R(417, 'obj', obj, 0) : J$.T(433, {
                                default: J$.R(425, 'obj', obj, 0)
                            }, 11, false)));
                        } catch (J$e) {
                            J$.Ex(1273, J$e);
                        } finally {
                            if (J$.Fr(1281))
                                continue jalangiLabel43;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function defaultOptions(options) {
                jalangiLabel44:
                    while (true) {
                        try {
                            J$.Fe(625, arguments.callee, this, arguments);
                            arguments = J$.N(633, 'arguments', arguments, 4);
                            options = J$.N(641, 'options', options, 4);
                            J$.N(649, '_defaultOptions', _defaultOptions, 0);
                            var _defaultOptions = J$.X1(505, J$.W(497, '_defaultOptions', J$.T(489, {
                                sourcemaps: J$.T(481, false, 23, false)
                            }, 11, false), _defaultOptions, 1));
                            if (J$.X1(1289, J$.C(24, J$.M(537, J$.G(521, J$.R(513, '_lodash2', _lodash2, 1), 'default', 0), 'isUndefined', 0)(J$.R(529, 'options', options, 0))))) {
                                J$.X1(561, options = J$.W(553, 'options', J$.T(545, {}, 11, false), options, 0));
                            }
                            return J$.X1(617, J$.Rt(609, J$.M(601, J$.G(577, J$.R(569, '_lodash2', _lodash2, 1), 'default', 0), 'defaults', 0)(J$.R(585, 'options', options, 0), J$.R(593, '_defaultOptions', _defaultOptions, 0))));
                        } catch (J$e) {
                            J$.Ex(1297, J$e);
                        } finally {
                            if (J$.Fr(1305))
                                continue jalangiLabel44;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function chunk(code, options) {
                jalangiLabel45:
                    while (true) {
                        try {
                            J$.Fe(745, arguments.callee, this, arguments);
                            arguments = J$.N(753, 'arguments', arguments, 4);
                            code = J$.N(761, 'code', code, 4);
                            options = J$.N(769, 'options', options, 4);
                            return J$.X1(737, J$.Rt(729, J$.F(721, (J$.X1(681, J$.T(657, 0, 22, false)), J$.G(673, J$.R(665, '_chunk2', _chunk2, 1), 'default', 0)), 0)(J$.R(689, 'code', code, 0), J$.F(713, J$.R(697, 'defaultOptions', defaultOptions, 1), 0)(J$.R(705, 'options', options, 0)))));
                        } catch (J$e) {
                            J$.Ex(1313, J$e);
                        } finally {
                            if (J$.Fr(1321))
                                continue jalangiLabel45;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function chunkFile(filepath, options) {
                jalangiLabel47:
                    while (true) {
                        try {
                            J$.Fe(1001, arguments.callee, this, arguments);
                            arguments = J$.N(1009, 'arguments', arguments, 4);
                            filepath = J$.N(1017, 'filepath', filepath, 4);
                            options = J$.N(1025, 'options', options, 4);
                            return J$.X1(993, J$.Rt(985, J$.M(977, J$.M(817, J$.G(785, J$.R(777, '_fsPromise2', _fsPromise2, 1), 'default', 0), 'readFile', 0)(J$.R(793, 'filepath', filepath, 0), J$.T(809, {
                                encoding: J$.T(801, 'utf8', 21, false)
                            }, 11, false)), 'then', 0)(J$.T(969, function (code) {
                                jalangiLabel46:
                                    while (true) {
                                        try {
                                            J$.Fe(937, arguments.callee, this, arguments);
                                            arguments = J$.N(945, 'arguments', arguments, 4);
                                            code = J$.N(953, 'code', code, 4);
                                            J$.N(961, 'chunkOptions', chunkOptions, 0);
                                            var chunkOptions = J$.X1(881, J$.W(873, 'chunkOptions', J$.M(865, J$.G(833, J$.R(825, '_lodash2', _lodash2, 1), 'default', 0), 'defaults', 0)(J$.R(841, 'options', options, 0), J$.T(857, {
                                                source: J$.R(849, 'filepath', filepath, 0)
                                            }, 11, false)), chunkOptions, 1));
                                            return J$.X1(929, J$.Rt(921, J$.F(913, J$.R(889, 'chunk', chunk, 1), 0)(J$.R(897, 'code', code, 0), J$.R(905, 'chunkOptions', chunkOptions, 0))));
                                        } catch (J$e) {
                                            J$.Ex(1329, J$e);
                                        } finally {
                                            if (J$.Fr(1337))
                                                continue jalangiLabel46;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false, 937))));
                        } catch (J$e) {
                            J$.Ex(1345, J$e);
                        } finally {
                            if (J$.Fr(1353))
                                continue jalangiLabel47;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(1137, '__secret__S$', __secret__S$, 0);
            J$.N(1145, '__secret__traits__', __secret__traits__, 0);
            J$.N(1153, '_fsPromise', _fsPromise, 0);
            J$.N(1161, '_fsPromise2', _fsPromise2, 0);
            J$.N(1169, '_chunk', _chunk, 0);
            J$.N(1177, '_chunk2', _chunk2, 0);
            J$.N(1185, '_lodash', _lodash, 0);
            J$.N(1193, '_lodash2', _lodash2, 0);
            _interopRequireDefault = J$.N(1209, '_interopRequireDefault', J$.T(1201, _interopRequireDefault, 12, false, 457), 0);
            defaultOptions = J$.N(1225, 'defaultOptions', J$.T(1217, defaultOptions, 12, false, 625), 0);
            chunk = J$.N(1241, 'chunk', J$.T(1233, chunk, 12, false, 745), 0);
            chunkFile = J$.N(1257, 'chunkFile', J$.T(1249, chunkFile, 12, false, 1001), 0);
            var __secret__S$ = J$.X1(41, J$.W(33, '__secret__S$', J$.F(25, J$.R(9, 'require', require, 2), 0)(J$.T(17, 'S$', 21, false)), __secret__S$, 3));
            var __secret__traits__ = J$.X1(73, J$.W(65, '__secret__traits__', J$.G(57, J$.R(49, '__secret__S$', __secret__S$, 1), 'Traits', 0), __secret__traits__, 3));
            J$.X1(89, J$.T(81, 'use strict', 21, false));
            J$.X1(145, J$.M(137, J$.R(97, 'Object', Object, 2), 'defineProperty', 0)(J$.R(105, 'exports', exports, 2), J$.T(113, '__esModule', 21, false), J$.T(129, {
                value: J$.T(121, true, 23, false)
            }, 11, false)));
            var _fsPromise = J$.X1(185, J$.W(177, '_fsPromise', J$.F(169, J$.R(153, 'require', require, 2), 0)(J$.T(161, 'fs-promise', 21, false)), _fsPromise, 3));
            var _fsPromise2 = J$.X1(225, J$.W(217, '_fsPromise2', J$.F(209, J$.R(193, '_interopRequireDefault', _interopRequireDefault, 1), 0)(J$.R(201, '_fsPromise', _fsPromise, 1)), _fsPromise2, 3));
            var _chunk = J$.X1(265, J$.W(257, '_chunk', J$.F(249, J$.R(233, 'require', require, 2), 0)(J$.T(241, './chunk', 21, false)), _chunk, 3));
            var _chunk2 = J$.X1(305, J$.W(297, '_chunk2', J$.F(289, J$.R(273, '_interopRequireDefault', _interopRequireDefault, 1), 0)(J$.R(281, '_chunk', _chunk, 1)), _chunk2, 3));
            var _lodash = J$.X1(345, J$.W(337, '_lodash', J$.F(329, J$.R(313, 'require', require, 2), 0)(J$.T(321, 'lodash', 21, false)), _lodash, 3));
            var _lodash2 = J$.X1(385, J$.W(377, '_lodash2', J$.F(369, J$.R(353, '_interopRequireDefault', _interopRequireDefault, 1), 0)(J$.R(361, '_lodash', _lodash, 1)), _lodash2, 3));
            J$.X1(1073, J$.P(1065, J$.R(1033, 'exports', exports, 2), 'default', J$.T(1057, {
                chunk: J$.R(1041, 'chunk', chunk, 1),
                chunkFile: J$.R(1049, 'chunkFile', chunkFile, 1)
            }, 11, false), 0));
            J$.X1(1121, J$.P(1113, J$.R(1081, 'module', module, 2), 'exports', J$.G(1105, J$.R(1089, 'exports', exports, 2), J$.T(1097, 'default', 21, false), 4), 0));
        } catch (J$e) {
            J$.Ex(1361, J$e);
        } finally {
            if (J$.Sr(1369)) {
                J$.L();
                continue jalangiLabel48;
            } else {
                J$.L();
                break jalangiLabel48;
            }
        }
    }
// JALANGI DO NOT INSTRUMENT
