// xml2js/lib/defaults_jalangi_.js
J$.iids = {"9":[1,20,1,27],"17":[1,28,1,32],"25":[1,20,1,33],"33":[1,20,1,33],"41":[1,20,1,33],"49":[2,26,2,38],"57":[2,26,2,45],"65":[2,26,2,45],"73":[2,26,2,45],"81":[4,5,4,12],"89":[6,30,6,35],"97":[7,19,7,23],"105":[8,24,8,28],"113":[9,28,9,33],"121":[10,22,10,25],"129":[11,22,11,25],"137":[12,28,12,33],"145":[13,26,13,31],"153":[14,25,14,30],"161":[15,27,15,32],"169":[16,24,16,28],"177":[17,20,17,25],"185":[18,31,18,36],"193":[19,23,19,27],"201":[20,30,20,35],"209":[21,32,21,37],"217":[22,20,22,25],"225":[23,21,23,25],"233":[24,33,24,37],"241":[25,34,25,38],"249":[26,32,26,36],"257":[27,30,27,34],"265":[28,23,28,25],"273":[5,16,29,10],"281":[31,30,31,35],"289":[32,19,32,24],"297":[33,24,33,29],"305":[34,28,34,33],"313":[35,22,35,25],"321":[36,22,36,25],"329":[37,28,37,32],"337":[38,26,38,31],"345":[39,25,39,30],"353":[40,27,40,31],"361":[41,24,41,28],"369":[42,20,42,25],"377":[43,31,43,36],"385":[44,36,44,41],"393":[45,23,45,27],"401":[46,30,46,35],"409":[47,32,47,37],"417":[48,20,48,25],"425":[49,21,49,25],"433":[50,33,50,37],"441":[51,34,51,38],"449":[52,32,52,36],"457":[53,30,53,34],"465":[54,23,54,29],"473":[56,28,56,33],"481":[57,29,57,36],"489":[58,31,58,35],"497":[55,21,59,14],"505":[60,22,60,26],"513":[62,27,62,31],"521":[63,27,63,31],"529":[64,28,64,32],"537":[61,25,65,14],"545":[66,23,66,28],"553":[67,24,67,29],"561":[68,23,68,25],"569":[69,20,69,25],"577":[30,16,70,10],"585":[4,24,71,6],"593":[4,5,71,6],"601":[4,5,71,7],"609":[3,2,72,2],"617":[3,2,72,2],"625":[3,2,72,2],"633":[72,8,72,12],"641":[3,2,72,13],"643":[3,2,72,7],"649":[3,1,72,15],"657":[1,1,72,15],"665":[1,1,72,15],"673":[1,1,72,15],"681":[3,2,72,2],"689":[3,2,72,2],"697":[1,1,72,15],"705":[1,1,72,15],"nBranches":0,"originalCodeFileName":"/home/blake/ExpoSE/Targets/xero-extension/node_modules/xml2js/lib/defaults.js","instrumentedCodeFileName":"/home/blake/ExpoSE/Targets/xero-extension/node_modules/xml2js/lib/defaults_jalangi_.js","code":"// Generated by CoffeeScript 1.12.7\n(function() {\n  exports.defaults = {\n    \"0.1\": {\n      explicitCharkey: false,\n      trim: true,\n      normalize: true,\n      normalizeTags: false,\n      attrkey: \"@\",\n      charkey: \"#\",\n      explicitArray: false,\n      ignoreAttrs: false,\n      mergeAttrs: false,\n      explicitRoot: false,\n      validator: null,\n      xmlns: false,\n      explicitChildren: false,\n      childkey: '@@',\n      charsAsChildren: false,\n      includeWhiteChars: false,\n      async: false,\n      strict: true,\n      attrNameProcessors: null,\n      attrValueProcessors: null,\n      tagNameProcessors: null,\n      valueProcessors: null,\n      emptyTag: ''\n    },\n    \"0.2\": {\n      explicitCharkey: false,\n      trim: false,\n      normalize: false,\n      normalizeTags: false,\n      attrkey: \"$\",\n      charkey: \"_\",\n      explicitArray: true,\n      ignoreAttrs: false,\n      mergeAttrs: false,\n      explicitRoot: true,\n      validator: null,\n      xmlns: false,\n      explicitChildren: false,\n      preserveChildrenOrder: false,\n      childkey: '$$',\n      charsAsChildren: false,\n      includeWhiteChars: false,\n      async: false,\n      strict: true,\n      attrNameProcessors: null,\n      attrValueProcessors: null,\n      tagNameProcessors: null,\n      valueProcessors: null,\n      rootName: 'root',\n      xmldec: {\n        'version': '1.0',\n        'encoding': 'UTF-8',\n        'standalone': true\n      },\n      doctype: null,\n      renderOpts: {\n        'pretty': true,\n        'indent': '  ',\n        'newline': '\\n'\n      },\n      headless: false,\n      chunkSize: 10000,\n      emptyTag: '',\n      cdata: false\n    }\n  };\n\n}).call(this);\n"};
jalangiLabel409:
    while (true) {
        try {
            J$.Se(657, '/home/blake/ExpoSE/Targets/xero-extension/node_modules/xml2js/lib/defaults_jalangi_.js', '/home/blake/ExpoSE/Targets/xero-extension/node_modules/xml2js/lib/defaults.js');
            J$.N(665, '__secret__S$', __secret__S$, 0);
            J$.N(673, '__secret__traits__', __secret__traits__, 0);
            var __secret__S$ = J$.X1(41, J$.W(33, '__secret__S$', J$.F(25, J$.R(9, 'require', require, 2), 0)(J$.T(17, 'S$', 21, false)), __secret__S$, 3));
            var __secret__traits__ = J$.X1(73, J$.W(65, '__secret__traits__', J$.G(57, J$.R(49, '__secret__S$', __secret__S$, 1), 'Traits', 0), __secret__traits__, 3));
            J$.X1(649, J$.M(641, J$.T(625, function () {
                jalangiLabel408:
                    while (true) {
                        try {
                            J$.Fe(609, arguments.callee, this, arguments);
                            arguments = J$.N(617, 'arguments', arguments, 4);
                            J$.X1(601, J$.P(593, J$.R(81, 'exports', exports, 2), 'defaults', J$.T(585, {
                                '0.1': J$.T(273, {
                                    explicitCharkey: J$.T(89, false, 23, false),
                                    trim: J$.T(97, true, 23, false),
                                    normalize: J$.T(105, true, 23, false),
                                    normalizeTags: J$.T(113, false, 23, false),
                                    attrkey: J$.T(121, '@', 21, false),
                                    charkey: J$.T(129, '#', 21, false),
                                    explicitArray: J$.T(137, false, 23, false),
                                    ignoreAttrs: J$.T(145, false, 23, false),
                                    mergeAttrs: J$.T(153, false, 23, false),
                                    explicitRoot: J$.T(161, false, 23, false),
                                    validator: J$.T(169, null, 25, false),
                                    xmlns: J$.T(177, false, 23, false),
                                    explicitChildren: J$.T(185, false, 23, false),
                                    childkey: J$.T(193, '@@', 21, false),
                                    charsAsChildren: J$.T(201, false, 23, false),
                                    includeWhiteChars: J$.T(209, false, 23, false),
                                    async: J$.T(217, false, 23, false),
                                    strict: J$.T(225, true, 23, false),
                                    attrNameProcessors: J$.T(233, null, 25, false),
                                    attrValueProcessors: J$.T(241, null, 25, false),
                                    tagNameProcessors: J$.T(249, null, 25, false),
                                    valueProcessors: J$.T(257, null, 25, false),
                                    emptyTag: J$.T(265, '', 21, false)
                                }, 11, false),
                                '0.2': J$.T(577, {
                                    explicitCharkey: J$.T(281, false, 23, false),
                                    trim: J$.T(289, false, 23, false),
                                    normalize: J$.T(297, false, 23, false),
                                    normalizeTags: J$.T(305, false, 23, false),
                                    attrkey: J$.T(313, '$', 21, false),
                                    charkey: J$.T(321, '_', 21, false),
                                    explicitArray: J$.T(329, true, 23, false),
                                    ignoreAttrs: J$.T(337, false, 23, false),
                                    mergeAttrs: J$.T(345, false, 23, false),
                                    explicitRoot: J$.T(353, true, 23, false),
                                    validator: J$.T(361, null, 25, false),
                                    xmlns: J$.T(369, false, 23, false),
                                    explicitChildren: J$.T(377, false, 23, false),
                                    preserveChildrenOrder: J$.T(385, false, 23, false),
                                    childkey: J$.T(393, '$$', 21, false),
                                    charsAsChildren: J$.T(401, false, 23, false),
                                    includeWhiteChars: J$.T(409, false, 23, false),
                                    async: J$.T(417, false, 23, false),
                                    strict: J$.T(425, true, 23, false),
                                    attrNameProcessors: J$.T(433, null, 25, false),
                                    attrValueProcessors: J$.T(441, null, 25, false),
                                    tagNameProcessors: J$.T(449, null, 25, false),
                                    valueProcessors: J$.T(457, null, 25, false),
                                    rootName: J$.T(465, 'root', 21, false),
                                    xmldec: J$.T(497, {
                                        'version': J$.T(473, '1.0', 21, false),
                                        'encoding': J$.T(481, 'UTF-8', 21, false),
                                        'standalone': J$.T(489, true, 23, false)
                                    }, 11, false),
                                    doctype: J$.T(505, null, 25, false),
                                    renderOpts: J$.T(537, {
                                        'pretty': J$.T(513, true, 23, false),
                                        'indent': J$.T(521, '  ', 21, false),
                                        'newline': J$.T(529, '\n', 21, false)
                                    }, 11, false),
                                    headless: J$.T(545, false, 23, false),
                                    chunkSize: J$.T(553, 10000, 22, false),
                                    emptyTag: J$.T(561, '', 21, false),
                                    cdata: J$.T(569, false, 23, false)
                                }, 11, false)
                            }, 11, false), 0));
                        } catch (J$e) {
                            J$.Ex(681, J$e);
                        } finally {
                            if (J$.Fr(689))
                                continue jalangiLabel408;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 609), 'call', 0)(J$.R(633, 'this', this, 0)));
        } catch (J$e) {
            J$.Ex(697, J$e);
        } finally {
            if (J$.Sr(705)) {
                J$.L();
                continue jalangiLabel409;
            } else {
                J$.L();
                break jalangiLabel409;
            }
        }
    }
// JALANGI DO NOT INSTRUMENT
