// bluebird/js/release/async_jalangi_.js
J$.iids = {"8":[39,9,39,25],"9":[1,20,1,27],"10":[48,42,48,60],"16":[44,12,44,55],"17":[1,28,1,32],"18":[48,30,48,75],"24":[48,42,48,60],"25":[1,20,1,33],"26":[48,30,48,82],"32":[47,9,47,15],"33":[1,20,1,33],"34":[55,9,55,31],"40":[55,9,55,31],"41":[1,20,1,33],"42":[61,9,61,26],"48":[61,9,61,42],"49":[2,26,2,38],"50":[61,9,61,42],"56":[92,13,92,36],"57":[2,26,2,45],"58":[86,5,86,22],"64":[103,13,103,36],"65":[2,26,2,45],"66":[122,12,122,30],"72":[112,13,112,36],"73":[2,26,2,45],"74":[124,13,124,22],"80":[86,5,86,22],"81":[3,1,3,13],"82":[124,13,124,37],"88":[124,13,124,37],"89":[3,1,3,14],"90":[140,9,140,26],"96":[122,12,122,30],"97":[6,15,6,20],"104":[140,9,140,26],"105":[6,11,6,22],"113":[6,11,6,22],"121":[6,5,6,23],"129":[8,22,8,23],"137":[8,22,8,23],"145":[8,5,8,24],"153":[7,3,9,2],"161":[10,16,10,23],"169":[10,24,10,36],"177":[10,16,10,37],"185":[10,16,10,37],"193":[10,16,10,37],"201":[11,13,11,20],"209":[11,21,11,30],"217":[11,13,11,31],"225":[11,13,11,31],"233":[11,13,11,31],"241":[12,12,12,19],"249":[12,20,12,28],"257":[12,12,12,29],"265":[12,12,12,29],"273":[12,12,12,29],"281":[14,5,14,9],"289":[14,29,14,34],"297":[14,5,14,34],"305":[14,5,14,35],"313":[15,5,15,9],"321":[15,24,15,29],"329":[15,5,15,29],"337":[15,5,15,30],"345":[16,5,16,9],"353":[16,27,16,32],"361":[16,33,16,35],"369":[16,23,16,36],"377":[16,5,16,36],"385":[16,5,16,37],"393":[17,5,17,9],"401":[17,29,17,34],"409":[17,35,17,37],"417":[17,25,17,38],"425":[17,5,17,38],"433":[17,5,17,39],"441":[18,5,18,9],"449":[18,31,18,36],"457":[18,5,18,36],"465":[18,5,18,37],"473":[19,5,19,9],"481":[19,31,19,35],"489":[19,5,19,35],"497":[19,5,19,36],"505":[20,16,20,20],"513":[20,16,20,20],"521":[20,16,20,20],"529":[21,5,21,9],"537":[22,9,22,13],"545":[22,9,22,28],"547":[22,9,22,26],"553":[22,9,22,29],"561":[21,24,23,6],"569":[21,24,23,6],"577":[21,24,23,6],"585":[21,5,23,6],"593":[21,5,23,7],"601":[24,5,24,9],"609":[24,22,24,30],"617":[24,5,24,30],"625":[24,5,24,31],"633":[13,1,25,2],"641":[13,1,25,2],"649":[13,1,25,2],"657":[26,1,26,6],"665":[26,1,26,16],"673":[27,16,27,20],"681":[27,16,27,30],"689":[27,16,27,30],"697":[27,16,27,30],"705":[28,5,28,9],"713":[28,22,28,24],"721":[28,5,28,24],"729":[28,5,28,25],"737":[29,5,29,9],"745":[29,29,29,33],"753":[29,5,29,33],"761":[29,5,29,34],"769":[30,12,30,16],"777":[30,12,30,16],"785":[30,5,30,17],"793":[26,32,31,2],"801":[26,32,31,2],"809":[26,32,31,2],"817":[26,32,31,2],"825":[26,32,31,2],"833":[26,1,31,2],"841":[26,1,31,3],"849":[32,1,32,6],"857":[32,1,32,16],"865":[33,12,33,16],"873":[33,12,33,33],"881":[33,12,33,33],"889":[33,5,33,34],"897":[32,38,34,2],"905":[32,38,34,2],"913":[32,38,34,2],"921":[32,1,34,2],"929":[32,1,34,3],"937":[35,1,35,6],"945":[35,1,35,16],"953":[36,5,36,9],"961":[36,31,36,35],"969":[36,5,36,35],"977":[36,5,36,36],"985":[35,36,37,2],"993":[35,36,37,2],"1001":[35,36,37,2],"1009":[35,1,37,2],"1017":[35,1,37,3],"1025":[38,1,38,6],"1033":[38,1,38,16],"1041":[39,9,39,13],"1049":[39,9,39,25],"1057":[40,9,40,13],"1065":[40,35,40,40],"1073":[40,9,40,40],"1081":[40,9,40,41],"1089":[38,48,42,2],"1097":[38,48,42,2],"1105":[38,48,42,2],"1113":[38,1,42,2],"1121":[38,1,42,3],"1129":[43,1,43,6],"1137":[43,1,43,16],"1145":[44,12,44,16],"1153":[44,12,44,28],"1161":[44,32,44,36],"1169":[44,32,44,55],"1177":[44,12,44,55],"1185":[44,5,44,56],"1193":[43,35,45,2],"1201":[43,35,45,2],"1209":[43,35,45,2],"1217":[43,1,45,2],"1225":[43,1,45,3],"1233":[46,1,46,6],"1241":[46,1,46,16],"1249":[47,9,47,15],"1257":[48,9,48,16],"1265":[48,9,48,23],"1273":[48,30,48,38],"1281":[48,42,48,43],"1289":[48,55,48,60],"1297":[48,63,48,64],"1305":[48,63,48,70],"1313":[48,73,48,74],"1321":[48,78,48,82],"1329":[48,9,48,83],"1331":[48,9,48,29],"1337":[48,9,48,84],"1345":[49,9,49,16],"1353":[49,22,49,23],"1361":[49,9,49,24],"1363":[49,9,49,21],"1369":[49,9,49,25],"1377":[51,9,51,13],"1385":[51,25,51,26],"1393":[51,9,51,27],"1395":[51,9,51,24],"1401":[51,9,51,28],"1409":[46,30,53,2],"1417":[46,30,53,2],"1425":[46,30,53,2],"1433":[46,30,53,2],"1441":[46,30,53,2],"1449":[46,1,53,2],"1457":[46,1,53,3],"1465":[54,1,54,6],"1473":[54,1,54,16],"1481":[55,9,55,18],"1489":[55,9,55,25],"1497":[55,30,55,31],"1505":[56,15,56,17],"1513":[56,15,56,17],"1521":[56,9,56,18],"1529":[58,19,58,22],"1537":[58,19,58,22],"1545":[58,13,58,23],"1553":[57,14,59,10],"1561":[57,14,59,10],"1569":[57,14,59,10],"1577":[57,14,59,10],"1585":[57,9,59,11],"1593":[61,16,61,26],"1601":[61,31,61,42],"1609":[62,9,62,19],"1617":[63,13,63,15],"1625":[63,16,63,19],"1633":[63,13,63,20],"1641":[63,13,63,21],"1649":[62,20,64,10],"1657":[62,20,64,10],"1665":[62,20,64,10],"1673":[64,12,64,13],"1681":[62,9,64,14],"1689":[62,9,64,15],"1697":[67,13,67,17],"1705":[68,17,68,19],"1713":[68,20,68,23],"1721":[68,17,68,24],"1729":[68,17,68,25],"1737":[67,28,69,14],"1745":[67,28,69,14],"1753":[67,28,69,14],"1761":[67,13,69,15],"1763":[67,13,67,27],"1769":[67,13,69,16],"1777":[71,23,71,28],"1785":[71,29,71,93],"1793":[71,19,71,94],"1801":[71,19,71,94],"1809":[71,13,71,95],"1817":[70,11,72,10],"1825":[54,30,73,2],"1833":[54,30,73,2],"1841":[54,30,73,2],"1849":[54,30,73,2],"1857":[54,30,73,2],"1865":[54,1,73,2],"1873":[54,1,73,3],"1881":[75,5,75,9],"1889":[75,5,75,20],"1897":[75,26,75,28],"1905":[75,30,75,38],"1913":[75,40,75,43],"1921":[75,5,75,44],"1923":[75,5,75,25],"1929":[75,5,75,45],"1937":[76,5,76,9],"1945":[76,5,76,22],"1947":[76,5,76,20],"1953":[76,5,76,23],"1961":[74,1,77,2],"1969":[74,1,77,2],"1977":[74,1,77,2],"1985":[74,1,77,2],"1993":[74,1,77,2],"2001":[79,5,79,9],"2009":[79,5,79,22],"2017":[79,28,79,30],"2025":[79,32,79,40],"2033":[79,42,79,45],"2041":[79,5,79,46],"2043":[79,5,79,27],"2049":[79,5,79,47],"2057":[80,5,80,9],"2065":[80,5,80,22],"2067":[80,5,80,20],"2073":[80,5,80,23],"2081":[78,1,81,2],"2089":[78,1,81,2],"2097":[78,1,81,2],"2105":[78,1,81,2],"2113":[78,1,81,2],"2121":[83,5,83,9],"2129":[83,5,83,22],"2137":[83,32,83,39],"2145":[83,5,83,40],"2147":[83,5,83,31],"2153":[83,5,83,41],"2161":[84,5,84,9],"2169":[84,5,84,22],"2171":[84,5,84,20],"2177":[84,5,84,23],"2185":[82,1,85,2],"2193":[82,1,85,2],"2201":[82,1,85,2],"2209":[86,6,86,10],"2217":[86,6,86,22],"2225":[87,5,87,10],"2233":[87,5,87,20],"2241":[87,35,87,51],"2249":[87,5,87,51],"2257":[87,5,87,52],"2265":[88,5,88,10],"2273":[88,5,88,20],"2281":[88,30,88,41],"2289":[88,5,88,41],"2297":[88,5,88,42],"2305":[89,5,89,10],"2313":[89,5,89,20],"2321":[89,38,89,57],"2329":[89,5,89,57],"2337":[89,5,89,58],"2345":[91,5,91,10],"2353":[91,5,91,20],"2361":[92,13,92,17],"2369":[92,13,92,36],"2377":[93,13,93,29],"2385":[93,35,93,39],"2393":[93,41,93,43],"2401":[93,45,93,53],"2409":[93,55,93,58],"2417":[93,13,93,59],"2419":[93,13,93,34],"2425":[93,13,93,60],"2433":[95,13,95,17],"2441":[96,17,96,27],"2449":[97,21,97,23],"2457":[97,29,97,37],"2465":[97,39,97,42],"2473":[97,21,97,43],"2475":[97,21,97,28],"2481":[97,21,97,44],"2489":[96,28,98,18],"2497":[96,28,98,18],"2505":[96,28,98,18],"2513":[98,20,98,23],"2521":[96,17,98,24],"2529":[96,17,98,25],"2537":[95,28,99,14],"2545":[95,28,99,14],"2553":[95,28,99,14],"2561":[95,13,99,15],"2563":[95,13,95,27],"2569":[95,13,99,16],"2577":[91,35,101,6],"2585":[91,35,101,6],"2593":[91,35,101,6],"2601":[91,35,101,6],"2609":[91,35,101,6],"2617":[91,35,101,6],"2625":[91,5,101,6],"2633":[91,5,101,7],"2641":[102,5,102,10],"2649":[102,5,102,20],"2657":[103,13,103,17],"2665":[103,13,103,36],"2673":[104,13,104,24],"2681":[104,30,104,34],"2689":[104,36,104,38],"2697":[104,40,104,48],"2705":[104,50,104,53],"2713":[104,13,104,54],"2715":[104,13,104,29],"2721":[104,13,104,55],"2729":[106,13,106,17],"2737":[107,17,107,19],"2745":[107,25,107,33],"2753":[107,35,107,38],"2761":[107,17,107,39],"2763":[107,17,107,24],"2769":[107,17,107,40],"2777":[106,28,108,14],"2785":[106,28,108,14],"2793":[106,28,108,14],"2801":[106,13,108,15],"2803":[106,13,106,27],"2809":[106,13,108,16],"2817":[102,30,110,6],"2825":[102,30,110,6],"2833":[102,30,110,6],"2841":[102,30,110,6],"2849":[102,30,110,6],"2857":[102,30,110,6],"2865":[102,5,110,6],"2873":[102,5,110,7],"2881":[111,5,111,10],"2889":[111,5,111,20],"2897":[112,13,112,17],"2905":[112,13,112,36],"2913":[113,13,113,32],"2921":[113,38,113,42],"2929":[113,44,113,51],"2937":[113,13,113,52],"2939":[113,13,113,37],"2945":[113,13,113,53],"2953":[115,13,115,17],"2961":[116,17,116,24],"2969":[116,17,116,42],"2971":[116,17,116,40],"2977":[116,17,116,43],"2985":[115,28,117,14],"2993":[115,28,117,14],"3001":[115,28,117,14],"3009":[115,13,117,15],"3011":[115,13,115,27],"3017":[115,13,117,16],"3025":[111,38,119,6],"3033":[111,38,119,6],"3041":[111,38,119,6],"3049":[111,38,119,6],"3057":[111,5,119,6],"3065":[111,5,119,7],"3073":[121,1,121,6],"3081":[121,1,121,16],"3089":[122,12,122,17],"3097":[122,12,122,26],"3099":[122,12,122,24],"3105":[122,29,122,30],"3113":[123,18,123,23],"3121":[123,18,123,31],"3123":[123,18,123,29],"3129":[123,18,123,31],"3137":[123,18,123,31],"3145":[124,20,124,22],"3153":[124,27,124,37],"3161":[125,13,125,15],"3169":[125,13,125,33],"3171":[125,13,125,31],"3177":[125,13,125,34],"3185":[128,24,128,29],"3193":[128,24,128,37],"3195":[128,24,128,35],"3201":[128,24,128,37],"3209":[128,24,128,37],"3217":[129,19,129,24],"3225":[129,19,129,32],"3227":[129,19,129,30],"3233":[129,19,129,32],"3241":[129,19,129,32],"3249":[130,9,130,11],"3257":[130,17,130,25],"3265":[130,27,130,30],"3273":[130,9,130,31],"3275":[130,9,130,16],"3281":[130,9,130,32],"3289":[121,31,132,2],"3297":[121,31,132,2],"3305":[121,31,132,2],"3313":[121,31,132,2],"3321":[121,31,132,2],"3329":[121,31,132,2],"3337":[121,31,132,2],"3345":[121,1,132,2],"3353":[121,1,132,3],"3361":[133,1,133,6],"3369":[133,1,133,16],"3377":[134,5,134,9],"3385":[134,22,134,26],"3393":[134,22,134,39],"3401":[134,5,134,40],"3403":[134,5,134,21],"3409":[134,5,134,41],"3417":[135,5,135,9],"3425":[135,5,135,18],"3427":[135,5,135,16],"3433":[135,5,135,19],"3441":[136,5,136,9],"3449":[136,31,136,35],"3457":[136,5,136,35],"3465":[136,5,136,36],"3473":[137,5,137,9],"3481":[137,22,137,26],"3489":[137,22,137,37],"3497":[137,5,137,38],"3499":[137,5,137,21],"3505":[137,5,137,39],"3513":[133,32,138,2],"3521":[133,32,138,2],"3529":[133,32,138,2],"3537":[133,1,138,2],"3545":[133,1,138,3],"3553":[139,1,139,6],"3561":[139,1,139,16],"3569":[140,10,140,14],"3577":[140,10,140,26],"3585":[141,9,141,13],"3593":[141,28,141,32],"3601":[141,9,141,32],"3609":[141,9,141,33],"3617":[142,9,142,13],"3625":[142,24,142,28],"3633":[142,24,142,40],"3641":[142,9,142,41],"3643":[142,9,142,23],"3649":[142,9,142,42],"3657":[139,30,144,2],"3665":[139,30,144,2],"3673":[139,30,144,2],"3681":[139,1,144,2],"3689":[139,1,144,3],"3697":[145,1,145,6],"3705":[145,1,145,16],"3713":[146,5,146,9],"3721":[146,24,146,29],"3729":[146,5,146,29],"3737":[146,5,146,30],"3745":[145,26,147,2],"3753":[145,26,147,2],"3761":[145,26,147,2],"3769":[145,1,147,2],"3777":[145,1,147,3],"3785":[148,1,148,7],"3793":[148,18,148,23],"3801":[148,1,148,23],"3809":[148,1,148,24],"3817":[149,1,149,7],"3825":[149,1,149,15],"3833":[149,33,149,47],"3841":[149,1,149,47],"3849":[149,1,149,48],"3857":[1,1,149,48],"3865":[1,1,149,48],"3873":[1,1,149,48],"3881":[1,1,149,48],"3889":[1,1,149,48],"3897":[1,1,149,48],"3905":[1,1,149,48],"3913":[13,1,25,2],"3921":[1,1,149,48],"3929":[74,1,77,2],"3937":[1,1,149,48],"3945":[78,1,81,2],"3953":[1,1,149,48],"3961":[82,1,85,2],"3969":[1,1,149,48],"3977":[21,24,23,6],"3985":[21,24,23,6],"3993":[13,1,25,2],"4001":[13,1,25,2],"4009":[26,32,31,2],"4017":[26,32,31,2],"4025":[32,38,34,2],"4033":[32,38,34,2],"4041":[35,36,37,2],"4049":[35,36,37,2],"4057":[39,5,41,6],"4065":[38,48,42,2],"4073":[38,48,42,2],"4081":[43,35,45,2],"4089":[43,35,45,2],"4097":[48,42,48,74],"4105":[47,5,52,6],"4113":[46,30,53,2],"4121":[46,30,53,2],"4129":[57,14,59,10],"4137":[57,14,59,10],"4145":[55,5,60,6],"4153":[62,20,64,10],"4161":[62,20,64,10],"4169":[67,28,69,14],"4177":[67,28,69,14],"4185":[61,5,72,10],"4193":[54,30,73,2],"4201":[54,30,73,2],"4209":[74,1,77,2],"4217":[74,1,77,2],"4225":[78,1,81,2],"4233":[78,1,81,2],"4241":[82,1,85,2],"4249":[82,1,85,2],"4257":[96,28,98,18],"4265":[96,28,98,18],"4273":[95,28,99,14],"4281":[95,28,99,14],"4289":[92,9,100,10],"4297":[91,35,101,6],"4305":[91,35,101,6],"4313":[106,28,108,14],"4321":[106,28,108,14],"4329":[103,9,109,10],"4337":[102,30,110,6],"4345":[102,30,110,6],"4353":[115,28,117,14],"4361":[115,28,117,14],"4369":[112,9,118,10],"4377":[111,38,119,6],"4385":[111,38,119,6],"4393":[86,1,120,2],"4401":[124,9,127,10],"4409":[122,5,131,6],"4417":[121,31,132,2],"4425":[121,31,132,2],"4433":[133,32,138,2],"4441":[133,32,138,2],"4449":[140,5,143,6],"4457":[139,30,144,2],"4465":[139,30,144,2],"4473":[145,26,147,2],"4481":[145,26,147,2],"4489":[1,1,149,48],"4497":[1,1,149,48],"nBranches":26,"originalCodeFileName":"/home/blake/ExpoSE/Targets/hapigerjs/node_modules/bluebird/js/release/async.js","instrumentedCodeFileName":"/home/blake/ExpoSE/Targets/hapigerjs/node_modules/bluebird/js/release/async_jalangi_.js","code":"\"use strict\";\nvar firstLineError;\ntry {throw new Error(); } catch (e) {firstLineError = e;}\nvar schedule = require(\"./schedule\");\nvar Queue = require(\"./queue\");\nvar util = require(\"./util\");\n\nfunction Async() {\n    this._customScheduler = false;\n    this._isTickUsed = false;\n    this._lateQueue = new Queue(16);\n    this._normalQueue = new Queue(16);\n    this._haveDrainedQueues = false;\n    this._trampolineEnabled = true;\n    var self = this;\n    this.drainQueues = function () {\n        self._drainQueues();\n    };\n    this._schedule = schedule;\n}\n\nAsync.prototype.setScheduler = function(fn) {\n    var prev = this._schedule;\n    this._schedule = fn;\n    this._customScheduler = true;\n    return prev;\n};\n\nAsync.prototype.hasCustomScheduler = function() {\n    return this._customScheduler;\n};\n\nAsync.prototype.enableTrampoline = function() {\n    this._trampolineEnabled = true;\n};\n\nAsync.prototype.disableTrampolineIfNecessary = function() {\n    if (util.hasDevTools) {\n        this._trampolineEnabled = false;\n    }\n};\n\nAsync.prototype.haveItemsQueued = function () {\n    return this._isTickUsed || this._haveDrainedQueues;\n};\n\n\nAsync.prototype.fatalError = function(e, isNode) {\n    if (isNode) {\n        process.stderr.write(\"Fatal \" + (e instanceof Error ? e.stack : e) +\n            \"\\n\");\n        process.exit(2);\n    } else {\n        this.throwLater(e);\n    }\n};\n\nAsync.prototype.throwLater = function(fn, arg) {\n    if (arguments.length === 1) {\n        arg = fn;\n        fn = function () { throw arg; };\n    }\n    if (typeof setTimeout !== \"undefined\") {\n        setTimeout(function() {\n            fn(arg);\n        }, 0);\n    } else try {\n        this._schedule(function() {\n            fn(arg);\n        });\n    } catch (e) {\n        throw new Error(\"No async scheduler available\\u000a\\u000a    See http://goo.gl/MqrFmX\\u000a\");\n    }\n};\n\nfunction AsyncInvokeLater(fn, receiver, arg) {\n    this._lateQueue.push(fn, receiver, arg);\n    this._queueTick();\n}\n\nfunction AsyncInvoke(fn, receiver, arg) {\n    this._normalQueue.push(fn, receiver, arg);\n    this._queueTick();\n}\n\nfunction AsyncSettlePromises(promise) {\n    this._normalQueue._pushOne(promise);\n    this._queueTick();\n}\n\nif (!util.hasDevTools) {\n    Async.prototype.invokeLater = AsyncInvokeLater;\n    Async.prototype.invoke = AsyncInvoke;\n    Async.prototype.settlePromises = AsyncSettlePromises;\n} else {\n    Async.prototype.invokeLater = function (fn, receiver, arg) {\n        if (this._trampolineEnabled) {\n            AsyncInvokeLater.call(this, fn, receiver, arg);\n        } else {\n            this._schedule(function() {\n                setTimeout(function() {\n                    fn.call(receiver, arg);\n                }, 100);\n            });\n        }\n    };\n\n    Async.prototype.invoke = function (fn, receiver, arg) {\n        if (this._trampolineEnabled) {\n            AsyncInvoke.call(this, fn, receiver, arg);\n        } else {\n            this._schedule(function() {\n                fn.call(receiver, arg);\n            });\n        }\n    };\n\n    Async.prototype.settlePromises = function(promise) {\n        if (this._trampolineEnabled) {\n            AsyncSettlePromises.call(this, promise);\n        } else {\n            this._schedule(function() {\n                promise._settlePromises();\n            });\n        }\n    };\n}\n\nAsync.prototype._drainQueue = function(queue) {\n    while (queue.length() > 0) {\n        var fn = queue.shift();\n        if (typeof fn !== \"function\") {\n            fn._settlePromises();\n            continue;\n        }\n        var receiver = queue.shift();\n        var arg = queue.shift();\n        fn.call(receiver, arg);\n    }\n};\n\nAsync.prototype._drainQueues = function () {\n    this._drainQueue(this._normalQueue);\n    this._reset();\n    this._haveDrainedQueues = true;\n    this._drainQueue(this._lateQueue);\n};\n\nAsync.prototype._queueTick = function () {\n    if (!this._isTickUsed) {\n        this._isTickUsed = true;\n        this._schedule(this.drainQueues);\n    }\n};\n\nAsync.prototype._reset = function () {\n    this._isTickUsed = false;\n};\n\nmodule.exports = Async;\nmodule.exports.firstLineError = firstLineError;\n"};
jalangiLabel207:
    while (true) {
        try {
            J$.Se(3857, '/home/blake/ExpoSE/Targets/hapigerjs/node_modules/bluebird/js/release/async_jalangi_.js', '/home/blake/ExpoSE/Targets/hapigerjs/node_modules/bluebird/js/release/async.js');
            function Async() {
                jalangiLabel182:
                    while (true) {
                        try {
                            J$.Fe(633, arguments.callee, this, arguments);
                            arguments = J$.N(641, 'arguments', arguments, 4);
                            J$.N(649, 'self', self, 0);
                            J$.X1(305, J$.P(297, J$.R(281, 'this', this, 0), '_customScheduler', J$.T(289, false, 23, false), 0));
                            J$.X1(337, J$.P(329, J$.R(313, 'this', this, 0), '_isTickUsed', J$.T(321, false, 23, false), 0));
                            J$.X1(385, J$.P(377, J$.R(345, 'this', this, 0), '_lateQueue', J$.F(369, J$.R(353, 'Queue', Queue, 1), 1)(J$.T(361, 16, 22, false)), 0));
                            J$.X1(433, J$.P(425, J$.R(393, 'this', this, 0), '_normalQueue', J$.F(417, J$.R(401, 'Queue', Queue, 1), 1)(J$.T(409, 16, 22, false)), 0));
                            J$.X1(465, J$.P(457, J$.R(441, 'this', this, 0), '_haveDrainedQueues', J$.T(449, false, 23, false), 0));
                            J$.X1(497, J$.P(489, J$.R(473, 'this', this, 0), '_trampolineEnabled', J$.T(481, true, 23, false), 0));
                            var self = J$.X1(521, J$.W(513, 'self', J$.R(505, 'this', this, 0), self, 1));
                            J$.X1(593, J$.P(585, J$.R(529, 'this', this, 0), 'drainQueues', J$.T(577, function () {
                                jalangiLabel181:
                                    while (true) {
                                        try {
                                            J$.Fe(561, arguments.callee, this, arguments);
                                            arguments = J$.N(569, 'arguments', arguments, 4);
                                            J$.X1(553, J$.M(545, J$.R(537, 'self', self, 0), '_drainQueues', 0)());
                                        } catch (J$e) {
                                            J$.Ex(3977, J$e);
                                        } finally {
                                            if (J$.Fr(3985))
                                                continue jalangiLabel181;
                                            else
                                                return J$.Ra();
                                        }
                                    }
                            }, 12, false, 561), 0));
                            J$.X1(625, J$.P(617, J$.R(601, 'this', this, 0), '_schedule', J$.R(609, 'schedule', schedule, 1), 0));
                        } catch (J$e) {
                            J$.Ex(3993, J$e);
                        } finally {
                            if (J$.Fr(4001))
                                continue jalangiLabel182;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function AsyncInvokeLater(fn, receiver, arg) {
                jalangiLabel193:
                    while (true) {
                        try {
                            J$.Fe(1961, arguments.callee, this, arguments);
                            arguments = J$.N(1969, 'arguments', arguments, 4);
                            fn = J$.N(1977, 'fn', fn, 4);
                            receiver = J$.N(1985, 'receiver', receiver, 4);
                            arg = J$.N(1993, 'arg', arg, 4);
                            J$.X1(1929, J$.M(1921, J$.G(1889, J$.R(1881, 'this', this, 0), '_lateQueue', 0), 'push', 0)(J$.R(1897, 'fn', fn, 0), J$.R(1905, 'receiver', receiver, 0), J$.R(1913, 'arg', arg, 0)));
                            J$.X1(1953, J$.M(1945, J$.R(1937, 'this', this, 0), '_queueTick', 0)());
                        } catch (J$e) {
                            J$.Ex(4209, J$e);
                        } finally {
                            if (J$.Fr(4217))
                                continue jalangiLabel193;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function AsyncInvoke(fn, receiver, arg) {
                jalangiLabel194:
                    while (true) {
                        try {
                            J$.Fe(2081, arguments.callee, this, arguments);
                            arguments = J$.N(2089, 'arguments', arguments, 4);
                            fn = J$.N(2097, 'fn', fn, 4);
                            receiver = J$.N(2105, 'receiver', receiver, 4);
                            arg = J$.N(2113, 'arg', arg, 4);
                            J$.X1(2049, J$.M(2041, J$.G(2009, J$.R(2001, 'this', this, 0), '_normalQueue', 0), 'push', 0)(J$.R(2017, 'fn', fn, 0), J$.R(2025, 'receiver', receiver, 0), J$.R(2033, 'arg', arg, 0)));
                            J$.X1(2073, J$.M(2065, J$.R(2057, 'this', this, 0), '_queueTick', 0)());
                        } catch (J$e) {
                            J$.Ex(4225, J$e);
                        } finally {
                            if (J$.Fr(4233))
                                continue jalangiLabel194;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function AsyncSettlePromises(promise) {
                jalangiLabel195:
                    while (true) {
                        try {
                            J$.Fe(2185, arguments.callee, this, arguments);
                            arguments = J$.N(2193, 'arguments', arguments, 4);
                            promise = J$.N(2201, 'promise', promise, 4);
                            J$.X1(2153, J$.M(2145, J$.G(2129, J$.R(2121, 'this', this, 0), '_normalQueue', 0), '_pushOne', 0)(J$.R(2137, 'promise', promise, 0)));
                            J$.X1(2177, J$.M(2169, J$.R(2161, 'this', this, 0), '_queueTick', 0)());
                        } catch (J$e) {
                            J$.Ex(4241, J$e);
                        } finally {
                            if (J$.Fr(4249))
                                continue jalangiLabel195;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(3865, '__secret__S$', __secret__S$, 0);
            J$.N(3873, '__secret__traits__', __secret__traits__, 0);
            J$.N(3881, 'firstLineError', firstLineError, 0);
            J$.N(3889, 'schedule', schedule, 0);
            J$.N(3897, 'Queue', Queue, 0);
            J$.N(3905, 'util', util, 0);
            Async = J$.N(3921, 'Async', J$.T(3913, Async, 12, false, 633), 0);
            AsyncInvokeLater = J$.N(3937, 'AsyncInvokeLater', J$.T(3929, AsyncInvokeLater, 12, false, 1961), 0);
            AsyncInvoke = J$.N(3953, 'AsyncInvoke', J$.T(3945, AsyncInvoke, 12, false, 2081), 0);
            AsyncSettlePromises = J$.N(3969, 'AsyncSettlePromises', J$.T(3961, AsyncSettlePromises, 12, false, 2185), 0);
            var __secret__S$ = J$.X1(41, J$.W(33, '__secret__S$', J$.F(25, J$.R(9, 'require', require, 2), 0)(J$.T(17, 'S$', 21, false)), __secret__S$, 3));
            var __secret__traits__ = J$.X1(73, J$.W(65, '__secret__traits__', J$.G(57, J$.R(49, '__secret__S$', __secret__S$, 1), 'Traits', 0), __secret__traits__, 3));
            J$.X1(89, J$.T(81, 'use strict', 21, false));
            var firstLineError;
            try {
                throw J$.X1(121, J$.Th(113, J$.F(105, J$.R(97, 'Error', Error, 2), 1)()));
            } catch (e) {
                e = J$.N(153, 'e', e, 1);
                J$.X1(145, firstLineError = J$.W(137, 'firstLineError', J$.R(129, 'e', e, 0), firstLineError, 2));
            }
            var schedule = J$.X1(193, J$.W(185, 'schedule', J$.F(177, J$.R(161, 'require', require, 2), 0)(J$.T(169, './schedule', 21, false)), schedule, 3));
            var Queue = J$.X1(233, J$.W(225, 'Queue', J$.F(217, J$.R(201, 'require', require, 2), 0)(J$.T(209, './queue', 21, false)), Queue, 3));
            var util = J$.X1(273, J$.W(265, 'util', J$.F(257, J$.R(241, 'require', require, 2), 0)(J$.T(249, './util', 21, false)), util, 3));
            J$.X1(841, J$.P(833, J$.G(665, J$.R(657, 'Async', Async, 1), 'prototype', 0), 'setScheduler', J$.T(825, function (fn) {
                jalangiLabel183:
                    while (true) {
                        try {
                            J$.Fe(793, arguments.callee, this, arguments);
                            arguments = J$.N(801, 'arguments', arguments, 4);
                            fn = J$.N(809, 'fn', fn, 4);
                            J$.N(817, 'prev', prev, 0);
                            var prev = J$.X1(697, J$.W(689, 'prev', J$.G(681, J$.R(673, 'this', this, 0), '_schedule', 0), prev, 1));
                            J$.X1(729, J$.P(721, J$.R(705, 'this', this, 0), '_schedule', J$.R(713, 'fn', fn, 0), 0));
                            J$.X1(761, J$.P(753, J$.R(737, 'this', this, 0), '_customScheduler', J$.T(745, true, 23, false), 0));
                            return J$.X1(785, J$.Rt(777, J$.R(769, 'prev', prev, 0)));
                        } catch (J$e) {
                            J$.Ex(4009, J$e);
                        } finally {
                            if (J$.Fr(4017))
                                continue jalangiLabel183;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 793), 0));
            J$.X1(929, J$.P(921, J$.G(857, J$.R(849, 'Async', Async, 1), 'prototype', 0), 'hasCustomScheduler', J$.T(913, function () {
                jalangiLabel184:
                    while (true) {
                        try {
                            J$.Fe(897, arguments.callee, this, arguments);
                            arguments = J$.N(905, 'arguments', arguments, 4);
                            return J$.X1(889, J$.Rt(881, J$.G(873, J$.R(865, 'this', this, 0), '_customScheduler', 0)));
                        } catch (J$e) {
                            J$.Ex(4025, J$e);
                        } finally {
                            if (J$.Fr(4033))
                                continue jalangiLabel184;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 897), 0));
            J$.X1(1017, J$.P(1009, J$.G(945, J$.R(937, 'Async', Async, 1), 'prototype', 0), 'enableTrampoline', J$.T(1001, function () {
                jalangiLabel185:
                    while (true) {
                        try {
                            J$.Fe(985, arguments.callee, this, arguments);
                            arguments = J$.N(993, 'arguments', arguments, 4);
                            J$.X1(977, J$.P(969, J$.R(953, 'this', this, 0), '_trampolineEnabled', J$.T(961, true, 23, false), 0));
                        } catch (J$e) {
                            J$.Ex(4041, J$e);
                        } finally {
                            if (J$.Fr(4049))
                                continue jalangiLabel185;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 985), 0));
            J$.X1(1121, J$.P(1113, J$.G(1033, J$.R(1025, 'Async', Async, 1), 'prototype', 0), 'disableTrampolineIfNecessary', J$.T(1105, function () {
                jalangiLabel186:
                    while (true) {
                        try {
                            J$.Fe(1089, arguments.callee, this, arguments);
                            arguments = J$.N(1097, 'arguments', arguments, 4);
                            if (J$.X1(4057, J$.C(8, J$.G(1049, J$.R(1041, 'util', util, 1), 'hasDevTools', 0)))) {
                                J$.X1(1081, J$.P(1073, J$.R(1057, 'this', this, 0), '_trampolineEnabled', J$.T(1065, false, 23, false), 0));
                            }
                        } catch (J$e) {
                            J$.Ex(4065, J$e);
                        } finally {
                            if (J$.Fr(4073))
                                continue jalangiLabel186;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 1089), 0));
            J$.X1(1225, J$.P(1217, J$.G(1137, J$.R(1129, 'Async', Async, 1), 'prototype', 0), 'haveItemsQueued', J$.T(1209, function () {
                jalangiLabel187:
                    while (true) {
                        try {
                            J$.Fe(1193, arguments.callee, this, arguments);
                            arguments = J$.N(1201, 'arguments', arguments, 4);
                            return J$.X1(1185, J$.Rt(1177, J$.C(16, J$.G(1153, J$.R(1145, 'this', this, 0), '_isTickUsed', 0)) ? J$._() : J$.G(1169, J$.R(1161, 'this', this, 0), '_haveDrainedQueues', 0)));
                        } catch (J$e) {
                            J$.Ex(4081, J$e);
                        } finally {
                            if (J$.Fr(4089))
                                continue jalangiLabel187;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 1193), 0));
            J$.X1(1457, J$.P(1449, J$.G(1241, J$.R(1233, 'Async', Async, 1), 'prototype', 0), 'fatalError', J$.T(1441, function (e, isNode) {
                jalangiLabel188:
                    while (true) {
                        try {
                            J$.Fe(1409, arguments.callee, this, arguments);
                            arguments = J$.N(1417, 'arguments', arguments, 4);
                            e = J$.N(1425, 'e', e, 4);
                            isNode = J$.N(1433, 'isNode', isNode, 4);
                            if (J$.X1(4105, J$.C(32, J$.R(1249, 'isNode', isNode, 0)))) {
                                J$.X1(1337, J$.M(1329, J$.G(1265, J$.R(1257, 'process', process, 2), 'stderr', 0), 'write', 0)(J$.B(26, '+', J$.B(18, '+', J$.T(1273, 'Fatal ', 21, false), J$.X1(4097, J$.C(24, J$.B(10, 'instanceof', J$.R(1281, 'e', e, 0), J$.R(1289, 'Error', Error, 2), 0))) ? J$.G(1305, J$.R(1297, 'e', e, 0), 'stack', 0) : J$.R(1313, 'e', e, 0), 0), J$.T(1321, '\n', 21, false), 0)));
                                J$.X1(1369, J$.M(1361, J$.R(1345, 'process', process, 2), 'exit', 0)(J$.T(1353, 2, 22, false)));
                            } else {
                                J$.X1(1401, J$.M(1393, J$.R(1377, 'this', this, 0), 'throwLater', 0)(J$.R(1385, 'e', e, 0)));
                            }
                        } catch (J$e) {
                            J$.Ex(4113, J$e);
                        } finally {
                            if (J$.Fr(4121))
                                continue jalangiLabel188;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 1409), 0));
            J$.X1(1873, J$.P(1865, J$.G(1473, J$.R(1465, 'Async', Async, 1), 'prototype', 0), 'throwLater', J$.T(1857, function (fn, arg) {
                jalangiLabel192:
                    while (true) {
                        try {
                            J$.Fe(1825, arguments.callee, this, arguments);
                            arguments = J$.N(1833, 'arguments', arguments, 4);
                            fn = J$.N(1841, 'fn', fn, 4);
                            arg = J$.N(1849, 'arg', arg, 4);
                            if (J$.X1(4145, J$.C(40, J$.B(34, '===', J$.G(1489, J$.R(1481, 'arguments', arguments, 2), 'length', 0), J$.T(1497, 1, 22, false), 0)))) {
                                J$.X1(1521, arg = J$.W(1513, 'arg', J$.R(1505, 'fn', fn, 0), arg, 0));
                                J$.X1(1585, fn = J$.W(1577, 'fn', J$.T(1569, function () {
                                    jalangiLabel189:
                                        while (true) {
                                            try {
                                                J$.Fe(1553, arguments.callee, this, arguments);
                                                arguments = J$.N(1561, 'arguments', arguments, 4);
                                                throw J$.X1(1545, J$.Th(1537, J$.R(1529, 'arg', arg, 0)));
                                            } catch (J$e) {
                                                J$.Ex(4129, J$e);
                                            } finally {
                                                if (J$.Fr(4137))
                                                    continue jalangiLabel189;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false, 1553), fn, 0));
                            }
                            if (J$.X1(4185, J$.C(48, J$.B(50, '!==', J$.U(42, 'typeof', J$.I(typeof setTimeout === 'undefined' ? J$.R(1593, 'setTimeout', undefined, 2) : J$.R(1593, 'setTimeout', setTimeout, 2))), J$.T(1601, 'undefined', 21, false), 0)))) {
                                J$.X1(1689, J$.F(1681, J$.R(1609, 'setTimeout', setTimeout, 2), 0)(J$.T(1665, function () {
                                    jalangiLabel190:
                                        while (true) {
                                            try {
                                                J$.Fe(1649, arguments.callee, this, arguments);
                                                arguments = J$.N(1657, 'arguments', arguments, 4);
                                                J$.X1(1641, J$.F(1633, J$.R(1617, 'fn', fn, 0), 0)(J$.R(1625, 'arg', arg, 0)));
                                            } catch (J$e) {
                                                J$.Ex(4153, J$e);
                                            } finally {
                                                if (J$.Fr(4161))
                                                    continue jalangiLabel190;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false, 1649), J$.T(1673, 0, 22, false)));
                            } else
                                try {
                                J$.X1(1769, J$.M(1761, J$.R(1697, 'this', this, 0), '_schedule', 0)(J$.T(1753, function () {
                                    jalangiLabel191:
                                        while (true) {
                                            try {
                                                J$.Fe(1737, arguments.callee, this, arguments);
                                                arguments = J$.N(1745, 'arguments', arguments, 4);
                                                J$.X1(1729, J$.F(1721, J$.R(1705, 'fn', fn, 0), 0)(J$.R(1713, 'arg', arg, 0)));
                                            } catch (J$e) {
                                                J$.Ex(4169, J$e);
                                            } finally {
                                                if (J$.Fr(4177))
                                                    continue jalangiLabel191;
                                                else
                                                    return J$.Ra();
                                            }
                                        }
                                }, 12, false, 1737)));
                            } catch (e) {
                                e = J$.N(1817, 'e', e, 1);
                                throw J$.X1(1809, J$.Th(1801, J$.F(1793, J$.R(1777, 'Error', Error, 2), 1)(J$.T(1785, 'No async scheduler available\n\n    See http://goo.gl/MqrFmX\n', 21, false))));
                            }
                        } catch (J$e) {
                            J$.Ex(4193, J$e);
                        } finally {
                            if (J$.Fr(4201))
                                continue jalangiLabel192;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 1825), 0));
            if (J$.X1(4393, J$.C(80, J$.U(58, '!', J$.G(2217, J$.R(2209, 'util', util, 1), 'hasDevTools', 0))))) {
                J$.X1(2257, J$.P(2249, J$.G(2233, J$.R(2225, 'Async', Async, 1), 'prototype', 0), 'invokeLater', J$.R(2241, 'AsyncInvokeLater', AsyncInvokeLater, 1), 0));
                J$.X1(2297, J$.P(2289, J$.G(2273, J$.R(2265, 'Async', Async, 1), 'prototype', 0), 'invoke', J$.R(2281, 'AsyncInvoke', AsyncInvoke, 1), 0));
                J$.X1(2337, J$.P(2329, J$.G(2313, J$.R(2305, 'Async', Async, 1), 'prototype', 0), 'settlePromises', J$.R(2321, 'AsyncSettlePromises', AsyncSettlePromises, 1), 0));
            } else {
                J$.X1(2633, J$.P(2625, J$.G(2353, J$.R(2345, 'Async', Async, 1), 'prototype', 0), 'invokeLater', J$.T(2617, function (fn, receiver, arg) {
                    jalangiLabel198:
                        while (true) {
                            try {
                                J$.Fe(2577, arguments.callee, this, arguments);
                                arguments = J$.N(2585, 'arguments', arguments, 4);
                                fn = J$.N(2593, 'fn', fn, 4);
                                receiver = J$.N(2601, 'receiver', receiver, 4);
                                arg = J$.N(2609, 'arg', arg, 4);
                                if (J$.X1(4289, J$.C(56, J$.G(2369, J$.R(2361, 'this', this, 0), '_trampolineEnabled', 0)))) {
                                    J$.X1(2425, J$.M(2417, J$.R(2377, 'AsyncInvokeLater', AsyncInvokeLater, 1), 'call', 0)(J$.R(2385, 'this', this, 0), J$.R(2393, 'fn', fn, 0), J$.R(2401, 'receiver', receiver, 0), J$.R(2409, 'arg', arg, 0)));
                                } else {
                                    J$.X1(2569, J$.M(2561, J$.R(2433, 'this', this, 0), '_schedule', 0)(J$.T(2553, function () {
                                        jalangiLabel197:
                                            while (true) {
                                                try {
                                                    J$.Fe(2537, arguments.callee, this, arguments);
                                                    arguments = J$.N(2545, 'arguments', arguments, 4);
                                                    J$.X1(2529, J$.F(2521, J$.R(2441, 'setTimeout', setTimeout, 2), 0)(J$.T(2505, function () {
                                                        jalangiLabel196:
                                                            while (true) {
                                                                try {
                                                                    J$.Fe(2489, arguments.callee, this, arguments);
                                                                    arguments = J$.N(2497, 'arguments', arguments, 4);
                                                                    J$.X1(2481, J$.M(2473, J$.R(2449, 'fn', fn, 0), 'call', 0)(J$.R(2457, 'receiver', receiver, 0), J$.R(2465, 'arg', arg, 0)));
                                                                } catch (J$e) {
                                                                    J$.Ex(4257, J$e);
                                                                } finally {
                                                                    if (J$.Fr(4265))
                                                                        continue jalangiLabel196;
                                                                    else
                                                                        return J$.Ra();
                                                                }
                                                            }
                                                    }, 12, false, 2489), J$.T(2513, 100, 22, false)));
                                                } catch (J$e) {
                                                    J$.Ex(4273, J$e);
                                                } finally {
                                                    if (J$.Fr(4281))
                                                        continue jalangiLabel197;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false, 2537)));
                                }
                            } catch (J$e) {
                                J$.Ex(4297, J$e);
                            } finally {
                                if (J$.Fr(4305))
                                    continue jalangiLabel198;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false, 2577), 0));
                J$.X1(2873, J$.P(2865, J$.G(2649, J$.R(2641, 'Async', Async, 1), 'prototype', 0), 'invoke', J$.T(2857, function (fn, receiver, arg) {
                    jalangiLabel200:
                        while (true) {
                            try {
                                J$.Fe(2817, arguments.callee, this, arguments);
                                arguments = J$.N(2825, 'arguments', arguments, 4);
                                fn = J$.N(2833, 'fn', fn, 4);
                                receiver = J$.N(2841, 'receiver', receiver, 4);
                                arg = J$.N(2849, 'arg', arg, 4);
                                if (J$.X1(4329, J$.C(64, J$.G(2665, J$.R(2657, 'this', this, 0), '_trampolineEnabled', 0)))) {
                                    J$.X1(2721, J$.M(2713, J$.R(2673, 'AsyncInvoke', AsyncInvoke, 1), 'call', 0)(J$.R(2681, 'this', this, 0), J$.R(2689, 'fn', fn, 0), J$.R(2697, 'receiver', receiver, 0), J$.R(2705, 'arg', arg, 0)));
                                } else {
                                    J$.X1(2809, J$.M(2801, J$.R(2729, 'this', this, 0), '_schedule', 0)(J$.T(2793, function () {
                                        jalangiLabel199:
                                            while (true) {
                                                try {
                                                    J$.Fe(2777, arguments.callee, this, arguments);
                                                    arguments = J$.N(2785, 'arguments', arguments, 4);
                                                    J$.X1(2769, J$.M(2761, J$.R(2737, 'fn', fn, 0), 'call', 0)(J$.R(2745, 'receiver', receiver, 0), J$.R(2753, 'arg', arg, 0)));
                                                } catch (J$e) {
                                                    J$.Ex(4313, J$e);
                                                } finally {
                                                    if (J$.Fr(4321))
                                                        continue jalangiLabel199;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false, 2777)));
                                }
                            } catch (J$e) {
                                J$.Ex(4337, J$e);
                            } finally {
                                if (J$.Fr(4345))
                                    continue jalangiLabel200;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false, 2817), 0));
                J$.X1(3065, J$.P(3057, J$.G(2889, J$.R(2881, 'Async', Async, 1), 'prototype', 0), 'settlePromises', J$.T(3049, function (promise) {
                    jalangiLabel202:
                        while (true) {
                            try {
                                J$.Fe(3025, arguments.callee, this, arguments);
                                arguments = J$.N(3033, 'arguments', arguments, 4);
                                promise = J$.N(3041, 'promise', promise, 4);
                                if (J$.X1(4369, J$.C(72, J$.G(2905, J$.R(2897, 'this', this, 0), '_trampolineEnabled', 0)))) {
                                    J$.X1(2945, J$.M(2937, J$.R(2913, 'AsyncSettlePromises', AsyncSettlePromises, 1), 'call', 0)(J$.R(2921, 'this', this, 0), J$.R(2929, 'promise', promise, 0)));
                                } else {
                                    J$.X1(3017, J$.M(3009, J$.R(2953, 'this', this, 0), '_schedule', 0)(J$.T(3001, function () {
                                        jalangiLabel201:
                                            while (true) {
                                                try {
                                                    J$.Fe(2985, arguments.callee, this, arguments);
                                                    arguments = J$.N(2993, 'arguments', arguments, 4);
                                                    J$.X1(2977, J$.M(2969, J$.R(2961, 'promise', promise, 0), '_settlePromises', 0)());
                                                } catch (J$e) {
                                                    J$.Ex(4353, J$e);
                                                } finally {
                                                    if (J$.Fr(4361))
                                                        continue jalangiLabel201;
                                                    else
                                                        return J$.Ra();
                                                }
                                            }
                                    }, 12, false, 2985)));
                                }
                            } catch (J$e) {
                                J$.Ex(4377, J$e);
                            } finally {
                                if (J$.Fr(4385))
                                    continue jalangiLabel202;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false, 3025), 0));
            }
            J$.X1(3353, J$.P(3345, J$.G(3081, J$.R(3073, 'Async', Async, 1), 'prototype', 0), '_drainQueue', J$.T(3337, function (queue) {
                jalangiLabel203:
                    while (true) {
                        try {
                            J$.Fe(3289, arguments.callee, this, arguments);
                            arguments = J$.N(3297, 'arguments', arguments, 4);
                            queue = J$.N(3305, 'queue', queue, 4);
                            J$.N(3313, 'fn', fn, 0);
                            J$.N(3321, 'receiver', receiver, 0);
                            J$.N(3329, 'arg', arg, 0);
                            while (J$.X1(4409, J$.C(96, J$.B(66, '>', J$.M(3097, J$.R(3089, 'queue', queue, 0), 'length', 0)(), J$.T(3105, 0, 22, false), 0)))) {
                                var fn = J$.X1(3137, J$.W(3129, 'fn', J$.M(3121, J$.R(3113, 'queue', queue, 0), 'shift', 0)(), fn, 1));
                                if (J$.X1(4401, J$.C(88, J$.B(82, '!==', J$.U(74, 'typeof', J$.R(3145, 'fn', fn, 0)), J$.T(3153, 'function', 21, false), 0)))) {
                                    J$.X1(3177, J$.M(3169, J$.R(3161, 'fn', fn, 0), '_settlePromises', 0)());
                                    continue;
                                }
                                var receiver = J$.X1(3209, J$.W(3201, 'receiver', J$.M(3193, J$.R(3185, 'queue', queue, 0), 'shift', 0)(), receiver, 1));
                                var arg = J$.X1(3241, J$.W(3233, 'arg', J$.M(3225, J$.R(3217, 'queue', queue, 0), 'shift', 0)(), arg, 1));
                                J$.X1(3281, J$.M(3273, J$.R(3249, 'fn', fn, 0), 'call', 0)(J$.R(3257, 'receiver', receiver, 0), J$.R(3265, 'arg', arg, 0)));
                            }
                        } catch (J$e) {
                            J$.Ex(4417, J$e);
                        } finally {
                            if (J$.Fr(4425))
                                continue jalangiLabel203;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 3289), 0));
            J$.X1(3545, J$.P(3537, J$.G(3369, J$.R(3361, 'Async', Async, 1), 'prototype', 0), '_drainQueues', J$.T(3529, function () {
                jalangiLabel204:
                    while (true) {
                        try {
                            J$.Fe(3513, arguments.callee, this, arguments);
                            arguments = J$.N(3521, 'arguments', arguments, 4);
                            J$.X1(3409, J$.M(3401, J$.R(3377, 'this', this, 0), '_drainQueue', 0)(J$.G(3393, J$.R(3385, 'this', this, 0), '_normalQueue', 0)));
                            J$.X1(3433, J$.M(3425, J$.R(3417, 'this', this, 0), '_reset', 0)());
                            J$.X1(3465, J$.P(3457, J$.R(3441, 'this', this, 0), '_haveDrainedQueues', J$.T(3449, true, 23, false), 0));
                            J$.X1(3505, J$.M(3497, J$.R(3473, 'this', this, 0), '_drainQueue', 0)(J$.G(3489, J$.R(3481, 'this', this, 0), '_lateQueue', 0)));
                        } catch (J$e) {
                            J$.Ex(4433, J$e);
                        } finally {
                            if (J$.Fr(4441))
                                continue jalangiLabel204;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 3513), 0));
            J$.X1(3689, J$.P(3681, J$.G(3561, J$.R(3553, 'Async', Async, 1), 'prototype', 0), '_queueTick', J$.T(3673, function () {
                jalangiLabel205:
                    while (true) {
                        try {
                            J$.Fe(3657, arguments.callee, this, arguments);
                            arguments = J$.N(3665, 'arguments', arguments, 4);
                            if (J$.X1(4449, J$.C(104, J$.U(90, '!', J$.G(3577, J$.R(3569, 'this', this, 0), '_isTickUsed', 0))))) {
                                J$.X1(3609, J$.P(3601, J$.R(3585, 'this', this, 0), '_isTickUsed', J$.T(3593, true, 23, false), 0));
                                J$.X1(3649, J$.M(3641, J$.R(3617, 'this', this, 0), '_schedule', 0)(J$.G(3633, J$.R(3625, 'this', this, 0), 'drainQueues', 0)));
                            }
                        } catch (J$e) {
                            J$.Ex(4457, J$e);
                        } finally {
                            if (J$.Fr(4465))
                                continue jalangiLabel205;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 3657), 0));
            J$.X1(3777, J$.P(3769, J$.G(3705, J$.R(3697, 'Async', Async, 1), 'prototype', 0), '_reset', J$.T(3761, function () {
                jalangiLabel206:
                    while (true) {
                        try {
                            J$.Fe(3745, arguments.callee, this, arguments);
                            arguments = J$.N(3753, 'arguments', arguments, 4);
                            J$.X1(3737, J$.P(3729, J$.R(3713, 'this', this, 0), '_isTickUsed', J$.T(3721, false, 23, false), 0));
                        } catch (J$e) {
                            J$.Ex(4473, J$e);
                        } finally {
                            if (J$.Fr(4481))
                                continue jalangiLabel206;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 3745), 0));
            J$.X1(3809, J$.P(3801, J$.R(3785, 'module', module, 2), 'exports', J$.R(3793, 'Async', Async, 1), 0));
            J$.X1(3849, J$.P(3841, J$.G(3825, J$.R(3817, 'module', module, 2), 'exports', 0), 'firstLineError', J$.R(3833, 'firstLineError', firstLineError, 1), 0));
        } catch (J$e) {
            J$.Ex(4489, J$e);
        } finally {
            if (J$.Sr(4497)) {
                J$.L();
                continue jalangiLabel207;
            } else {
                J$.L();
                break jalangiLabel207;
            }
        }
    }
// JALANGI DO NOT INSTRUMENT
