// bluebird/js/release/es5_jalangi_.js
J$.iids = {"8":[19,23,19,57],"9":[1,20,1,27],"10":[5,12,5,30],"16":[19,23,19,75],"17":[1,28,1,32],"18":[19,23,19,34],"24":[29,17,29,33],"25":[1,20,1,33],"26":[19,21,19,76],"32":[7,5,7,10],"33":[1,20,1,33],"34":[19,20,19,76],"41":[1,20,1,33],"42":[54,20,54,54],"49":[2,26,2,38],"57":[2,26,2,45],"65":[2,26,2,45],"73":[2,26,2,45],"81":[4,5,4,17],"89":[4,5,4,18],"97":[5,12,5,16],"105":[5,21,5,30],"113":[5,12,5,30],"121":[5,5,5,31],"129":[3,13,6,2],"137":[3,13,6,2],"145":[3,13,6,2],"153":[3,13,6,4],"161":[3,13,6,4],"169":[3,13,6,4],"177":[7,5,7,10],"185":[8,5,8,11],"193":[9,17,9,23],"201":[9,17,9,30],"209":[10,25,10,31],"217":[10,25,10,46],"225":[11,24,11,30],"233":[11,24,11,55],"241":[12,15,12,21],"249":[12,15,12,26],"257":[13,16,13,22],"265":[13,16,13,42],"273":[14,25,14,31],"281":[14,25,14,46],"289":[15,18,15,23],"297":[15,18,15,31],"305":[16,16,16,21],"313":[18,30,18,36],"321":[18,62,18,65],"329":[18,67,18,71],"337":[18,30,18,72],"339":[18,30,18,61],"345":[18,30,18,72],"353":[18,30,18,72],"361":[19,24,19,34],"369":[19,38,19,48],"377":[19,38,19,57],"385":[19,61,19,71],"393":[19,61,19,75],"401":[19,20,19,76],"409":[19,13,19,77],"417":[17,29,20,10],"425":[17,29,20,10],"433":[17,29,20,10],"441":[17,29,20,10],"449":[17,29,20,10],"457":[17,29,20,10],"465":[8,22,21,6],"473":[8,5,21,6],"481":[8,5,21,7],"489":[23,15,23,17],"497":[23,15,23,32],"505":[23,15,23,32],"513":[23,15,23,32],"521":[24,15,24,17],"529":[24,15,24,26],"537":[24,15,24,26],"545":[24,15,24,26],"553":[25,17,25,19],"561":[25,17,25,31],"569":[25,17,25,41],"577":[25,17,25,41],"585":[25,17,25,41],"593":[27,19,27,21],"601":[27,19,27,21],"609":[27,19,27,21],"617":[28,25,28,26],"625":[29,17,29,20],"633":[29,26,29,27],"641":[29,29,29,32],"649":[29,17,29,33],"651":[29,17,29,25],"657":[30,17,30,20],"665":[30,26,30,29],"673":[30,17,30,30],"675":[30,17,30,25],"681":[30,17,30,31],"689":[28,25,28,26],"697":[28,9,32,10],"705":[28,9,32,10],"713":[33,16,33,19],"721":[33,16,33,19],"729":[33,9,33,20],"737":[26,22,34,6],"745":[26,22,34,6],"753":[26,22,34,6],"761":[26,22,34,6],"769":[26,22,34,6],"777":[26,22,34,6],"785":[26,22,34,6],"793":[26,22,34,6],"801":[36,25,36,26],"809":[36,27,36,30],"817":[36,25,36,31],"825":[36,16,36,33],"833":[36,16,36,33],"841":[36,9,36,34],"849":[35,31,37,6],"857":[35,31,37,6],"865":[35,31,37,6],"873":[35,31,37,6],"881":[35,31,37,6],"889":[35,31,37,6],"897":[35,31,37,6],"905":[39,9,39,10],"913":[39,11,39,14],"921":[39,18,39,22],"929":[39,18,39,28],"937":[39,9,39,28],"945":[39,9,39,29],"953":[40,16,40,17],"961":[40,16,40,17],"969":[40,9,40,18],"977":[38,32,41,6],"985":[38,32,41,6],"993":[38,32,41,6],"1001":[38,32,41,6],"1009":[38,32,41,6],"1017":[38,32,41,6],"1025":[38,32,41,6],"1033":[38,32,41,6],"1041":[43,16,43,19],"1049":[43,16,43,19],"1057":[43,9,43,20],"1065":[42,24,44,6],"1073":[42,24,44,6],"1081":[42,24,44,6],"1089":[42,24,44,6],"1097":[42,24,44,6],"1105":[42,24,44,6],"1113":[47,20,47,26],"1121":[47,27,47,30],"1129":[47,20,47,31],"1137":[47,20,47,43],"1145":[47,20,47,53],"1153":[47,20,47,53],"1161":[47,13,47,54],"1169":[49,20,49,25],"1177":[49,20,49,25],"1185":[49,13,49,26],"1193":[48,11,50,10],"1201":[45,32,51,6],"1209":[45,32,51,6],"1217":[45,32,51,6],"1225":[45,32,51,6],"1233":[45,32,51,6],"1241":[45,32,51,6],"1249":[54,20,54,23],"1257":[54,29,54,32],"1265":[54,20,54,33],"1267":[54,20,54,28],"1273":[54,38,54,54],"1281":[54,20,54,54],"1289":[54,13,54,55],"1297":[56,20,56,25],"1305":[56,20,56,25],"1313":[56,13,56,26],"1321":[55,11,57,10],"1329":[52,24,58,6],"1337":[52,24,58,6],"1345":[52,24,58,6],"1353":[52,24,58,6],"1361":[52,24,58,6],"1369":[52,24,58,6],"1377":[59,5,59,11],"1385":[60,18,60,30],"1393":[61,15,61,25],"1401":[62,16,62,26],"1409":[63,25,63,45],"1417":[64,24,64,43],"1425":[65,17,65,29],"1433":[66,25,66,45],"1441":[67,16,67,21],"1449":[69,20,69,24],"1457":[69,20,69,24],"1465":[69,13,69,25],"1473":[68,29,70,10],"1481":[68,29,70,10],"1489":[68,29,70,10],"1497":[59,22,71,6],"1505":[59,5,71,6],"1513":[59,5,71,7],"1521":[1,1,72,2],"1529":[1,1,72,2],"1537":[1,1,72,2],"1545":[1,1,72,2],"1553":[1,1,72,2],"1561":[1,1,72,2],"1569":[1,1,72,2],"1577":[1,1,72,2],"1585":[1,1,72,2],"1593":[1,1,72,2],"1601":[1,1,72,2],"1609":[1,1,72,2],"1617":[1,1,72,2],"1625":[3,13,6,2],"1633":[3,13,6,2],"1641":[17,29,20,10],"1649":[17,29,20,10],"1657":[29,13,31,14],"1665":[26,22,34,6],"1673":[26,22,34,6],"1681":[35,31,37,6],"1689":[35,31,37,6],"1697":[38,32,41,6],"1705":[38,32,41,6],"1713":[42,24,44,6],"1721":[42,24,44,6],"1729":[45,32,51,6],"1737":[45,32,51,6],"1745":[52,24,58,6],"1753":[52,24,58,6],"1761":[68,29,70,10],"1769":[68,29,70,10],"1777":[7,1,72,2],"1785":[1,1,72,2],"1793":[1,1,72,2],"nBranches":8,"originalCodeFileName":"/home/blake/ExpoSE/Targets/hapigerjs/node_modules/bluebird/js/release/es5.js","instrumentedCodeFileName":"/home/blake/ExpoSE/Targets/hapigerjs/node_modules/bluebird/js/release/es5_jalangi_.js","code":"var isES5 = (function(){\n    \"use strict\";\n    return this === undefined;\n})();\n\nif (isES5) {\n    module.exports = {\n        freeze: Object.freeze,\n        defineProperty: Object.defineProperty,\n        getDescriptor: Object.getOwnPropertyDescriptor,\n        keys: Object.keys,\n        names: Object.getOwnPropertyNames,\n        getPrototypeOf: Object.getPrototypeOf,\n        isArray: Array.isArray,\n        isES5: isES5,\n        propertyIsWritable: function(obj, prop) {\n            var descriptor = Object.getOwnPropertyDescriptor(obj, prop);\n            return !!(!descriptor || descriptor.writable || descriptor.set);\n        }\n    };\n} else {\n    var has = {}.hasOwnProperty;\n    var str = {}.toString;\n    var proto = {}.constructor.prototype;\n\n    var ObjectKeys = function (o) {\n        var ret = [];\n        for (var key in o) {\n            if (has.call(o, key)) {\n                ret.push(key);\n            }\n        }\n        return ret;\n    };\n\n    var ObjectGetDescriptor = function(o, key) {\n        return {value: o[key]};\n    };\n\n    var ObjectDefineProperty = function (o, key, desc) {\n        o[key] = desc.value;\n        return o;\n    };\n\n    var ObjectFreeze = function (obj) {\n        return obj;\n    };\n\n    var ObjectGetPrototypeOf = function (obj) {\n        try {\n            return Object(obj).constructor.prototype;\n        }\n        catch (e) {\n            return proto;\n        }\n    };\n\n    var ArrayIsArray = function (obj) {\n        try {\n            return str.call(obj) === \"[object Array]\";\n        }\n        catch(e) {\n            return false;\n        }\n    };\n\n    module.exports = {\n        isArray: ArrayIsArray,\n        keys: ObjectKeys,\n        names: ObjectKeys,\n        defineProperty: ObjectDefineProperty,\n        getDescriptor: ObjectGetDescriptor,\n        freeze: ObjectFreeze,\n        getPrototypeOf: ObjectGetPrototypeOf,\n        isES5: isES5,\n        propertyIsWritable: function() {\n            return true;\n        }\n    };\n}\n"};
jalangiLabel180:
    while (true) {
        try {
            J$.Se(1521, '/home/blake/ExpoSE/Targets/hapigerjs/node_modules/bluebird/js/release/es5_jalangi_.js', '/home/blake/ExpoSE/Targets/hapigerjs/node_modules/bluebird/js/release/es5.js');
            J$.N(1529, '__secret__S$', __secret__S$, 0);
            J$.N(1537, '__secret__traits__', __secret__traits__, 0);
            J$.N(1545, 'isES5', isES5, 0);
            J$.N(1553, 'has', has, 0);
            J$.N(1561, 'str', str, 0);
            J$.N(1569, 'proto', proto, 0);
            J$.N(1577, 'ObjectKeys', ObjectKeys, 0);
            J$.N(1585, 'ObjectGetDescriptor', ObjectGetDescriptor, 0);
            J$.N(1593, 'ObjectDefineProperty', ObjectDefineProperty, 0);
            J$.N(1601, 'ObjectFreeze', ObjectFreeze, 0);
            J$.N(1609, 'ObjectGetPrototypeOf', ObjectGetPrototypeOf, 0);
            J$.N(1617, 'ArrayIsArray', ArrayIsArray, 0);
            var __secret__S$ = J$.X1(41, J$.W(33, '__secret__S$', J$.F(25, J$.R(9, 'require', require, 2), 0)(J$.T(17, 'S$', 21, false)), __secret__S$, 3));
            var __secret__traits__ = J$.X1(73, J$.W(65, '__secret__traits__', J$.G(57, J$.R(49, '__secret__S$', __secret__S$, 1), 'Traits', 0), __secret__traits__, 3));
            var isES5 = J$.X1(169, J$.W(161, 'isES5', J$.F(153, J$.T(145, function () {
                jalangiLabel171:
                    while (true) {
                        try {
                            J$.Fe(129, arguments.callee, this, arguments);
                            arguments = J$.N(137, 'arguments', arguments, 4);
                            J$.X1(89, J$.T(81, 'use strict', 21, false));
                            return J$.X1(121, J$.Rt(113, J$.B(10, '===', J$.R(97, 'this', this, 0), J$.T(105, undefined, 24, false), 0)));
                        } catch (J$e) {
                            J$.Ex(1625, J$e);
                        } finally {
                            if (J$.Fr(1633))
                                continue jalangiLabel171;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 129), 0)(), isES5, 3));
            if (J$.X1(1777, J$.C(32, J$.R(177, 'isES5', isES5, 1)))) {
                J$.X1(481, J$.P(473, J$.R(185, 'module', module, 2), 'exports', J$.T(465, {
                    freeze: J$.G(201, J$.R(193, 'Object', Object, 2), 'freeze', 0),
                    defineProperty: J$.G(217, J$.R(209, 'Object', Object, 2), 'defineProperty', 0),
                    getDescriptor: J$.G(233, J$.R(225, 'Object', Object, 2), 'getOwnPropertyDescriptor', 0),
                    keys: J$.G(249, J$.R(241, 'Object', Object, 2), 'keys', 0),
                    names: J$.G(265, J$.R(257, 'Object', Object, 2), 'getOwnPropertyNames', 0),
                    getPrototypeOf: J$.G(281, J$.R(273, 'Object', Object, 2), 'getPrototypeOf', 0),
                    isArray: J$.G(297, J$.R(289, 'Array', Array, 2), 'isArray', 0),
                    isES5: J$.R(305, 'isES5', isES5, 1),
                    propertyIsWritable: J$.T(457, function (obj, prop) {
                        jalangiLabel172:
                            while (true) {
                                try {
                                    J$.Fe(417, arguments.callee, this, arguments);
                                    arguments = J$.N(425, 'arguments', arguments, 4);
                                    obj = J$.N(433, 'obj', obj, 4);
                                    prop = J$.N(441, 'prop', prop, 4);
                                    J$.N(449, 'descriptor', descriptor, 0);
                                    var descriptor = J$.X1(353, J$.W(345, 'descriptor', J$.M(337, J$.R(313, 'Object', Object, 2), 'getOwnPropertyDescriptor', 0)(J$.R(321, 'obj', obj, 0), J$.R(329, 'prop', prop, 0)), descriptor, 1));
                                    return J$.X1(409, J$.Rt(401, J$.U(34, '!', J$.U(26, '!', J$.C(16, J$.C(8, J$.U(18, '!', J$.R(361, 'descriptor', descriptor, 0))) ? J$._() : J$.G(377, J$.R(369, 'descriptor', descriptor, 0), 'writable', 0)) ? J$._() : J$.G(393, J$.R(385, 'descriptor', descriptor, 0), 'set', 0)))));
                                } catch (J$e) {
                                    J$.Ex(1641, J$e);
                                } finally {
                                    if (J$.Fr(1649))
                                        continue jalangiLabel172;
                                    else
                                        return J$.Ra();
                                }
                            }
                    }, 12, false, 417)
                }, 11, false), 0));
            } else {
                var has = J$.X1(513, J$.W(505, 'has', J$.G(497, J$.T(489, {}, 11, false), 'hasOwnProperty', 0), has, 3));
                var str = J$.X1(545, J$.W(537, 'str', J$.G(529, J$.T(521, {}, 11, false), 'toString', 0), str, 3));
                var proto = J$.X1(585, J$.W(577, 'proto', J$.G(569, J$.G(561, J$.T(553, {}, 11, false), 'constructor', 0), 'prototype', 0), proto, 3));
                var ObjectKeys = J$.X1(793, J$.W(785, 'ObjectKeys', J$.T(777, function (o) {
                    jalangiLabel173:
                        while (true) {
                            try {
                                J$.Fe(737, arguments.callee, this, arguments);
                                arguments = J$.N(745, 'arguments', arguments, 4);
                                o = J$.N(753, 'o', o, 4);
                                J$.N(761, 'ret', ret, 0);
                                J$.N(769, 'key', key, 0);
                                var ret = J$.X1(609, J$.W(601, 'ret', J$.T(593, [], 10, false), ret, 1));
                                for (J$._tm_p in J$.H(689, J$.R(617, 'o', o, 0))) {
                                    var key = J$.X1(705, J$.W(697, 'key', J$._tm_p, key, 1));
                                    {
                                        {
                                            if (J$.X1(1657, J$.C(24, J$.M(649, J$.R(625, 'has', has, 1), 'call', 0)(J$.R(633, 'o', o, 0), J$.R(641, 'key', key, 0))))) {
                                                J$.X1(681, J$.M(673, J$.R(657, 'ret', ret, 0), 'push', 0)(J$.R(665, 'key', key, 0)));
                                            }
                                        }
                                    }
                                }
                                return J$.X1(729, J$.Rt(721, J$.R(713, 'ret', ret, 0)));
                            } catch (J$e) {
                                J$.Ex(1665, J$e);
                            } finally {
                                if (J$.Fr(1673))
                                    continue jalangiLabel173;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false, 737), ObjectKeys, 3));
                var ObjectGetDescriptor = J$.X1(897, J$.W(889, 'ObjectGetDescriptor', J$.T(881, function (o, key) {
                    jalangiLabel174:
                        while (true) {
                            try {
                                J$.Fe(849, arguments.callee, this, arguments);
                                arguments = J$.N(857, 'arguments', arguments, 4);
                                o = J$.N(865, 'o', o, 4);
                                key = J$.N(873, 'key', key, 4);
                                return J$.X1(841, J$.Rt(833, J$.T(825, {
                                    value: J$.G(817, J$.R(801, 'o', o, 0), J$.R(809, 'key', key, 0), 4)
                                }, 11, false)));
                            } catch (J$e) {
                                J$.Ex(1681, J$e);
                            } finally {
                                if (J$.Fr(1689))
                                    continue jalangiLabel174;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false, 849), ObjectGetDescriptor, 3));
                var ObjectDefineProperty = J$.X1(1033, J$.W(1025, 'ObjectDefineProperty', J$.T(1017, function (o, key, desc) {
                    jalangiLabel175:
                        while (true) {
                            try {
                                J$.Fe(977, arguments.callee, this, arguments);
                                arguments = J$.N(985, 'arguments', arguments, 4);
                                o = J$.N(993, 'o', o, 4);
                                key = J$.N(1001, 'key', key, 4);
                                desc = J$.N(1009, 'desc', desc, 4);
                                J$.X1(945, J$.P(937, J$.R(905, 'o', o, 0), J$.R(913, 'key', key, 0), J$.G(929, J$.R(921, 'desc', desc, 0), 'value', 0), 2));
                                return J$.X1(969, J$.Rt(961, J$.R(953, 'o', o, 0)));
                            } catch (J$e) {
                                J$.Ex(1697, J$e);
                            } finally {
                                if (J$.Fr(1705))
                                    continue jalangiLabel175;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false, 977), ObjectDefineProperty, 3));
                var ObjectFreeze = J$.X1(1105, J$.W(1097, 'ObjectFreeze', J$.T(1089, function (obj) {
                    jalangiLabel176:
                        while (true) {
                            try {
                                J$.Fe(1065, arguments.callee, this, arguments);
                                arguments = J$.N(1073, 'arguments', arguments, 4);
                                obj = J$.N(1081, 'obj', obj, 4);
                                return J$.X1(1057, J$.Rt(1049, J$.R(1041, 'obj', obj, 0)));
                            } catch (J$e) {
                                J$.Ex(1713, J$e);
                            } finally {
                                if (J$.Fr(1721))
                                    continue jalangiLabel176;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false, 1065), ObjectFreeze, 3));
                var ObjectGetPrototypeOf = J$.X1(1241, J$.W(1233, 'ObjectGetPrototypeOf', J$.T(1225, function (obj) {
                    jalangiLabel177:
                        while (true) {
                            try {
                                J$.Fe(1201, arguments.callee, this, arguments);
                                arguments = J$.N(1209, 'arguments', arguments, 4);
                                obj = J$.N(1217, 'obj', obj, 4);
                                try {
                                    return J$.X1(1161, J$.Rt(1153, J$.G(1145, J$.G(1137, J$.F(1129, J$.R(1113, 'Object', Object, 2), 0)(J$.R(1121, 'obj', obj, 0)), 'constructor', 0), 'prototype', 0)));
                                } catch (e) {
                                    e = J$.N(1193, 'e', e, 1);
                                    return J$.X1(1185, J$.Rt(1177, J$.R(1169, 'proto', proto, 1)));
                                }
                            } catch (J$e) {
                                J$.Ex(1729, J$e);
                            } finally {
                                if (J$.Fr(1737))
                                    continue jalangiLabel177;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false, 1201), ObjectGetPrototypeOf, 3));
                var ArrayIsArray = J$.X1(1369, J$.W(1361, 'ArrayIsArray', J$.T(1353, function (obj) {
                    jalangiLabel178:
                        while (true) {
                            try {
                                J$.Fe(1329, arguments.callee, this, arguments);
                                arguments = J$.N(1337, 'arguments', arguments, 4);
                                obj = J$.N(1345, 'obj', obj, 4);
                                try {
                                    return J$.X1(1289, J$.Rt(1281, J$.B(42, '===', J$.M(1265, J$.R(1249, 'str', str, 1), 'call', 0)(J$.R(1257, 'obj', obj, 0)), J$.T(1273, '[object Array]', 21, false), 0)));
                                } catch (e) {
                                    e = J$.N(1321, 'e', e, 1);
                                    return J$.X1(1313, J$.Rt(1305, J$.T(1297, false, 23, false)));
                                }
                            } catch (J$e) {
                                J$.Ex(1745, J$e);
                            } finally {
                                if (J$.Fr(1753))
                                    continue jalangiLabel178;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false, 1329), ArrayIsArray, 3));
                J$.X1(1513, J$.P(1505, J$.R(1377, 'module', module, 2), 'exports', J$.T(1497, {
                    isArray: J$.R(1385, 'ArrayIsArray', ArrayIsArray, 1),
                    keys: J$.R(1393, 'ObjectKeys', ObjectKeys, 1),
                    names: J$.R(1401, 'ObjectKeys', ObjectKeys, 1),
                    defineProperty: J$.R(1409, 'ObjectDefineProperty', ObjectDefineProperty, 1),
                    getDescriptor: J$.R(1417, 'ObjectGetDescriptor', ObjectGetDescriptor, 1),
                    freeze: J$.R(1425, 'ObjectFreeze', ObjectFreeze, 1),
                    getPrototypeOf: J$.R(1433, 'ObjectGetPrototypeOf', ObjectGetPrototypeOf, 1),
                    isES5: J$.R(1441, 'isES5', isES5, 1),
                    propertyIsWritable: J$.T(1489, function () {
                        jalangiLabel179:
                            while (true) {
                                try {
                                    J$.Fe(1473, arguments.callee, this, arguments);
                                    arguments = J$.N(1481, 'arguments', arguments, 4);
                                    return J$.X1(1465, J$.Rt(1457, J$.T(1449, true, 23, false)));
                                } catch (J$e) {
                                    J$.Ex(1761, J$e);
                                } finally {
                                    if (J$.Fr(1769))
                                        continue jalangiLabel179;
                                    else
                                        return J$.Ra();
                                }
                            }
                    }, 12, false, 1473)
                }, 11, false), 0));
            }
        } catch (J$e) {
            J$.Ex(1785, J$e);
        } finally {
            if (J$.Sr(1793)) {
                J$.L();
                continue jalangiLabel180;
            } else {
                J$.L();
                break jalangiLabel180;
            }
        }
    }
// JALANGI DO NOT INSTRUMENT
