// elizabot/elizadata_jalangi_.js
J$.iids = {"9":[1,20,1,27],"17":[1,28,1,32],"25":[1,20,1,33],"33":[1,20,1,33],"41":[1,20,1,33],"49":[2,26,2,38],"57":[2,26,2,45],"65":[2,26,2,45],"73":[2,26,2,45],"81":[3,1,3,8],"89":[4,5,4,51],"97":[5,5,5,49],"105":[6,5,6,35],"113":[3,25,7,2],"121":[3,1,7,2],"129":[3,1,7,3],"137":[8,1,8,8],"145":[9,5,9,44],"153":[10,5,10,45],"161":[11,5,11,58],"169":[12,5,12,78],"177":[13,5,13,76],"185":[8,23,14,2],"193":[8,1,14,2],"201":[8,1,14,3],"209":[15,1,15,8],"217":[16,5,16,10],"225":[17,5,17,14],"233":[18,5,18,11],"241":[19,5,19,11],"249":[20,5,20,11],"257":[15,22,21,2],"265":[15,1,21,2],"273":[15,1,21,3],"281":[22,1,22,8],"289":[23,5,23,11],"297":[24,5,24,13],"305":[25,5,25,11],"313":[26,5,26,13],"321":[27,5,27,11],"329":[28,5,28,13],"337":[29,5,29,16],"345":[30,5,30,15],"353":[31,5,31,13],"361":[32,5,32,15],"369":[33,5,33,13],"377":[34,5,34,14],"385":[35,5,35,13],"393":[36,5,36,12],"401":[37,5,37,12],"409":[38,5,38,14],"417":[39,5,39,16],"425":[40,5,40,10],"433":[41,5,41,14],"441":[42,5,42,15],"449":[43,5,43,15],"457":[44,5,44,15],"465":[45,5,45,16],"473":[46,5,46,15],"481":[47,5,47,11],"489":[48,5,48,10],"497":[49,5,49,14],"505":[50,5,50,14],"513":[51,5,51,11],"521":[52,5,52,11],"529":[53,5,53,11],"537":[54,5,54,12],"545":[55,5,55,16],"553":[56,5,56,12],"561":[57,5,57,17],"569":[58,5,58,12],"577":[22,21,59,2],"585":[22,1,59,2],"593":[22,1,59,3],"601":[60,1,60,8],"609":[61,5,61,9],"617":[62,5,62,10],"625":[63,5,63,11],"633":[64,5,64,9],"641":[65,5,65,9],"649":[66,5,66,10],"657":[67,5,67,13],"665":[68,5,68,15],"673":[69,5,69,15],"681":[70,5,70,13],"689":[71,5,71,8],"697":[72,5,72,10],"705":[73,5,73,10],"713":[74,5,74,8],"721":[75,5,75,9],"729":[76,5,76,11],"737":[77,5,77,11],"745":[78,5,78,14],"753":[60,22,79,2],"761":[60,1,79,2],"769":[60,1,79,3],"777":[80,1,80,8],"785":[82,9,82,13],"793":[83,9,83,13],"801":[84,9,84,14],"809":[85,9,85,14],"817":[81,11,86,6],"825":[88,9,88,15],"833":[89,9,89,16],"841":[90,9,90,18],"849":[91,9,91,15],"857":[87,15,92,6],"865":[93,16,93,24],"873":[93,15,93,25],"881":[95,9,95,15],"889":[96,9,96,15],"897":[94,15,97,6],"905":[99,9,99,20],"913":[100,9,100,17],"921":[101,9,101,16],"929":[98,17,102,6],"937":[104,9,104,17],"945":[105,9,105,14],"953":[106,9,106,17],"961":[107,9,107,14],"969":[108,9,108,17],"977":[109,9,109,18],"985":[110,9,110,15],"993":[111,9,111,19],"1001":[112,9,112,16],"1009":[103,15,113,6],"1017":[115,9,115,17],"1025":[116,9,116,15],"1033":[117,9,117,17],"1041":[114,14,118,6],"1049":[120,9,120,18],"1057":[121,9,121,20],"1065":[122,9,122,15],"1073":[119,12,123,6],"1081":[80,23,124,2],"1089":[80,1,124,2],"1097":[80,1,124,3],"1105":[125,1,125,8],"1113":[127,9,127,16],"1121":[128,9,128,10],"1129":[130,17,130,20],"1137":[132,21,132,60],"1145":[133,21,133,36],"1153":[134,21,134,54],"1161":[135,21,135,74],"1169":[136,21,136,61],"1177":[137,21,137,47],"1185":[138,21,138,59],"1193":[131,17,139,18],"1201":[129,10,140,14],"1209":[129,9,140,15],"1217":[126,5,141,6],"1225":[143,9,143,16],"1233":[144,9,144,10],"1241":[146,17,146,20],"1249":[148,21,148,47],"1257":[149,21,149,51],"1265":[150,21,150,70],"1273":[151,21,151,62],"1281":[147,17,152,18],"1289":[145,10,153,14],"1297":[145,9,153,15],"1305":[142,5,154,6],"1313":[156,9,156,20],"1321":[157,9,157,10],"1329":[159,17,159,20],"1337":[160,18,160,30],"1345":[160,17,160,31],"1353":[158,10,161,14],"1361":[158,9,161,15],"1369":[155,5,162,6],"1377":[164,9,164,19],"1385":[165,9,165,10],"1393":[168,17,168,33],"1401":[170,21,170,50],"1409":[171,21,171,73],"1417":[172,21,172,51],"1425":[173,21,173,57],"1433":[174,21,174,73],"1441":[175,21,175,66],"1449":[176,21,176,57],"1457":[169,17,177,18],"1465":[167,13,178,14],"1473":[180,17,180,38],"1481":[182,21,182,57],"1489":[183,21,183,65],"1497":[184,21,184,39],"1505":[185,21,185,32],"1513":[186,21,186,42],"1521":[181,17,187,18],"1529":[179,13,188,14],"1537":[190,17,190,35],"1545":[192,21,192,47],"1553":[193,21,193,57],"1561":[194,21,194,31],"1569":[191,17,195,18],"1577":[189,13,196,14],"1585":[166,9,197,10],"1593":[163,5,198,6],"1601":[200,9,200,17],"1609":[201,9,201,10],"1617":[204,17,204,31],"1625":[206,21,206,66],"1633":[207,21,207,52],"1641":[208,21,208,54],"1649":[209,21,209,58],"1657":[210,21,210,51],"1665":[211,21,211,52],"1673":[212,21,212,61],"1681":[205,17,213,18],"1689":[203,13,214,14],"1697":[216,17,216,37],"1705":[218,21,218,39],"1713":[219,21,219,49],"1721":[220,21,220,60],"1729":[221,21,221,57],"1737":[222,21,222,32],"1745":[223,21,223,46],"1753":[217,17,224,18],"1761":[215,13,225,14],"1769":[202,9,226,10],"1777":[199,5,227,6],"1785":[229,9,229,13],"1793":[230,9,230,10],"1801":[232,17,232,25],"1809":[234,21,234,59],"1817":[235,21,235,45],"1825":[236,21,236,51],"1833":[237,21,237,39],"1841":[238,21,238,49],"1849":[239,21,239,58],"1857":[240,21,240,59],"1865":[233,17,241,18],"1873":[231,10,242,14],"1881":[231,9,242,15],"1889":[228,5,243,6],"1897":[245,9,245,18],"1905":[246,9,246,10],"1913":[248,17,248,32],"1921":[250,21,250,36],"1929":[251,21,251,74],"1937":[252,21,252,57],"1945":[253,21,253,33],"1953":[249,17,254,18],"1961":[247,10,255,14],"1969":[247,9,255,15],"1977":[244,5,256,6],"1985":[258,9,258,16],"1993":[259,9,259,10],"2001":[261,17,261,20],"2009":[263,21,263,60],"2017":[264,21,264,43],"2025":[265,21,265,59],"2033":[266,21,266,90],"2041":[262,17,267,18],"2049":[260,10,268,14],"2057":[260,9,268,15],"2065":[257,5,269,6],"2073":[271,9,271,18],"2081":[272,9,272,10],"2089":[274,17,274,20],"2097":[276,21,276,53],"2105":[277,21,277,47],"2113":[278,21,278,52],"2121":[279,21,279,41],"2129":[280,21,280,40],"2137":[281,21,281,55],"2145":[275,17,282,18],"2153":[273,10,283,14],"2161":[273,9,283,15],"2169":[270,5,284,6],"2177":[286,9,286,15],"2185":[287,9,287,11],"2193":[289,17,289,20],"2201":[291,21,291,52],"2209":[292,21,292,91],"2217":[290,17,293,18],"2225":[288,10,294,14],"2233":[288,9,294,15],"2241":[285,5,295,6],"2249":[297,9,297,18],"2257":[298,9,298,10],"2265":[300,17,300,20],"2273":[302,21,302,36],"2281":[303,21,303,69],"2289":[301,17,304,18],"2297":[299,10,305,14],"2305":[299,9,305,15],"2313":[296,5,306,6],"2321":[308,9,308,19],"2329":[309,9,309,10],"2337":[311,17,311,20],"2345":[313,21,313,36],"2353":[314,21,314,69],"2361":[312,17,315,18],"2369":[310,10,316,14],"2377":[310,9,316,15],"2385":[307,5,317,6],"2393":[319,9,319,19],"2401":[320,9,320,10],"2409":[322,17,322,20],"2417":[324,21,324,36],"2425":[325,21,325,70],"2433":[323,17,326,18],"2441":[321,10,327,14],"2449":[321,9,327,15],"2457":[318,5,328,6],"2465":[330,9,330,18],"2473":[331,9,331,10],"2481":[333,17,333,20],"2489":[335,21,335,36],"2497":[336,21,336,70],"2505":[334,17,337,18],"2513":[332,10,338,14],"2521":[332,9,338,15],"2529":[329,5,339,6],"2537":[341,9,341,19],"2545":[342,9,342,10],"2553":[344,17,344,20],"2561":[345,18,345,41],"2569":[345,17,345,42],"2577":[343,10,346,14],"2585":[343,9,346,15],"2593":[340,5,347,6],"2601":[349,9,349,16],"2609":[350,9,350,10],"2617":[352,17,352,20],"2625":[354,21,354,65],"2633":[355,21,355,59],"2641":[353,17,356,18],"2649":[351,10,357,14],"2657":[351,9,357,15],"2665":[348,5,358,6],"2673":[360,9,360,19],"2681":[361,9,361,11],"2689":[363,17,363,20],"2697":[365,21,365,47],"2705":[366,21,366,53],"2713":[367,21,367,80],"2721":[368,21,368,67],"2729":[369,21,369,56],"2737":[370,21,370,57],"2745":[371,21,371,73],"2753":[364,17,372,18],"2761":[362,10,373,14],"2769":[362,9,373,15],"2777":[359,5,374,6],"2785":[376,9,376,13],"2793":[377,9,377,10],"2801":[380,17,380,27],"2809":[382,21,382,51],"2817":[383,21,383,49],"2825":[384,21,384,65],"2833":[385,21,385,59],"2841":[386,21,386,32],"2849":[381,17,387,18],"2857":[379,13,388,14],"2865":[390,17,390,27],"2873":[391,18,391,26],"2881":[391,17,391,27],"2889":[389,13,392,14],"2897":[394,17,394,20],"2905":[396,21,396,46],"2913":[397,21,397,48],"2921":[395,17,398,18],"2929":[393,13,399,14],"2937":[378,9,400,10],"2945":[375,5,401,6],"2953":[403,9,403,14],"2961":[404,9,404,10],"2969":[407,17,407,30],"2977":[409,21,409,74],"2985":[410,21,410,59],"2993":[411,21,411,58],"3001":[412,21,412,56],"3009":[413,21,413,32],"3017":[414,21,414,47],"3025":[415,21,415,43],"3033":[408,17,416,18],"3041":[406,13,417,14],"3049":[419,17,419,30],"3057":[420,18,420,28],"3065":[420,17,420,29],"3073":[418,13,421,14],"3081":[423,17,423,26],"3089":[425,21,425,60],"3097":[426,21,426,63],"3105":[427,21,427,50],"3113":[428,21,428,44],"3121":[429,21,429,45],"3129":[430,21,430,54],"3137":[424,17,431,18],"3145":[422,13,432,14],"3153":[405,9,433,10],"3161":[402,5,434,6],"3169":[436,9,436,15],"3177":[437,9,437,10],"3185":[439,17,439,27],"3193":[441,21,441,58],"3201":[442,21,442,48],"3209":[443,21,443,66],"3217":[444,21,444,39],"3225":[445,21,445,55],"3233":[446,21,446,43],"3241":[440,17,447,18],"3249":[438,10,448,14],"3257":[438,9,448,15],"3265":[435,5,449,6],"3273":[451,9,451,14],"3281":[452,9,452,10],"3289":[455,17,455,28],"3297":[457,21,457,45],"3305":[458,21,458,50],"3313":[459,21,459,37],"3321":[460,21,460,59],"3329":[461,21,461,59],"3337":[462,21,462,32],"3345":[456,17,463,18],"3353":[454,13,464,14],"3361":[466,17,466,28],"3369":[468,21,468,40],"3377":[469,21,469,60],"3385":[470,21,470,59],"3393":[467,17,471,18],"3401":[465,13,472,14],"3409":[474,17,474,30],"3417":[476,21,476,60],"3425":[477,21,477,53],"3433":[478,21,478,42],"3441":[479,21,479,41],"3449":[480,21,480,47],"3457":[475,17,481,18],"3465":[473,13,482,14],"3473":[453,9,483,10],"3481":[450,5,484,6],"3489":[486,9,486,12],"3497":[487,9,487,10],"3505":[490,17,490,32],"3513":[492,21,492,65],"3521":[493,21,493,44],"3529":[494,21,494,48],"3537":[495,21,495,50],"3545":[496,21,496,59],"3553":[497,21,497,78],"3561":[491,17,498,18],"3569":[489,13,499,14],"3577":[501,17,501,33],"3585":[503,21,503,59],"3593":[504,21,504,77],"3601":[505,21,505,62],"3609":[506,21,506,58],"3617":[502,17,507,18],"3625":[500,13,508,14],"3633":[510,17,510,35],"3641":[512,21,512,56],"3649":[513,21,513,56],"3657":[514,21,514,52],"3665":[515,21,515,65],"3673":[511,17,516,18],"3681":[509,13,517,14],"3689":[519,17,519,28],"3697":[520,18,520,28],"3705":[520,17,520,29],"3713":[518,13,521,14],"3721":[523,17,523,34],"3729":[525,21,525,47],"3737":[526,21,526,52],"3745":[527,21,527,52],"3753":[524,17,528,18],"3761":[522,13,529,14],"3769":[531,17,531,38],"3777":[532,18,532,28],"3785":[532,17,532,29],"3793":[530,13,533,14],"3801":[535,17,535,27],"3809":[537,21,537,70],"3817":[538,21,538,51],"3825":[539,21,539,62],"3833":[540,21,540,47],"3841":[541,21,541,59],"3849":[536,17,542,18],"3857":[534,13,543,14],"3865":[545,17,545,32],"3873":[547,21,547,60],"3881":[548,21,548,39],"3889":[549,21,549,49],"3897":[550,21,550,61],"3905":[551,21,551,46],"3913":[546,17,552,18],"3921":[544,13,553,14],"3929":[555,17,555,31],"3937":[557,21,557,46],"3945":[558,21,558,43],"3953":[559,21,559,54],"3961":[560,21,560,46],"3969":[556,17,561,18],"3977":[554,13,562,14],"3985":[564,17,564,29],"3993":[566,21,566,56],"4001":[567,21,567,46],"4009":[568,21,568,49],"4017":[569,21,569,60],"4025":[565,17,570,18],"4033":[563,13,571,14],"4041":[573,17,573,30],"4049":[575,21,575,67],"4057":[576,21,576,46],"4065":[577,21,577,50],"4073":[578,21,578,47],"4081":[574,17,579,18],"4089":[572,13,580,14],"4097":[582,17,582,20],"4105":[584,21,584,36],"4113":[585,21,585,50],"4121":[586,21,586,63],"4129":[587,21,587,49],"4137":[583,17,588,18],"4145":[581,13,589,14],"4153":[488,9,590,10],"4161":[485,5,591,6],"4169":[593,9,593,14],"4177":[594,9,594,10],"4185":[597,17,597,39],"4193":[598,18,598,30],"4201":[598,17,598,31],"4209":[596,13,599,14],"4217":[601,17,601,30],"4225":[603,21,603,54],"4233":[604,21,604,63],"4241":[605,21,605,59],"4249":[606,21,606,56],"4257":[602,17,607,18],"4265":[600,13,608,14],"4273":[610,17,610,30],"4281":[612,21,612,51],"4289":[613,21,613,66],"4297":[614,21,614,55],"4305":[615,21,615,42],"4313":[616,21,616,57],"4321":[617,21,617,70],"4329":[618,21,618,60],"4337":[611,17,619,18],"4345":[609,13,620,14],"4353":[622,17,622,26],"4361":[624,21,624,56],"4369":[625,21,625,34],"4377":[626,21,626,71],"4385":[627,21,627,51],"4393":[623,17,628,18],"4401":[621,13,629,14],"4409":[595,9,630,10],"4417":[592,5,631,6],"4425":[633,9,633,14],"4433":[634,9,634,10],"4441":[636,17,636,20],"4449":[638,21,638,53],"4457":[639,21,639,36],"4465":[640,21,640,29],"4473":[641,21,641,36],"4481":[637,17,642,18],"4489":[635,10,643,14],"4497":[635,9,643,15],"4505":[632,5,644,6],"4513":[646,9,646,13],"4521":[647,9,647,10],"4529":[650,17,650,29],"4537":[652,21,652,49],"4545":[653,21,653,43],"4553":[654,21,654,55],"4561":[655,21,655,66],"4569":[656,21,656,39],"4577":[657,21,657,73],"4585":[658,21,658,63],"4593":[651,17,659,18],"4601":[649,13,660,14],"4609":[662,17,662,20],"4617":[664,21,664,61],"4625":[665,21,665,52],"4633":[666,21,666,32],"4641":[667,21,667,35],"4649":[663,17,668,18],"4657":[661,13,669,14],"4665":[648,9,670,10],"4673":[645,5,671,6],"4681":[673,9,673,13],"4689":[674,9,674,10],"4697":[677,17,677,27],"4705":[679,21,679,82],"4713":[680,21,680,57],"4721":[681,21,681,49],"4729":[682,21,682,36],"4737":[678,17,683,18],"4745":[676,13,684,14],"4753":[686,17,686,34],"4761":[688,21,688,54],"4769":[689,21,689,52],"4777":[690,21,690,33],"4785":[691,21,691,80],"4793":[687,17,692,18],"4801":[685,13,693,14],"4809":[695,17,695,25],"4817":[697,21,697,33],"4825":[698,21,698,48],"4833":[699,21,699,77],"4841":[700,21,700,61],"4849":[696,17,701,18],"4857":[694,13,702,14],"4865":[675,9,703,10],"4873":[672,5,704,6],"4881":[706,9,706,14],"4889":[707,9,707,10],"4897":[710,17,710,30],"4905":[712,21,712,57],"4913":[713,21,713,32],"4921":[714,21,714,53],"4929":[715,21,715,73],"4937":[711,17,716,18],"4945":[709,13,717,14],"4953":[719,17,719,28],"4961":[721,21,721,81],"4969":[722,21,722,54],"4977":[723,21,723,54],"4985":[724,21,724,32],"4993":[720,17,725,18],"5001":[718,13,726,14],"5009":[708,9,727,10],"5017":[705,5,728,6],"5025":[730,9,730,15],"5033":[731,9,731,10],"5041":[733,17,733,20],"5049":[735,21,735,39],"5057":[736,21,736,56],"5065":[737,21,737,59],"5073":[738,21,738,61],"5081":[739,21,739,58],"5089":[740,21,740,42],"5097":[741,21,741,61],"5105":[742,21,742,61],"5113":[743,21,743,51],"5121":[734,17,744,18],"5129":[732,10,745,14],"5137":[732,9,745,15],"5145":[729,5,746,6],"5153":[748,9,748,14],"5161":[749,9,749,10],"5169":[751,17,751,24],"5177":[752,18,752,29],"5185":[752,17,752,30],"5193":[750,10,753,14],"5201":[750,9,753,15],"5209":[747,5,754,6],"5217":[756,9,756,15],"5225":[757,9,757,10],"5233":[759,17,759,25],"5241":[760,18,760,29],"5249":[760,17,760,30],"5257":[758,10,761,14],"5265":[758,9,761,15],"5273":[755,5,762,6],"5281":[764,9,764,16],"5289":[765,9,765,10],"5297":[767,17,767,26],"5305":[768,18,768,29],"5313":[768,17,768,30],"5321":[766,10,769,14],"5329":[766,9,769,15],"5337":[763,5,770,6],"5345":[772,9,772,14],"5353":[773,9,773,10],"5361":[775,17,775,24],"5369":[776,18,776,29],"5377":[776,17,776,30],"5385":[774,10,777,14],"5393":[774,9,777,15],"5401":[771,5,778,6],"5409":[780,9,780,18],"5417":[781,9,781,10],"5425":[783,17,783,20],"5433":[785,21,785,48],"5441":[786,21,786,62],"5449":[787,21,787,71],"5457":[788,21,788,58],"5465":[784,17,789,18],"5473":[782,10,790,14],"5481":[782,9,790,15],"5489":[779,5,791,6],"5497":[793,9,793,14],"5505":[794,9,794,10],"5513":[797,17,797,37],"5521":[799,21,799,52],"5529":[800,21,800,55],"5537":[801,21,801,48],"5545":[802,21,802,43],"5553":[803,21,803,32],"5561":[798,17,804,18],"5569":[796,13,805,14],"5577":[807,17,807,35],"5585":[809,21,809,63],"5593":[810,21,810,54],"5601":[811,21,811,65],"5609":[812,21,812,61],"5617":[813,21,813,32],"5625":[808,17,814,18],"5633":[806,13,815,14],"5641":[817,17,817,20],"5649":[818,18,818,29],"5657":[818,17,818,30],"5665":[816,13,819,14],"5673":[795,9,820,10],"5681":[792,5,821,6],"5689":[823,9,823,19],"5697":[824,9,824,10],"5705":[826,17,826,32],"5713":[828,21,828,36],"5721":[829,21,829,38],"5729":[830,21,830,62],"5737":[831,21,831,40],"5745":[832,21,832,66],"5753":[833,21,833,39],"5761":[834,21,834,48],"5769":[835,21,835,73],"5777":[836,21,836,63],"5785":[827,17,837,18],"5793":[825,10,838,14],"5801":[825,9,838,15],"5809":[822,5,839,6],"5817":[841,9,841,20],"5825":[842,9,842,10],"5833":[844,17,844,20],"5841":[845,18,845,33],"5849":[845,17,845,34],"5857":[843,10,846,14],"5865":[843,9,846,15],"5873":[840,5,847,6],"5881":[849,9,849,17],"5889":[850,9,850,10],"5897":[852,17,852,20],"5905":[853,18,853,33],"5913":[853,17,853,34],"5921":[851,10,854,14],"5929":[851,9,854,15],"5937":[848,5,855,6],"5945":[857,9,857,16],"5953":[858,9,858,10],"5961":[860,17,860,20],"5969":[861,18,861,33],"5977":[861,17,861,34],"5985":[859,10,862,14],"5993":[859,9,862,15],"6001":[856,5,863,6],"6009":[865,9,865,17],"6017":[866,9,866,10],"6025":[868,17,868,20],"6033":[870,21,870,60],"6041":[871,21,871,29],"6049":[872,21,872,58],"6057":[873,21,873,39],"6065":[869,17,874,18],"6073":[867,10,875,14],"6081":[867,9,875,15],"6089":[864,5,876,6],"6097":[878,9,878,16],"6105":[879,9,879,11],"6113":[881,17,881,20],"6121":[883,21,883,36],"6129":[884,21,884,52],"6137":[885,21,885,65],"6145":[886,21,886,58],"6153":[887,21,887,67],"6161":[888,21,888,63],"6169":[889,21,889,62],"6177":[890,21,890,28],"6185":[882,17,891,18],"6193":[880,10,892,14],"6201":[880,9,892,15],"6209":[877,5,893,6],"6217":[895,9,895,15],"6225":[896,9,896,11],"6233":[898,17,898,32],"6241":[899,18,899,30],"6249":[899,17,899,31],"6257":[897,10,900,14],"6265":[897,9,900,15],"6273":[894,5,901,6],"6281":[903,9,903,20],"6289":[904,9,904,10],"6297":[906,17,906,20],"6305":[908,21,908,44],"6313":[909,21,909,52],"6321":[910,21,910,65],"6329":[911,21,911,59],"6337":[912,21,912,65],"6345":[913,21,913,71],"6353":[914,21,914,28],"6361":[907,17,915,18],"6369":[905,10,916,14],"6377":[905,9,916,15],"6385":[902,5,917,6],"6393":[125,25,918,2],"6401":[125,1,918,2],"6409":[125,1,918,3],"6417":[919,1,919,8],"6425":[920,5,920,16],"6433":[921,5,921,11],"6441":[922,5,922,31],"6449":[923,5,923,18],"6457":[924,5,924,30],"6465":[925,5,925,17],"6473":[926,5,926,30],"6481":[927,5,927,17],"6489":[928,5,928,41],"6497":[929,5,929,32],"6505":[930,5,930,24],"6513":[931,5,931,16],"6521":[932,5,932,50],"6529":[933,5,933,40],"6537":[919,31,934,2],"6545":[919,1,934,2],"6553":[919,1,934,3],"6561":[1,1,934,3],"6569":[1,1,934,3],"6577":[1,1,934,3],"6585":[1,1,934,3],"6593":[1,1,934,3],"nBranches":0,"originalCodeFileName":"/home/blake/ExpoSE/Targets/elizabot/node_modules/elizabot/elizadata.js","instrumentedCodeFileName":"/home/blake/ExpoSE/Targets/elizabot/node_modules/elizabot/elizadata_jalangi_.js","code":"\n// start of file - elizadata.js\n// data for elizabot.js\n// entries prestructured as layed out in Weizenbaum's description\n// [cf: Communications of the ACM, Vol. 9, #1 (January 1966): p 36-45.]\n\nexports.elizaInitials = [\n\t\"How do you do.  Please tell me your problem.\",\n// additions (not original)\n\t\"Please tell me what's been bothering you.\",\n\t\"Is something troubling you ?\"\n];\n\nexports.elizaFinals = [\n\t\"Goodbye.  It was nice talking to you.\",\n// additions (not original)\n\t\"Goodbye.  This was really a nice talk.\",\n\t\"Goodbye.  I'm looking forward to our next session.\",\n\t\"This was a good session, wasn't it -- but time is over now.   Goodbye.\",\n\t\"Maybe we could discuss this moreover in our next session ?   Goodbye.\"\n];\n\nexports.elizaQuits = [\n\t\"bye\",\n\t\"goodbye\",\n\t\"done\",\n\t\"exit\",\n\t\"quit\"\n];\n\nexports.elizaPres = [\n\t\"dont\", \"don't\",\n\t\"cant\", \"can't\",\n\t\"wont\", \"won't\",\n\t\"recollect\", \"remember\",\n\t\"recall\", \"remember\",\n\t\"dreamt\", \"dreamed\",\n\t\"dreams\", \"dream\",\n\t\"maybe\", \"perhaps\",\n\t\"certainly\", \"yes\",\n\t\"machine\", \"computer\",\n\t\"machines\", \"computer\",\n\t\"computers\", \"computer\",\n\t\"were\", \"was\",\n\t\"you're\", \"you are\",\n\t\"i'm\", \"i am\",\n\t\"same\", \"alike\",\n\t\"identical\", \"alike\",\n\t\"equivalent\", \"alike\"\n];\n\nexports.elizaPosts = [\n\t\"am\", \"are\",\n\t\"your\", \"my\",\n\t\"me\", \"you\",\n\t\"myself\", \"yourself\",\n\t\"yourself\", \"myself\",\n\t\"i\", \"you\",\n\t\"you\", \"I\",\n\t\"my\", \"your\",\n\t\"i'm\", \"you are\"\n];\n\nexports.elizaSynons = {\n\t\"be\": [\"am\", \"is\", \"are\", \"was\"],\n\t\"belief\": [\"feel\", \"think\", \"believe\", \"wish\"],\n\t\"cannot\": [\"can't\"],\n\t\"desire\": [\"want\", \"need\"],\n\t\"everyone\": [\"everybody\", \"nobody\", \"noone\"],\n\t\"family\": [\"mother\", \"mom\", \"father\", \"dad\", \"sister\", \"brother\", \"wife\", \"children\", \"child\"],\n\t\"happy\": [\"elated\", \"glad\", \"better\"],\n\t\"sad\": [\"unhappy\", \"depressed\", \"sick\"]\n};\n\nexports.elizaKeywords = [\n\n\t/*\n\t Array of\n\t [\"<key>\", <rank>, [\n\t [\"<decomp>\", [\n\t \"<reasmb>\",\n\t \"<reasmb>\",\n\t \"<reasmb>\"\n\t ]],\n\t [\"<decomp>\", [\n\t \"<reasmb>\",\n\t \"<reasmb>\",\n\t \"<reasmb>\"\n\t ]]\n\t ]]\n\t */\n\n\t[\"xnone\", 0, [\n\t\t[\"*\", [\n\t\t\t\"I'm not sure I understand you fully.\",\n\t\t\t\"Please go on.\",\n\t\t\t\"What does that suggest to you ?\",\n\t\t\t\"Do you feel strongly about discussing such things ?\",\n\t\t\t\"That is interesting.  Please continue.\",\n\t\t\t\"Tell me more about that.\",\n\t\t\t\"Does talking about this bother you ?\"\n\t\t]]\n\t]],\n\t[\"sorry\", 0, [\n\t\t[\"*\", [\n\t\t\t\"Please don't apologise.\",\n\t\t\t\"Apologies are not necessary.\",\n\t\t\t\"I've told you that apologies are not required.\",\n\t\t\t\"It did not bother me.  Please continue.\"\n\t\t]]\n\t]],\n\t[\"apologise\", 0, [\n\t\t[\"*\", [\n\t\t\t\"goto sorry\"\n\t\t]]\n\t]],\n\t[\"remember\", 5, [\n\t\t[\"* i remember *\", [\n\t\t\t\"Do you often think of (2) ?\",\n\t\t\t\"Does thinking of (2) bring anything else to mind ?\",\n\t\t\t\"What else do you recollect ?\",\n\t\t\t\"Why do you remember (2) just now ?\",\n\t\t\t\"What in the present situation reminds you of (2) ?\",\n\t\t\t\"What is the connection between me and (2) ?\",\n\t\t\t\"What else does (2) remind you of ?\"\n\t\t]],\n\t\t[\"* do you remember *\", [\n\t\t\t\"Did you think I would forget (2) ?\",\n\t\t\t\"Why do you think I should recall (2) now ?\",\n\t\t\t\"What about (2) ?\",\n\t\t\t\"goto what\",\n\t\t\t\"You mentioned (2) ?\"\n\t\t]],\n\t\t[\"* you remember *\", [\n\t\t\t\"How could I forget (2) ?\",\n\t\t\t\"What about (2) should I remember ?\",\n\t\t\t\"goto you\"\n\t\t]]\n\t]],\n\t[\"forget\", 5, [\n\t\t[\"* i forget *\", [\n\t\t\t\"Can you think of why you might forget (2) ?\",\n\t\t\t\"Why can't you remember (2) ?\",\n\t\t\t\"How often do you think of (2) ?\",\n\t\t\t\"Does it bother you to forget that ?\",\n\t\t\t\"Could it be a mental block ?\",\n\t\t\t\"Are you generally forgetful ?\",\n\t\t\t\"Do you think you are suppressing (2) ?\"\n\t\t]],\n\t\t[\"* did you forget *\", [\n\t\t\t\"Why do you ask ?\",\n\t\t\t\"Are you sure you told me ?\",\n\t\t\t\"Would it bother you if I forgot (2) ?\",\n\t\t\t\"Why should I recall (2) just now ?\",\n\t\t\t\"goto what\",\n\t\t\t\"Tell me more about (2).\"\n\t\t]]\n\t]],\n\t[\"if\", 3, [\n\t\t[\"* if *\", [\n\t\t\t\"Do you think it's likely that (2) ?\",\n\t\t\t\"Do you wish that (2) ?\",\n\t\t\t\"What do you know about (2) ?\",\n\t\t\t\"Really, if (2) ?\",\n\t\t\t\"What would you do if (2) ?\",\n\t\t\t\"But what are the chances that (2) ?\",\n\t\t\t\"What does this speculation lead to ?\"\n\t\t]]\n\t]],\n\t[\"dreamed\", 4, [\n\t\t[\"* i dreamed *\", [\n\t\t\t\"Really, (2) ?\",\n\t\t\t\"Have you ever fantasized (2) while you were awake ?\",\n\t\t\t\"Have you ever dreamed (2) before ?\",\n\t\t\t\"goto dream\"\n\t\t]]\n\t]],\n\t[\"dream\", 3, [\n\t\t[\"*\", [\n\t\t\t\"What does that dream suggest to you ?\",\n\t\t\t\"Do you dream often ?\",\n\t\t\t\"What persons appear in your dreams ?\",\n\t\t\t\"Do you believe that dreams have something to do with your problem ?\"\n\t\t]]\n\t]],\n\t[\"perhaps\", 0, [\n\t\t[\"*\", [\n\t\t\t\"You don't seem quite certain.\",\n\t\t\t\"Why the uncertain tone ?\",\n\t\t\t\"Can't you be more positive ?\",\n\t\t\t\"You aren't sure ?\",\n\t\t\t\"Don't you know ?\",\n\t\t\t\"How likely, would you estimate ?\"\n\t\t]]\n\t]],\n\t[\"name\", 15, [\n\t\t[\"*\", [\n\t\t\t\"I am not interested in names.\",\n\t\t\t\"I've told you before, I don't care about names -- please continue.\"\n\t\t]]\n\t]],\n\t[\"deutsch\", 0, [\n\t\t[\"*\", [\n\t\t\t\"goto xforeign\",\n\t\t\t\"I told you before, I don't understand German.\"\n\t\t]]\n\t]],\n\t[\"francais\", 0, [\n\t\t[\"*\", [\n\t\t\t\"goto xforeign\",\n\t\t\t\"I told you before, I don't understand French.\"\n\t\t]]\n\t]],\n\t[\"italiano\", 0, [\n\t\t[\"*\", [\n\t\t\t\"goto xforeign\",\n\t\t\t\"I told you before, I don't understand Italian.\"\n\t\t]]\n\t]],\n\t[\"espanol\", 0, [\n\t\t[\"*\", [\n\t\t\t\"goto xforeign\",\n\t\t\t\"I told you before, I don't understand Spanish.\"\n\t\t]]\n\t]],\n\t[\"xforeign\", 0, [\n\t\t[\"*\", [\n\t\t\t\"I speak only English.\"\n\t\t]]\n\t]],\n\t[\"hello\", 0, [\n\t\t[\"*\", [\n\t\t\t\"How do you do.  Please state your problem.\",\n\t\t\t\"Hi.  What seems to be your problem ?\"\n\t\t]]\n\t]],\n\t[\"computer\", 50, [\n\t\t[\"*\", [\n\t\t\t\"Do computers worry you ?\",\n\t\t\t\"Why do you mention computers ?\",\n\t\t\t\"What do you think machines have to do with your problem ?\",\n\t\t\t\"Don't you think computers can help people ?\",\n\t\t\t\"What about machines worries you ?\",\n\t\t\t\"What do you think about machines ?\",\n\t\t\t\"You don't think I am a computer program, do you ?\"\n\t\t]]\n\t]],\n\t[\"am\", 0, [\n\t\t[\"* am i *\", [\n\t\t\t\"Do you believe you are (2) ?\",\n\t\t\t\"Would you want to be (2) ?\",\n\t\t\t\"Do you wish I would tell you you are (2) ?\",\n\t\t\t\"What would it mean if you were (2) ?\",\n\t\t\t\"goto what\"\n\t\t]],\n\t\t[\"* i am *\", [\n\t\t\t\"goto i\"\n\t\t]],\n\t\t[\"*\", [\n\t\t\t\"Why do you say 'am' ?\",\n\t\t\t\"I don't understand that.\"\n\t\t]]\n\t]],\n\t[\"are\", 0, [\n\t\t[\"* are you *\", [\n\t\t\t\"Why are you interested in whether I am (2) or not ?\",\n\t\t\t\"Would you prefer if I weren't (2) ?\",\n\t\t\t\"Perhaps I am (2) in your fantasies.\",\n\t\t\t\"Do you sometimes think I am (2) ?\",\n\t\t\t\"goto what\",\n\t\t\t\"Would it matter to you ?\",\n\t\t\t\"What if I were (2) ?\"\n\t\t]],\n\t\t[\"* you are *\", [\n\t\t\t\"goto you\"\n\t\t]],\n\t\t[\"* are *\", [\n\t\t\t\"Did you think they might not be (2) ?\",\n\t\t\t\"Would you like it if they were not (2) ?\",\n\t\t\t\"What if they were not (2) ?\",\n\t\t\t\"Are they always (2) ?\",\n\t\t\t\"Possibly they are (2).\",\n\t\t\t\"Are you positive they are (2) ?\"\n\t\t]]\n\t]],\n\t[\"your\", 0, [\n\t\t[\"* your *\", [\n\t\t\t\"Why are you concerned over my (2) ?\",\n\t\t\t\"What about your own (2) ?\",\n\t\t\t\"Are you worried about someone else's (2) ?\",\n\t\t\t\"Really, my (2) ?\",\n\t\t\t\"What makes you think of my (2) ?\",\n\t\t\t\"Do you want my (2) ?\"\n\t\t]]\n\t]],\n\t[\"was\", 2, [\n\t\t[\"* was i *\", [\n\t\t\t\"What if you were (2) ?\",\n\t\t\t\"Do you think you were (2) ?\",\n\t\t\t\"Were you (2) ?\",\n\t\t\t\"What would it mean if you were (2) ?\",\n\t\t\t\"What does ' (2) ' suggest to you ?\",\n\t\t\t\"goto what\"\n\t\t]],\n\t\t[\"* i was *\", [\n\t\t\t\"Were you really ?\",\n\t\t\t\"Why do you tell me you were (2) now ?\",\n\t\t\t\"Perhaps I already know you were (2).\"\n\t\t]],\n\t\t[\"* was you *\", [\n\t\t\t\"Would you like to believe I was (2) ?\",\n\t\t\t\"What suggests that I was (2) ?\",\n\t\t\t\"What do you think ?\",\n\t\t\t\"Perhaps I was (2).\",\n\t\t\t\"What if I had been (2) ?\"\n\t\t]]\n\t]],\n\t[\"i\", 0, [\n\t\t[\"* i @desire *\", [\n\t\t\t\"What would it mean to you if you got (3) ?\",\n\t\t\t\"Why do you want (3) ?\",\n\t\t\t\"Suppose you got (3) soon.\",\n\t\t\t\"What if you never got (3) ?\",\n\t\t\t\"What would getting (3) mean to you ?\",\n\t\t\t\"What does wanting (3) have to do with this discussion ?\"\n\t\t]],\n\t\t[\"* i am* @sad *\", [\n\t\t\t\"I am sorry to hear that you are (3).\",\n\t\t\t\"Do you think coming here will help you not to be (3) ?\",\n\t\t\t\"I'm sure it's not pleasant to be (3).\",\n\t\t\t\"Can you explain what made you (3) ?\"\n\t\t]],\n\t\t[\"* i am* @happy *\", [\n\t\t\t\"How have I helped you to be (3) ?\",\n\t\t\t\"Has your treatment made you (3) ?\",\n\t\t\t\"What makes you (3) just now ?\",\n\t\t\t\"Can you explain why you are suddenly (3) ?\"\n\t\t]],\n\t\t[\"* i was *\", [\n\t\t\t\"goto was\"\n\t\t]],\n\t\t[\"* i @belief i *\", [\n\t\t\t\"Do you really think so ?\",\n\t\t\t\"But you are not sure you (3).\",\n\t\t\t\"Do you really doubt you (3) ?\"\n\t\t]],\n\t\t[\"* i* @belief *you *\", [\n\t\t\t\"goto you\"\n\t\t]],\n\t\t[\"* i am *\", [\n\t\t\t\"Is it because you are (2) that you came to me ?\",\n\t\t\t\"How long have you been (2) ?\",\n\t\t\t\"Do you believe it is normal to be (2) ?\",\n\t\t\t\"Do you enjoy being (2) ?\",\n\t\t\t\"Do you know anyone else who is (2) ?\"\n\t\t]],\n\t\t[\"* i @cannot *\", [\n\t\t\t\"How do you know that you can't (3) ?\",\n\t\t\t\"Have you tried ?\",\n\t\t\t\"Perhaps you could (3) now.\",\n\t\t\t\"Do you really want to be able to (3) ?\",\n\t\t\t\"What if you could (3) ?\"\n\t\t]],\n\t\t[\"* i don't *\", [\n\t\t\t\"Don't you really (2) ?\",\n\t\t\t\"Why don't you (2) ?\",\n\t\t\t\"Do you wish to be able to (2) ?\",\n\t\t\t\"Does that trouble you ?\"\n\t\t]],\n\t\t[\"* i feel *\", [\n\t\t\t\"Tell me more about such feelings.\",\n\t\t\t\"Do you often feel (2) ?\",\n\t\t\t\"Do you enjoy feeling (2) ?\",\n\t\t\t\"Of what does feeling (2) remind you ?\"\n\t\t]],\n\t\t[\"* i * you *\", [\n\t\t\t\"Perhaps in your fantasies we (2) each other.\",\n\t\t\t\"Do you wish to (2) me ?\",\n\t\t\t\"You seem to need to (2) me.\",\n\t\t\t\"Do you (2) anyone else ?\"\n\t\t]],\n\t\t[\"*\", [\n\t\t\t\"You say (1) ?\",\n\t\t\t\"Can you elaborate on that ?\",\n\t\t\t\"Do you say (1) for some special reason ?\",\n\t\t\t\"That's quite interesting.\"\n\t\t]]\n\t]],\n\t[\"you\", 0, [\n\t\t[\"* you remind me of *\", [\n\t\t\t\"goto alike\"\n\t\t]],\n\t\t[\"* you are *\", [\n\t\t\t\"What makes you think I am (2) ?\",\n\t\t\t\"Does it please you to believe I am (2) ?\",\n\t\t\t\"Do you sometimes wish you were (2) ?\",\n\t\t\t\"Perhaps you would like to be (2).\"\n\t\t]],\n\t\t[\"* you* me *\", [\n\t\t\t\"Why do you think I (2) you ?\",\n\t\t\t\"You like to think I (2) you -- don't you ?\",\n\t\t\t\"What makes you think I (2) you ?\",\n\t\t\t\"Really, I (2) you ?\",\n\t\t\t\"Do you wish to believe I (2) you ?\",\n\t\t\t\"Suppose I did (2) you -- what would that mean ?\",\n\t\t\t\"Does someone else believe I (2) you ?\"\n\t\t]],\n\t\t[\"* you *\", [\n\t\t\t\"We were discussing you -- not me.\",\n\t\t\t\"Oh, I (2) ?\",\n\t\t\t\"You're not really talking about me -- are you ?\",\n\t\t\t\"What are your feelings now ?\"\n\t\t]]\n\t]],\n\t[\"yes\", 0, [\n\t\t[\"*\", [\n\t\t\t\"You seem to be quite positive.\",\n\t\t\t\"You are sure.\",\n\t\t\t\"I see.\",\n\t\t\t\"I understand.\"\n\t\t]]\n\t]],\n\t[\"no\", 0, [\n\t\t[\"* no one *\", [\n\t\t\t\"Are you sure, no one (2) ?\",\n\t\t\t\"Surely someone (2) .\",\n\t\t\t\"Can you think of anyone at all ?\",\n\t\t\t\"Are you thinking of a very special person ?\",\n\t\t\t\"Who, may I ask ?\",\n\t\t\t\"You have a particular person in mind, don't you ?\",\n\t\t\t\"Who do you think you are talking about ?\"\n\t\t]],\n\t\t[\"*\", [\n\t\t\t\"Are you saying no just to be negative?\",\n\t\t\t\"You are being a bit negative.\",\n\t\t\t\"Why not ?\",\n\t\t\t\"Why 'no' ?\"\n\t\t]]\n\t]],\n\t[\"my\", 2, [\n\t\t[\"$ * my *\", [\n\t\t\t\"Does that have anything to do with the fact that your (2) ?\",\n\t\t\t\"Lets discuss further why your (2).\",\n\t\t\t\"Earlier you said your (2).\",\n\t\t\t\"But your (2).\"\n\t\t]],\n\t\t[\"* my* @family *\", [\n\t\t\t\"Tell me more about your family.\",\n\t\t\t\"Who else in your family (4) ?\",\n\t\t\t\"Your (3) ?\",\n\t\t\t\"What else comes to your mind when you think of your (3) ?\"\n\t\t]],\n\t\t[\"* my *\", [\n\t\t\t\"Your (2) ?\",\n\t\t\t\"Why do you say your (2) ?\",\n\t\t\t\"Does that suggest anything else which belongs to you ?\",\n\t\t\t\"Is it important to you that your (2) ?\"\n\t\t]]\n\t]],\n\t[\"can\", 0, [\n\t\t[\"* can you *\", [\n\t\t\t\"You believe I can (2) don't you ?\",\n\t\t\t\"goto what\",\n\t\t\t\"You want me to be able to (2).\",\n\t\t\t\"Perhaps you would like to be able to (2) yourself.\"\n\t\t]],\n\t\t[\"* can i *\", [\n\t\t\t\"Whether or not you can (2) depends on you more than on me.\",\n\t\t\t\"Do you want to be able to (2) ?\",\n\t\t\t\"Perhaps you don't want to (2).\",\n\t\t\t\"goto what\"\n\t\t]]\n\t]],\n\t[\"what\", 0, [\n\t\t[\"*\", [\n\t\t\t\"Why do you ask ?\",\n\t\t\t\"Does that question interest you ?\",\n\t\t\t\"What is it you really want to know ?\",\n\t\t\t\"Are such questions much on your mind ?\",\n\t\t\t\"What answer would please you most ?\",\n\t\t\t\"What do you think ?\",\n\t\t\t\"What comes to mind when you ask that ?\",\n\t\t\t\"Have you asked such questions before ?\",\n\t\t\t\"Have you asked anyone else ?\"\n\t\t]]\n\t]],\n\t[\"who\", 0, [\n\t\t[\"who *\", [\n\t\t\t\"goto what\"\n\t\t]]\n\t]],\n\t[\"when\", 0, [\n\t\t[\"when *\", [\n\t\t\t\"goto what\"\n\t\t]]\n\t]],\n\t[\"where\", 0, [\n\t\t[\"where *\", [\n\t\t\t\"goto what\"\n\t\t]]\n\t]],\n\t[\"how\", 0, [\n\t\t[\"how *\", [\n\t\t\t\"goto what\"\n\t\t]]\n\t]],\n\t[\"because\", 0, [\n\t\t[\"*\", [\n\t\t\t\"Is that the real reason ?\",\n\t\t\t\"Don't any other reasons come to mind ?\",\n\t\t\t\"Does that reason seem to explain anything else ?\",\n\t\t\t\"What other reasons might there be ?\"\n\t\t]]\n\t]],\n\t[\"why\", 0, [\n\t\t[\"* why don't you *\", [\n\t\t\t\"Do you believe I don't (2) ?\",\n\t\t\t\"Perhaps I will (2) in good time.\",\n\t\t\t\"Should you (2) yourself ?\",\n\t\t\t\"You want me to (2) ?\",\n\t\t\t\"goto what\"\n\t\t]],\n\t\t[\"* why can't i *\", [\n\t\t\t\"Do you think you should be able to (2) ?\",\n\t\t\t\"Do you want to be able to (2) ?\",\n\t\t\t\"Do you believe this will help you to (2) ?\",\n\t\t\t\"Have you any idea why you can't (2) ?\",\n\t\t\t\"goto what\"\n\t\t]],\n\t\t[\"*\", [\n\t\t\t\"goto what\"\n\t\t]]\n\t]],\n\t[\"everyone\", 2, [\n\t\t[\"* @everyone *\", [\n\t\t\t\"Really, (2) ?\",\n\t\t\t\"Surely not (2).\",\n\t\t\t\"Can you think of anyone in particular ?\",\n\t\t\t\"Who, for example?\",\n\t\t\t\"Are you thinking of a very special person ?\",\n\t\t\t\"Who, may I ask ?\",\n\t\t\t\"Someone special perhaps ?\",\n\t\t\t\"You have a particular person in mind, don't you ?\",\n\t\t\t\"Who do you think you're talking about ?\"\n\t\t]]\n\t]],\n\t[\"everybody\", 2, [\n\t\t[\"*\", [\n\t\t\t\"goto everyone\"\n\t\t]]\n\t]],\n\t[\"nobody\", 2, [\n\t\t[\"*\", [\n\t\t\t\"goto everyone\"\n\t\t]]\n\t]],\n\t[\"noone\", 2, [\n\t\t[\"*\", [\n\t\t\t\"goto everyone\"\n\t\t]]\n\t]],\n\t[\"always\", 1, [\n\t\t[\"*\", [\n\t\t\t\"Can you think of a specific example ?\",\n\t\t\t\"When ?\",\n\t\t\t\"What incident are you thinking of ?\",\n\t\t\t\"Really, always ?\"\n\t\t]]\n\t]],\n\t[\"alike\", 10, [\n\t\t[\"*\", [\n\t\t\t\"In what way ?\",\n\t\t\t\"What resemblence do you see ?\",\n\t\t\t\"What does that similarity suggest to you ?\",\n\t\t\t\"What other connections do you see ?\",\n\t\t\t\"What do you suppose that resemblence means ?\",\n\t\t\t\"What is the connection, do you suppose ?\",\n\t\t\t\"Could there really be some connection ?\",\n\t\t\t\"How ?\"\n\t\t]]\n\t]],\n\t[\"like\", 10, [\n\t\t[\"* @be *like *\", [\n\t\t\t\"goto alike\"\n\t\t]]\n\t]],\n\t[\"different\", 0, [\n\t\t[\"*\", [\n\t\t\t\"How is it different ?\",\n\t\t\t\"What differences do you see ?\",\n\t\t\t\"What does that difference suggest to you ?\",\n\t\t\t\"What other distinctions do you see ?\",\n\t\t\t\"What do you suppose that disparity means ?\",\n\t\t\t\"Could there be some connection, do you suppose ?\",\n\t\t\t\"How ?\"\n\t\t]]\n\t]]\n\n];\n\n// regexp/replacement pairs to be performed as final cleanings\n// here: cleanings for multiple bots talking to each other\nexports.elizaPostTransforms = [\n\t/ old old/g, \" old\",\n\t/\\bthey were( not)? me\\b/g, \"it was$1 me\",\n\t/\\bthey are( not)? me\\b/g, \"it is$1 me\",\n\t/Are they( always)? me\\b/, \"it is$1 me\",\n\t/\\bthat your( own)? (\\w+)( now)? \\?/, \"that you have your$1 $2 ?\",\n\t/\\bI to have (\\w+)/, \"I have $1\",\n\t/Earlier you said your( own)? (\\w+)( now)?\\./, \"Earlier you talked about your $2.\"\n];\n\n// eof"};
jalangiLabel58:
    while (true) {
        try {
            J$.Se(6561, '/home/blake/ExpoSE/Targets/elizabot/node_modules/elizabot/elizadata_jalangi_.js', '/home/blake/ExpoSE/Targets/elizabot/node_modules/elizabot/elizadata.js');
            J$.N(6569, '__secret__S$', __secret__S$, 0);
            J$.N(6577, '__secret__traits__', __secret__traits__, 0);
            var __secret__S$ = J$.X1(41, J$.W(33, '__secret__S$', J$.F(25, J$.R(9, 'require', require, 2), 0)(J$.T(17, 'S$', 21, false)), __secret__S$, 3));
            var __secret__traits__ = J$.X1(73, J$.W(65, '__secret__traits__', J$.G(57, J$.R(49, '__secret__S$', __secret__S$, 1), 'Traits', 0), __secret__traits__, 3));
            J$.X1(129, J$.P(121, J$.R(81, 'exports', exports, 2), 'elizaInitials', J$.T(113, [
                J$.T(89, 'How do you do.  Please tell me your problem.', 21, false),
                J$.T(97, 'Please tell me what\'s been bothering you.', 21, false),
                J$.T(105, 'Is something troubling you ?', 21, false)
            ], 10, false), 0));
            J$.X1(201, J$.P(193, J$.R(137, 'exports', exports, 2), 'elizaFinals', J$.T(185, [
                J$.T(145, 'Goodbye.  It was nice talking to you.', 21, false),
                J$.T(153, 'Goodbye.  This was really a nice talk.', 21, false),
                J$.T(161, 'Goodbye.  I\'m looking forward to our next session.', 21, false),
                J$.T(169, 'This was a good session, wasn\'t it -- but time is over now.   Goodbye.', 21, false),
                J$.T(177, 'Maybe we could discuss this moreover in our next session ?   Goodbye.', 21, false)
            ], 10, false), 0));
            J$.X1(273, J$.P(265, J$.R(209, 'exports', exports, 2), 'elizaQuits', J$.T(257, [
                J$.T(217, 'bye', 21, false),
                J$.T(225, 'goodbye', 21, false),
                J$.T(233, 'done', 21, false),
                J$.T(241, 'exit', 21, false),
                J$.T(249, 'quit', 21, false)
            ], 10, false), 0));
            J$.X1(593, J$.P(585, J$.R(281, 'exports', exports, 2), 'elizaPres', J$.T(577, [
                J$.T(289, 'dont', 21, false),
                J$.T(297, 'don\'t', 21, false),
                J$.T(305, 'cant', 21, false),
                J$.T(313, 'can\'t', 21, false),
                J$.T(321, 'wont', 21, false),
                J$.T(329, 'won\'t', 21, false),
                J$.T(337, 'recollect', 21, false),
                J$.T(345, 'remember', 21, false),
                J$.T(353, 'recall', 21, false),
                J$.T(361, 'remember', 21, false),
                J$.T(369, 'dreamt', 21, false),
                J$.T(377, 'dreamed', 21, false),
                J$.T(385, 'dreams', 21, false),
                J$.T(393, 'dream', 21, false),
                J$.T(401, 'maybe', 21, false),
                J$.T(409, 'perhaps', 21, false),
                J$.T(417, 'certainly', 21, false),
                J$.T(425, 'yes', 21, false),
                J$.T(433, 'machine', 21, false),
                J$.T(441, 'computer', 21, false),
                J$.T(449, 'machines', 21, false),
                J$.T(457, 'computer', 21, false),
                J$.T(465, 'computers', 21, false),
                J$.T(473, 'computer', 21, false),
                J$.T(481, 'were', 21, false),
                J$.T(489, 'was', 21, false),
                J$.T(497, 'you\'re', 21, false),
                J$.T(505, 'you are', 21, false),
                J$.T(513, 'i\'m', 21, false),
                J$.T(521, 'i am', 21, false),
                J$.T(529, 'same', 21, false),
                J$.T(537, 'alike', 21, false),
                J$.T(545, 'identical', 21, false),
                J$.T(553, 'alike', 21, false),
                J$.T(561, 'equivalent', 21, false),
                J$.T(569, 'alike', 21, false)
            ], 10, false), 0));
            J$.X1(769, J$.P(761, J$.R(601, 'exports', exports, 2), 'elizaPosts', J$.T(753, [
                J$.T(609, 'am', 21, false),
                J$.T(617, 'are', 21, false),
                J$.T(625, 'your', 21, false),
                J$.T(633, 'my', 21, false),
                J$.T(641, 'me', 21, false),
                J$.T(649, 'you', 21, false),
                J$.T(657, 'myself', 21, false),
                J$.T(665, 'yourself', 21, false),
                J$.T(673, 'yourself', 21, false),
                J$.T(681, 'myself', 21, false),
                J$.T(689, 'i', 21, false),
                J$.T(697, 'you', 21, false),
                J$.T(705, 'you', 21, false),
                J$.T(713, 'I', 21, false),
                J$.T(721, 'my', 21, false),
                J$.T(729, 'your', 21, false),
                J$.T(737, 'i\'m', 21, false),
                J$.T(745, 'you are', 21, false)
            ], 10, false), 0));
            J$.X1(1097, J$.P(1089, J$.R(777, 'exports', exports, 2), 'elizaSynons', J$.T(1081, {
                'be': J$.T(817, [
                    J$.T(785, 'am', 21, false),
                    J$.T(793, 'is', 21, false),
                    J$.T(801, 'are', 21, false),
                    J$.T(809, 'was', 21, false)
                ], 10, false),
                'belief': J$.T(857, [
                    J$.T(825, 'feel', 21, false),
                    J$.T(833, 'think', 21, false),
                    J$.T(841, 'believe', 21, false),
                    J$.T(849, 'wish', 21, false)
                ], 10, false),
                'cannot': J$.T(873, [J$.T(865, 'can\'t', 21, false)], 10, false),
                'desire': J$.T(897, [
                    J$.T(881, 'want', 21, false),
                    J$.T(889, 'need', 21, false)
                ], 10, false),
                'everyone': J$.T(929, [
                    J$.T(905, 'everybody', 21, false),
                    J$.T(913, 'nobody', 21, false),
                    J$.T(921, 'noone', 21, false)
                ], 10, false),
                'family': J$.T(1009, [
                    J$.T(937, 'mother', 21, false),
                    J$.T(945, 'mom', 21, false),
                    J$.T(953, 'father', 21, false),
                    J$.T(961, 'dad', 21, false),
                    J$.T(969, 'sister', 21, false),
                    J$.T(977, 'brother', 21, false),
                    J$.T(985, 'wife', 21, false),
                    J$.T(993, 'children', 21, false),
                    J$.T(1001, 'child', 21, false)
                ], 10, false),
                'happy': J$.T(1041, [
                    J$.T(1017, 'elated', 21, false),
                    J$.T(1025, 'glad', 21, false),
                    J$.T(1033, 'better', 21, false)
                ], 10, false),
                'sad': J$.T(1073, [
                    J$.T(1049, 'unhappy', 21, false),
                    J$.T(1057, 'depressed', 21, false),
                    J$.T(1065, 'sick', 21, false)
                ], 10, false)
            }, 11, false), 0));
            J$.X1(6409, J$.P(6401, J$.R(1105, 'exports', exports, 2), 'elizaKeywords', J$.T(6393, [
                J$.T(1217, [
                    J$.T(1113, 'xnone', 21, false),
                    J$.T(1121, 0, 22, false),
                    J$.T(1209, [J$.T(1201, [
                            J$.T(1129, '*', 21, false),
                            J$.T(1193, [
                                J$.T(1137, 'I\'m not sure I understand you fully.', 21, false),
                                J$.T(1145, 'Please go on.', 21, false),
                                J$.T(1153, 'What does that suggest to you ?', 21, false),
                                J$.T(1161, 'Do you feel strongly about discussing such things ?', 21, false),
                                J$.T(1169, 'That is interesting.  Please continue.', 21, false),
                                J$.T(1177, 'Tell me more about that.', 21, false),
                                J$.T(1185, 'Does talking about this bother you ?', 21, false)
                            ], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false),
                J$.T(1305, [
                    J$.T(1225, 'sorry', 21, false),
                    J$.T(1233, 0, 22, false),
                    J$.T(1297, [J$.T(1289, [
                            J$.T(1241, '*', 21, false),
                            J$.T(1281, [
                                J$.T(1249, 'Please don\'t apologise.', 21, false),
                                J$.T(1257, 'Apologies are not necessary.', 21, false),
                                J$.T(1265, 'I\'ve told you that apologies are not required.', 21, false),
                                J$.T(1273, 'It did not bother me.  Please continue.', 21, false)
                            ], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false),
                J$.T(1369, [
                    J$.T(1313, 'apologise', 21, false),
                    J$.T(1321, 0, 22, false),
                    J$.T(1361, [J$.T(1353, [
                            J$.T(1329, '*', 21, false),
                            J$.T(1345, [J$.T(1337, 'goto sorry', 21, false)], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false),
                J$.T(1593, [
                    J$.T(1377, 'remember', 21, false),
                    J$.T(1385, 5, 22, false),
                    J$.T(1585, [
                        J$.T(1465, [
                            J$.T(1393, '* i remember *', 21, false),
                            J$.T(1457, [
                                J$.T(1401, 'Do you often think of (2) ?', 21, false),
                                J$.T(1409, 'Does thinking of (2) bring anything else to mind ?', 21, false),
                                J$.T(1417, 'What else do you recollect ?', 21, false),
                                J$.T(1425, 'Why do you remember (2) just now ?', 21, false),
                                J$.T(1433, 'What in the present situation reminds you of (2) ?', 21, false),
                                J$.T(1441, 'What is the connection between me and (2) ?', 21, false),
                                J$.T(1449, 'What else does (2) remind you of ?', 21, false)
                            ], 10, false)
                        ], 10, false),
                        J$.T(1529, [
                            J$.T(1473, '* do you remember *', 21, false),
                            J$.T(1521, [
                                J$.T(1481, 'Did you think I would forget (2) ?', 21, false),
                                J$.T(1489, 'Why do you think I should recall (2) now ?', 21, false),
                                J$.T(1497, 'What about (2) ?', 21, false),
                                J$.T(1505, 'goto what', 21, false),
                                J$.T(1513, 'You mentioned (2) ?', 21, false)
                            ], 10, false)
                        ], 10, false),
                        J$.T(1577, [
                            J$.T(1537, '* you remember *', 21, false),
                            J$.T(1569, [
                                J$.T(1545, 'How could I forget (2) ?', 21, false),
                                J$.T(1553, 'What about (2) should I remember ?', 21, false),
                                J$.T(1561, 'goto you', 21, false)
                            ], 10, false)
                        ], 10, false)
                    ], 10, false)
                ], 10, false),
                J$.T(1777, [
                    J$.T(1601, 'forget', 21, false),
                    J$.T(1609, 5, 22, false),
                    J$.T(1769, [
                        J$.T(1689, [
                            J$.T(1617, '* i forget *', 21, false),
                            J$.T(1681, [
                                J$.T(1625, 'Can you think of why you might forget (2) ?', 21, false),
                                J$.T(1633, 'Why can\'t you remember (2) ?', 21, false),
                                J$.T(1641, 'How often do you think of (2) ?', 21, false),
                                J$.T(1649, 'Does it bother you to forget that ?', 21, false),
                                J$.T(1657, 'Could it be a mental block ?', 21, false),
                                J$.T(1665, 'Are you generally forgetful ?', 21, false),
                                J$.T(1673, 'Do you think you are suppressing (2) ?', 21, false)
                            ], 10, false)
                        ], 10, false),
                        J$.T(1761, [
                            J$.T(1697, '* did you forget *', 21, false),
                            J$.T(1753, [
                                J$.T(1705, 'Why do you ask ?', 21, false),
                                J$.T(1713, 'Are you sure you told me ?', 21, false),
                                J$.T(1721, 'Would it bother you if I forgot (2) ?', 21, false),
                                J$.T(1729, 'Why should I recall (2) just now ?', 21, false),
                                J$.T(1737, 'goto what', 21, false),
                                J$.T(1745, 'Tell me more about (2).', 21, false)
                            ], 10, false)
                        ], 10, false)
                    ], 10, false)
                ], 10, false),
                J$.T(1889, [
                    J$.T(1785, 'if', 21, false),
                    J$.T(1793, 3, 22, false),
                    J$.T(1881, [J$.T(1873, [
                            J$.T(1801, '* if *', 21, false),
                            J$.T(1865, [
                                J$.T(1809, 'Do you think it\'s likely that (2) ?', 21, false),
                                J$.T(1817, 'Do you wish that (2) ?', 21, false),
                                J$.T(1825, 'What do you know about (2) ?', 21, false),
                                J$.T(1833, 'Really, if (2) ?', 21, false),
                                J$.T(1841, 'What would you do if (2) ?', 21, false),
                                J$.T(1849, 'But what are the chances that (2) ?', 21, false),
                                J$.T(1857, 'What does this speculation lead to ?', 21, false)
                            ], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false),
                J$.T(1977, [
                    J$.T(1897, 'dreamed', 21, false),
                    J$.T(1905, 4, 22, false),
                    J$.T(1969, [J$.T(1961, [
                            J$.T(1913, '* i dreamed *', 21, false),
                            J$.T(1953, [
                                J$.T(1921, 'Really, (2) ?', 21, false),
                                J$.T(1929, 'Have you ever fantasized (2) while you were awake ?', 21, false),
                                J$.T(1937, 'Have you ever dreamed (2) before ?', 21, false),
                                J$.T(1945, 'goto dream', 21, false)
                            ], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false),
                J$.T(2065, [
                    J$.T(1985, 'dream', 21, false),
                    J$.T(1993, 3, 22, false),
                    J$.T(2057, [J$.T(2049, [
                            J$.T(2001, '*', 21, false),
                            J$.T(2041, [
                                J$.T(2009, 'What does that dream suggest to you ?', 21, false),
                                J$.T(2017, 'Do you dream often ?', 21, false),
                                J$.T(2025, 'What persons appear in your dreams ?', 21, false),
                                J$.T(2033, 'Do you believe that dreams have something to do with your problem ?', 21, false)
                            ], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false),
                J$.T(2169, [
                    J$.T(2073, 'perhaps', 21, false),
                    J$.T(2081, 0, 22, false),
                    J$.T(2161, [J$.T(2153, [
                            J$.T(2089, '*', 21, false),
                            J$.T(2145, [
                                J$.T(2097, 'You don\'t seem quite certain.', 21, false),
                                J$.T(2105, 'Why the uncertain tone ?', 21, false),
                                J$.T(2113, 'Can\'t you be more positive ?', 21, false),
                                J$.T(2121, 'You aren\'t sure ?', 21, false),
                                J$.T(2129, 'Don\'t you know ?', 21, false),
                                J$.T(2137, 'How likely, would you estimate ?', 21, false)
                            ], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false),
                J$.T(2241, [
                    J$.T(2177, 'name', 21, false),
                    J$.T(2185, 15, 22, false),
                    J$.T(2233, [J$.T(2225, [
                            J$.T(2193, '*', 21, false),
                            J$.T(2217, [
                                J$.T(2201, 'I am not interested in names.', 21, false),
                                J$.T(2209, 'I\'ve told you before, I don\'t care about names -- please continue.', 21, false)
                            ], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false),
                J$.T(2313, [
                    J$.T(2249, 'deutsch', 21, false),
                    J$.T(2257, 0, 22, false),
                    J$.T(2305, [J$.T(2297, [
                            J$.T(2265, '*', 21, false),
                            J$.T(2289, [
                                J$.T(2273, 'goto xforeign', 21, false),
                                J$.T(2281, 'I told you before, I don\'t understand German.', 21, false)
                            ], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false),
                J$.T(2385, [
                    J$.T(2321, 'francais', 21, false),
                    J$.T(2329, 0, 22, false),
                    J$.T(2377, [J$.T(2369, [
                            J$.T(2337, '*', 21, false),
                            J$.T(2361, [
                                J$.T(2345, 'goto xforeign', 21, false),
                                J$.T(2353, 'I told you before, I don\'t understand French.', 21, false)
                            ], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false),
                J$.T(2457, [
                    J$.T(2393, 'italiano', 21, false),
                    J$.T(2401, 0, 22, false),
                    J$.T(2449, [J$.T(2441, [
                            J$.T(2409, '*', 21, false),
                            J$.T(2433, [
                                J$.T(2417, 'goto xforeign', 21, false),
                                J$.T(2425, 'I told you before, I don\'t understand Italian.', 21, false)
                            ], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false),
                J$.T(2529, [
                    J$.T(2465, 'espanol', 21, false),
                    J$.T(2473, 0, 22, false),
                    J$.T(2521, [J$.T(2513, [
                            J$.T(2481, '*', 21, false),
                            J$.T(2505, [
                                J$.T(2489, 'goto xforeign', 21, false),
                                J$.T(2497, 'I told you before, I don\'t understand Spanish.', 21, false)
                            ], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false),
                J$.T(2593, [
                    J$.T(2537, 'xforeign', 21, false),
                    J$.T(2545, 0, 22, false),
                    J$.T(2585, [J$.T(2577, [
                            J$.T(2553, '*', 21, false),
                            J$.T(2569, [J$.T(2561, 'I speak only English.', 21, false)], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false),
                J$.T(2665, [
                    J$.T(2601, 'hello', 21, false),
                    J$.T(2609, 0, 22, false),
                    J$.T(2657, [J$.T(2649, [
                            J$.T(2617, '*', 21, false),
                            J$.T(2641, [
                                J$.T(2625, 'How do you do.  Please state your problem.', 21, false),
                                J$.T(2633, 'Hi.  What seems to be your problem ?', 21, false)
                            ], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false),
                J$.T(2777, [
                    J$.T(2673, 'computer', 21, false),
                    J$.T(2681, 50, 22, false),
                    J$.T(2769, [J$.T(2761, [
                            J$.T(2689, '*', 21, false),
                            J$.T(2753, [
                                J$.T(2697, 'Do computers worry you ?', 21, false),
                                J$.T(2705, 'Why do you mention computers ?', 21, false),
                                J$.T(2713, 'What do you think machines have to do with your problem ?', 21, false),
                                J$.T(2721, 'Don\'t you think computers can help people ?', 21, false),
                                J$.T(2729, 'What about machines worries you ?', 21, false),
                                J$.T(2737, 'What do you think about machines ?', 21, false),
                                J$.T(2745, 'You don\'t think I am a computer program, do you ?', 21, false)
                            ], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false),
                J$.T(2945, [
                    J$.T(2785, 'am', 21, false),
                    J$.T(2793, 0, 22, false),
                    J$.T(2937, [
                        J$.T(2857, [
                            J$.T(2801, '* am i *', 21, false),
                            J$.T(2849, [
                                J$.T(2809, 'Do you believe you are (2) ?', 21, false),
                                J$.T(2817, 'Would you want to be (2) ?', 21, false),
                                J$.T(2825, 'Do you wish I would tell you you are (2) ?', 21, false),
                                J$.T(2833, 'What would it mean if you were (2) ?', 21, false),
                                J$.T(2841, 'goto what', 21, false)
                            ], 10, false)
                        ], 10, false),
                        J$.T(2889, [
                            J$.T(2865, '* i am *', 21, false),
                            J$.T(2881, [J$.T(2873, 'goto i', 21, false)], 10, false)
                        ], 10, false),
                        J$.T(2929, [
                            J$.T(2897, '*', 21, false),
                            J$.T(2921, [
                                J$.T(2905, 'Why do you say \'am\' ?', 21, false),
                                J$.T(2913, 'I don\'t understand that.', 21, false)
                            ], 10, false)
                        ], 10, false)
                    ], 10, false)
                ], 10, false),
                J$.T(3161, [
                    J$.T(2953, 'are', 21, false),
                    J$.T(2961, 0, 22, false),
                    J$.T(3153, [
                        J$.T(3041, [
                            J$.T(2969, '* are you *', 21, false),
                            J$.T(3033, [
                                J$.T(2977, 'Why are you interested in whether I am (2) or not ?', 21, false),
                                J$.T(2985, 'Would you prefer if I weren\'t (2) ?', 21, false),
                                J$.T(2993, 'Perhaps I am (2) in your fantasies.', 21, false),
                                J$.T(3001, 'Do you sometimes think I am (2) ?', 21, false),
                                J$.T(3009, 'goto what', 21, false),
                                J$.T(3017, 'Would it matter to you ?', 21, false),
                                J$.T(3025, 'What if I were (2) ?', 21, false)
                            ], 10, false)
                        ], 10, false),
                        J$.T(3073, [
                            J$.T(3049, '* you are *', 21, false),
                            J$.T(3065, [J$.T(3057, 'goto you', 21, false)], 10, false)
                        ], 10, false),
                        J$.T(3145, [
                            J$.T(3081, '* are *', 21, false),
                            J$.T(3137, [
                                J$.T(3089, 'Did you think they might not be (2) ?', 21, false),
                                J$.T(3097, 'Would you like it if they were not (2) ?', 21, false),
                                J$.T(3105, 'What if they were not (2) ?', 21, false),
                                J$.T(3113, 'Are they always (2) ?', 21, false),
                                J$.T(3121, 'Possibly they are (2).', 21, false),
                                J$.T(3129, 'Are you positive they are (2) ?', 21, false)
                            ], 10, false)
                        ], 10, false)
                    ], 10, false)
                ], 10, false),
                J$.T(3265, [
                    J$.T(3169, 'your', 21, false),
                    J$.T(3177, 0, 22, false),
                    J$.T(3257, [J$.T(3249, [
                            J$.T(3185, '* your *', 21, false),
                            J$.T(3241, [
                                J$.T(3193, 'Why are you concerned over my (2) ?', 21, false),
                                J$.T(3201, 'What about your own (2) ?', 21, false),
                                J$.T(3209, 'Are you worried about someone else\'s (2) ?', 21, false),
                                J$.T(3217, 'Really, my (2) ?', 21, false),
                                J$.T(3225, 'What makes you think of my (2) ?', 21, false),
                                J$.T(3233, 'Do you want my (2) ?', 21, false)
                            ], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false),
                J$.T(3481, [
                    J$.T(3273, 'was', 21, false),
                    J$.T(3281, 2, 22, false),
                    J$.T(3473, [
                        J$.T(3353, [
                            J$.T(3289, '* was i *', 21, false),
                            J$.T(3345, [
                                J$.T(3297, 'What if you were (2) ?', 21, false),
                                J$.T(3305, 'Do you think you were (2) ?', 21, false),
                                J$.T(3313, 'Were you (2) ?', 21, false),
                                J$.T(3321, 'What would it mean if you were (2) ?', 21, false),
                                J$.T(3329, 'What does \' (2) \' suggest to you ?', 21, false),
                                J$.T(3337, 'goto what', 21, false)
                            ], 10, false)
                        ], 10, false),
                        J$.T(3401, [
                            J$.T(3361, '* i was *', 21, false),
                            J$.T(3393, [
                                J$.T(3369, 'Were you really ?', 21, false),
                                J$.T(3377, 'Why do you tell me you were (2) now ?', 21, false),
                                J$.T(3385, 'Perhaps I already know you were (2).', 21, false)
                            ], 10, false)
                        ], 10, false),
                        J$.T(3465, [
                            J$.T(3409, '* was you *', 21, false),
                            J$.T(3457, [
                                J$.T(3417, 'Would you like to believe I was (2) ?', 21, false),
                                J$.T(3425, 'What suggests that I was (2) ?', 21, false),
                                J$.T(3433, 'What do you think ?', 21, false),
                                J$.T(3441, 'Perhaps I was (2).', 21, false),
                                J$.T(3449, 'What if I had been (2) ?', 21, false)
                            ], 10, false)
                        ], 10, false)
                    ], 10, false)
                ], 10, false),
                J$.T(4161, [
                    J$.T(3489, 'i', 21, false),
                    J$.T(3497, 0, 22, false),
                    J$.T(4153, [
                        J$.T(3569, [
                            J$.T(3505, '* i @desire *', 21, false),
                            J$.T(3561, [
                                J$.T(3513, 'What would it mean to you if you got (3) ?', 21, false),
                                J$.T(3521, 'Why do you want (3) ?', 21, false),
                                J$.T(3529, 'Suppose you got (3) soon.', 21, false),
                                J$.T(3537, 'What if you never got (3) ?', 21, false),
                                J$.T(3545, 'What would getting (3) mean to you ?', 21, false),
                                J$.T(3553, 'What does wanting (3) have to do with this discussion ?', 21, false)
                            ], 10, false)
                        ], 10, false),
                        J$.T(3625, [
                            J$.T(3577, '* i am* @sad *', 21, false),
                            J$.T(3617, [
                                J$.T(3585, 'I am sorry to hear that you are (3).', 21, false),
                                J$.T(3593, 'Do you think coming here will help you not to be (3) ?', 21, false),
                                J$.T(3601, 'I\'m sure it\'s not pleasant to be (3).', 21, false),
                                J$.T(3609, 'Can you explain what made you (3) ?', 21, false)
                            ], 10, false)
                        ], 10, false),
                        J$.T(3681, [
                            J$.T(3633, '* i am* @happy *', 21, false),
                            J$.T(3673, [
                                J$.T(3641, 'How have I helped you to be (3) ?', 21, false),
                                J$.T(3649, 'Has your treatment made you (3) ?', 21, false),
                                J$.T(3657, 'What makes you (3) just now ?', 21, false),
                                J$.T(3665, 'Can you explain why you are suddenly (3) ?', 21, false)
                            ], 10, false)
                        ], 10, false),
                        J$.T(3713, [
                            J$.T(3689, '* i was *', 21, false),
                            J$.T(3705, [J$.T(3697, 'goto was', 21, false)], 10, false)
                        ], 10, false),
                        J$.T(3761, [
                            J$.T(3721, '* i @belief i *', 21, false),
                            J$.T(3753, [
                                J$.T(3729, 'Do you really think so ?', 21, false),
                                J$.T(3737, 'But you are not sure you (3).', 21, false),
                                J$.T(3745, 'Do you really doubt you (3) ?', 21, false)
                            ], 10, false)
                        ], 10, false),
                        J$.T(3793, [
                            J$.T(3769, '* i* @belief *you *', 21, false),
                            J$.T(3785, [J$.T(3777, 'goto you', 21, false)], 10, false)
                        ], 10, false),
                        J$.T(3857, [
                            J$.T(3801, '* i am *', 21, false),
                            J$.T(3849, [
                                J$.T(3809, 'Is it because you are (2) that you came to me ?', 21, false),
                                J$.T(3817, 'How long have you been (2) ?', 21, false),
                                J$.T(3825, 'Do you believe it is normal to be (2) ?', 21, false),
                                J$.T(3833, 'Do you enjoy being (2) ?', 21, false),
                                J$.T(3841, 'Do you know anyone else who is (2) ?', 21, false)
                            ], 10, false)
                        ], 10, false),
                        J$.T(3921, [
                            J$.T(3865, '* i @cannot *', 21, false),
                            J$.T(3913, [
                                J$.T(3873, 'How do you know that you can\'t (3) ?', 21, false),
                                J$.T(3881, 'Have you tried ?', 21, false),
                                J$.T(3889, 'Perhaps you could (3) now.', 21, false),
                                J$.T(3897, 'Do you really want to be able to (3) ?', 21, false),
                                J$.T(3905, 'What if you could (3) ?', 21, false)
                            ], 10, false)
                        ], 10, false),
                        J$.T(3977, [
                            J$.T(3929, '* i don\'t *', 21, false),
                            J$.T(3969, [
                                J$.T(3937, 'Don\'t you really (2) ?', 21, false),
                                J$.T(3945, 'Why don\'t you (2) ?', 21, false),
                                J$.T(3953, 'Do you wish to be able to (2) ?', 21, false),
                                J$.T(3961, 'Does that trouble you ?', 21, false)
                            ], 10, false)
                        ], 10, false),
                        J$.T(4033, [
                            J$.T(3985, '* i feel *', 21, false),
                            J$.T(4025, [
                                J$.T(3993, 'Tell me more about such feelings.', 21, false),
                                J$.T(4001, 'Do you often feel (2) ?', 21, false),
                                J$.T(4009, 'Do you enjoy feeling (2) ?', 21, false),
                                J$.T(4017, 'Of what does feeling (2) remind you ?', 21, false)
                            ], 10, false)
                        ], 10, false),
                        J$.T(4089, [
                            J$.T(4041, '* i * you *', 21, false),
                            J$.T(4081, [
                                J$.T(4049, 'Perhaps in your fantasies we (2) each other.', 21, false),
                                J$.T(4057, 'Do you wish to (2) me ?', 21, false),
                                J$.T(4065, 'You seem to need to (2) me.', 21, false),
                                J$.T(4073, 'Do you (2) anyone else ?', 21, false)
                            ], 10, false)
                        ], 10, false),
                        J$.T(4145, [
                            J$.T(4097, '*', 21, false),
                            J$.T(4137, [
                                J$.T(4105, 'You say (1) ?', 21, false),
                                J$.T(4113, 'Can you elaborate on that ?', 21, false),
                                J$.T(4121, 'Do you say (1) for some special reason ?', 21, false),
                                J$.T(4129, 'That\'s quite interesting.', 21, false)
                            ], 10, false)
                        ], 10, false)
                    ], 10, false)
                ], 10, false),
                J$.T(4417, [
                    J$.T(4169, 'you', 21, false),
                    J$.T(4177, 0, 22, false),
                    J$.T(4409, [
                        J$.T(4209, [
                            J$.T(4185, '* you remind me of *', 21, false),
                            J$.T(4201, [J$.T(4193, 'goto alike', 21, false)], 10, false)
                        ], 10, false),
                        J$.T(4265, [
                            J$.T(4217, '* you are *', 21, false),
                            J$.T(4257, [
                                J$.T(4225, 'What makes you think I am (2) ?', 21, false),
                                J$.T(4233, 'Does it please you to believe I am (2) ?', 21, false),
                                J$.T(4241, 'Do you sometimes wish you were (2) ?', 21, false),
                                J$.T(4249, 'Perhaps you would like to be (2).', 21, false)
                            ], 10, false)
                        ], 10, false),
                        J$.T(4345, [
                            J$.T(4273, '* you* me *', 21, false),
                            J$.T(4337, [
                                J$.T(4281, 'Why do you think I (2) you ?', 21, false),
                                J$.T(4289, 'You like to think I (2) you -- don\'t you ?', 21, false),
                                J$.T(4297, 'What makes you think I (2) you ?', 21, false),
                                J$.T(4305, 'Really, I (2) you ?', 21, false),
                                J$.T(4313, 'Do you wish to believe I (2) you ?', 21, false),
                                J$.T(4321, 'Suppose I did (2) you -- what would that mean ?', 21, false),
                                J$.T(4329, 'Does someone else believe I (2) you ?', 21, false)
                            ], 10, false)
                        ], 10, false),
                        J$.T(4401, [
                            J$.T(4353, '* you *', 21, false),
                            J$.T(4393, [
                                J$.T(4361, 'We were discussing you -- not me.', 21, false),
                                J$.T(4369, 'Oh, I (2) ?', 21, false),
                                J$.T(4377, 'You\'re not really talking about me -- are you ?', 21, false),
                                J$.T(4385, 'What are your feelings now ?', 21, false)
                            ], 10, false)
                        ], 10, false)
                    ], 10, false)
                ], 10, false),
                J$.T(4505, [
                    J$.T(4425, 'yes', 21, false),
                    J$.T(4433, 0, 22, false),
                    J$.T(4497, [J$.T(4489, [
                            J$.T(4441, '*', 21, false),
                            J$.T(4481, [
                                J$.T(4449, 'You seem to be quite positive.', 21, false),
                                J$.T(4457, 'You are sure.', 21, false),
                                J$.T(4465, 'I see.', 21, false),
                                J$.T(4473, 'I understand.', 21, false)
                            ], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false),
                J$.T(4673, [
                    J$.T(4513, 'no', 21, false),
                    J$.T(4521, 0, 22, false),
                    J$.T(4665, [
                        J$.T(4601, [
                            J$.T(4529, '* no one *', 21, false),
                            J$.T(4593, [
                                J$.T(4537, 'Are you sure, no one (2) ?', 21, false),
                                J$.T(4545, 'Surely someone (2) .', 21, false),
                                J$.T(4553, 'Can you think of anyone at all ?', 21, false),
                                J$.T(4561, 'Are you thinking of a very special person ?', 21, false),
                                J$.T(4569, 'Who, may I ask ?', 21, false),
                                J$.T(4577, 'You have a particular person in mind, don\'t you ?', 21, false),
                                J$.T(4585, 'Who do you think you are talking about ?', 21, false)
                            ], 10, false)
                        ], 10, false),
                        J$.T(4657, [
                            J$.T(4609, '*', 21, false),
                            J$.T(4649, [
                                J$.T(4617, 'Are you saying no just to be negative?', 21, false),
                                J$.T(4625, 'You are being a bit negative.', 21, false),
                                J$.T(4633, 'Why not ?', 21, false),
                                J$.T(4641, 'Why \'no\' ?', 21, false)
                            ], 10, false)
                        ], 10, false)
                    ], 10, false)
                ], 10, false),
                J$.T(4873, [
                    J$.T(4681, 'my', 21, false),
                    J$.T(4689, 2, 22, false),
                    J$.T(4865, [
                        J$.T(4745, [
                            J$.T(4697, '$ * my *', 21, false),
                            J$.T(4737, [
                                J$.T(4705, 'Does that have anything to do with the fact that your (2) ?', 21, false),
                                J$.T(4713, 'Lets discuss further why your (2).', 21, false),
                                J$.T(4721, 'Earlier you said your (2).', 21, false),
                                J$.T(4729, 'But your (2).', 21, false)
                            ], 10, false)
                        ], 10, false),
                        J$.T(4801, [
                            J$.T(4753, '* my* @family *', 21, false),
                            J$.T(4793, [
                                J$.T(4761, 'Tell me more about your family.', 21, false),
                                J$.T(4769, 'Who else in your family (4) ?', 21, false),
                                J$.T(4777, 'Your (3) ?', 21, false),
                                J$.T(4785, 'What else comes to your mind when you think of your (3) ?', 21, false)
                            ], 10, false)
                        ], 10, false),
                        J$.T(4857, [
                            J$.T(4809, '* my *', 21, false),
                            J$.T(4849, [
                                J$.T(4817, 'Your (2) ?', 21, false),
                                J$.T(4825, 'Why do you say your (2) ?', 21, false),
                                J$.T(4833, 'Does that suggest anything else which belongs to you ?', 21, false),
                                J$.T(4841, 'Is it important to you that your (2) ?', 21, false)
                            ], 10, false)
                        ], 10, false)
                    ], 10, false)
                ], 10, false),
                J$.T(5017, [
                    J$.T(4881, 'can', 21, false),
                    J$.T(4889, 0, 22, false),
                    J$.T(5009, [
                        J$.T(4945, [
                            J$.T(4897, '* can you *', 21, false),
                            J$.T(4937, [
                                J$.T(4905, 'You believe I can (2) don\'t you ?', 21, false),
                                J$.T(4913, 'goto what', 21, false),
                                J$.T(4921, 'You want me to be able to (2).', 21, false),
                                J$.T(4929, 'Perhaps you would like to be able to (2) yourself.', 21, false)
                            ], 10, false)
                        ], 10, false),
                        J$.T(5001, [
                            J$.T(4953, '* can i *', 21, false),
                            J$.T(4993, [
                                J$.T(4961, 'Whether or not you can (2) depends on you more than on me.', 21, false),
                                J$.T(4969, 'Do you want to be able to (2) ?', 21, false),
                                J$.T(4977, 'Perhaps you don\'t want to (2).', 21, false),
                                J$.T(4985, 'goto what', 21, false)
                            ], 10, false)
                        ], 10, false)
                    ], 10, false)
                ], 10, false),
                J$.T(5145, [
                    J$.T(5025, 'what', 21, false),
                    J$.T(5033, 0, 22, false),
                    J$.T(5137, [J$.T(5129, [
                            J$.T(5041, '*', 21, false),
                            J$.T(5121, [
                                J$.T(5049, 'Why do you ask ?', 21, false),
                                J$.T(5057, 'Does that question interest you ?', 21, false),
                                J$.T(5065, 'What is it you really want to know ?', 21, false),
                                J$.T(5073, 'Are such questions much on your mind ?', 21, false),
                                J$.T(5081, 'What answer would please you most ?', 21, false),
                                J$.T(5089, 'What do you think ?', 21, false),
                                J$.T(5097, 'What comes to mind when you ask that ?', 21, false),
                                J$.T(5105, 'Have you asked such questions before ?', 21, false),
                                J$.T(5113, 'Have you asked anyone else ?', 21, false)
                            ], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false),
                J$.T(5209, [
                    J$.T(5153, 'who', 21, false),
                    J$.T(5161, 0, 22, false),
                    J$.T(5201, [J$.T(5193, [
                            J$.T(5169, 'who *', 21, false),
                            J$.T(5185, [J$.T(5177, 'goto what', 21, false)], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false),
                J$.T(5273, [
                    J$.T(5217, 'when', 21, false),
                    J$.T(5225, 0, 22, false),
                    J$.T(5265, [J$.T(5257, [
                            J$.T(5233, 'when *', 21, false),
                            J$.T(5249, [J$.T(5241, 'goto what', 21, false)], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false),
                J$.T(5337, [
                    J$.T(5281, 'where', 21, false),
                    J$.T(5289, 0, 22, false),
                    J$.T(5329, [J$.T(5321, [
                            J$.T(5297, 'where *', 21, false),
                            J$.T(5313, [J$.T(5305, 'goto what', 21, false)], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false),
                J$.T(5401, [
                    J$.T(5345, 'how', 21, false),
                    J$.T(5353, 0, 22, false),
                    J$.T(5393, [J$.T(5385, [
                            J$.T(5361, 'how *', 21, false),
                            J$.T(5377, [J$.T(5369, 'goto what', 21, false)], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false),
                J$.T(5489, [
                    J$.T(5409, 'because', 21, false),
                    J$.T(5417, 0, 22, false),
                    J$.T(5481, [J$.T(5473, [
                            J$.T(5425, '*', 21, false),
                            J$.T(5465, [
                                J$.T(5433, 'Is that the real reason ?', 21, false),
                                J$.T(5441, 'Don\'t any other reasons come to mind ?', 21, false),
                                J$.T(5449, 'Does that reason seem to explain anything else ?', 21, false),
                                J$.T(5457, 'What other reasons might there be ?', 21, false)
                            ], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false),
                J$.T(5681, [
                    J$.T(5497, 'why', 21, false),
                    J$.T(5505, 0, 22, false),
                    J$.T(5673, [
                        J$.T(5569, [
                            J$.T(5513, '* why don\'t you *', 21, false),
                            J$.T(5561, [
                                J$.T(5521, 'Do you believe I don\'t (2) ?', 21, false),
                                J$.T(5529, 'Perhaps I will (2) in good time.', 21, false),
                                J$.T(5537, 'Should you (2) yourself ?', 21, false),
                                J$.T(5545, 'You want me to (2) ?', 21, false),
                                J$.T(5553, 'goto what', 21, false)
                            ], 10, false)
                        ], 10, false),
                        J$.T(5633, [
                            J$.T(5577, '* why can\'t i *', 21, false),
                            J$.T(5625, [
                                J$.T(5585, 'Do you think you should be able to (2) ?', 21, false),
                                J$.T(5593, 'Do you want to be able to (2) ?', 21, false),
                                J$.T(5601, 'Do you believe this will help you to (2) ?', 21, false),
                                J$.T(5609, 'Have you any idea why you can\'t (2) ?', 21, false),
                                J$.T(5617, 'goto what', 21, false)
                            ], 10, false)
                        ], 10, false),
                        J$.T(5665, [
                            J$.T(5641, '*', 21, false),
                            J$.T(5657, [J$.T(5649, 'goto what', 21, false)], 10, false)
                        ], 10, false)
                    ], 10, false)
                ], 10, false),
                J$.T(5809, [
                    J$.T(5689, 'everyone', 21, false),
                    J$.T(5697, 2, 22, false),
                    J$.T(5801, [J$.T(5793, [
                            J$.T(5705, '* @everyone *', 21, false),
                            J$.T(5785, [
                                J$.T(5713, 'Really, (2) ?', 21, false),
                                J$.T(5721, 'Surely not (2).', 21, false),
                                J$.T(5729, 'Can you think of anyone in particular ?', 21, false),
                                J$.T(5737, 'Who, for example?', 21, false),
                                J$.T(5745, 'Are you thinking of a very special person ?', 21, false),
                                J$.T(5753, 'Who, may I ask ?', 21, false),
                                J$.T(5761, 'Someone special perhaps ?', 21, false),
                                J$.T(5769, 'You have a particular person in mind, don\'t you ?', 21, false),
                                J$.T(5777, 'Who do you think you\'re talking about ?', 21, false)
                            ], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false),
                J$.T(5873, [
                    J$.T(5817, 'everybody', 21, false),
                    J$.T(5825, 2, 22, false),
                    J$.T(5865, [J$.T(5857, [
                            J$.T(5833, '*', 21, false),
                            J$.T(5849, [J$.T(5841, 'goto everyone', 21, false)], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false),
                J$.T(5937, [
                    J$.T(5881, 'nobody', 21, false),
                    J$.T(5889, 2, 22, false),
                    J$.T(5929, [J$.T(5921, [
                            J$.T(5897, '*', 21, false),
                            J$.T(5913, [J$.T(5905, 'goto everyone', 21, false)], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false),
                J$.T(6001, [
                    J$.T(5945, 'noone', 21, false),
                    J$.T(5953, 2, 22, false),
                    J$.T(5993, [J$.T(5985, [
                            J$.T(5961, '*', 21, false),
                            J$.T(5977, [J$.T(5969, 'goto everyone', 21, false)], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false),
                J$.T(6089, [
                    J$.T(6009, 'always', 21, false),
                    J$.T(6017, 1, 22, false),
                    J$.T(6081, [J$.T(6073, [
                            J$.T(6025, '*', 21, false),
                            J$.T(6065, [
                                J$.T(6033, 'Can you think of a specific example ?', 21, false),
                                J$.T(6041, 'When ?', 21, false),
                                J$.T(6049, 'What incident are you thinking of ?', 21, false),
                                J$.T(6057, 'Really, always ?', 21, false)
                            ], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false),
                J$.T(6209, [
                    J$.T(6097, 'alike', 21, false),
                    J$.T(6105, 10, 22, false),
                    J$.T(6201, [J$.T(6193, [
                            J$.T(6113, '*', 21, false),
                            J$.T(6185, [
                                J$.T(6121, 'In what way ?', 21, false),
                                J$.T(6129, 'What resemblence do you see ?', 21, false),
                                J$.T(6137, 'What does that similarity suggest to you ?', 21, false),
                                J$.T(6145, 'What other connections do you see ?', 21, false),
                                J$.T(6153, 'What do you suppose that resemblence means ?', 21, false),
                                J$.T(6161, 'What is the connection, do you suppose ?', 21, false),
                                J$.T(6169, 'Could there really be some connection ?', 21, false),
                                J$.T(6177, 'How ?', 21, false)
                            ], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false),
                J$.T(6273, [
                    J$.T(6217, 'like', 21, false),
                    J$.T(6225, 10, 22, false),
                    J$.T(6265, [J$.T(6257, [
                            J$.T(6233, '* @be *like *', 21, false),
                            J$.T(6249, [J$.T(6241, 'goto alike', 21, false)], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false),
                J$.T(6385, [
                    J$.T(6281, 'different', 21, false),
                    J$.T(6289, 0, 22, false),
                    J$.T(6377, [J$.T(6369, [
                            J$.T(6297, '*', 21, false),
                            J$.T(6361, [
                                J$.T(6305, 'How is it different ?', 21, false),
                                J$.T(6313, 'What differences do you see ?', 21, false),
                                J$.T(6321, 'What does that difference suggest to you ?', 21, false),
                                J$.T(6329, 'What other distinctions do you see ?', 21, false),
                                J$.T(6337, 'What do you suppose that disparity means ?', 21, false),
                                J$.T(6345, 'Could there be some connection, do you suppose ?', 21, false),
                                J$.T(6353, 'How ?', 21, false)
                            ], 10, false)
                        ], 10, false)], 10, false)
                ], 10, false)
            ], 10, false), 0));
            J$.X1(6553, J$.P(6545, J$.R(6417, 'exports', exports, 2), 'elizaPostTransforms', J$.T(6537, [
                J$.T(6425, / old old/g, 14, false),
                J$.T(6433, ' old', 21, false),
                J$.T(6441, /\bthey were( not)? me\b/g, 14, false),
                J$.T(6449, 'it was$1 me', 21, false),
                J$.T(6457, /\bthey are( not)? me\b/g, 14, false),
                J$.T(6465, 'it is$1 me', 21, false),
                J$.T(6473, /Are they( always)? me\b/, 14, false),
                J$.T(6481, 'it is$1 me', 21, false),
                J$.T(6489, /\bthat your( own)? (\w+)( now)? \?/, 14, false),
                J$.T(6497, 'that you have your$1 $2 ?', 21, false),
                J$.T(6505, /\bI to have (\w+)/, 14, false),
                J$.T(6513, 'I have $1', 21, false),
                J$.T(6521, /Earlier you said your( own)? (\w+)( now)?\./, 14, false),
                J$.T(6529, 'Earlier you talked about your $2.', 21, false)
            ], 10, false), 0));
        } catch (J$e) {
            J$.Ex(6585, J$e);
        } finally {
            if (J$.Sr(6593)) {
                J$.L();
                continue jalangiLabel58;
            } else {
                J$.L();
                break jalangiLabel58;
            }
        }
    }
// JALANGI DO NOT INSTRUMENT
