// bson/lib/bson/symbol_jalangi_.js
J$.iids = {"8":[4,9,4,34],"9":[1,20,1,27],"10":[4,11,4,33],"17":[1,28,1,32],"18":[4,9,4,34],"25":[1,20,1,33],"33":[1,20,1,33],"41":[1,20,1,33],"49":[2,26,2,38],"57":[2,26,2,45],"65":[2,26,2,45],"73":[2,26,2,45],"81":[4,11,4,15],"89":[4,27,4,33],"97":[5,20,5,26],"105":[5,27,5,32],"113":[5,16,5,33],"121":[5,16,5,33],"129":[5,9,5,34],"137":[6,5,6,9],"145":[6,22,6,30],"153":[6,5,6,30],"161":[6,5,6,31],"169":[7,5,7,9],"177":[7,18,7,23],"185":[7,5,7,23],"193":[7,5,7,24],"201":[3,1,8,2],"209":[3,1,8,2],"217":[3,1,8,2],"225":[9,1,9,7],"233":[9,1,9,17],"241":[10,12,10,16],"249":[10,12,10,22],"257":[10,12,10,22],"265":[10,5,10,23],"273":[9,28,11,2],"281":[9,28,11,2],"289":[9,28,11,2],"297":[9,1,11,2],"305":[9,1,11,3],"313":[12,1,12,7],"321":[12,1,12,17],"329":[13,12,13,16],"337":[13,12,13,22],"345":[13,12,13,22],"353":[13,5,13,23],"361":[12,29,14,2],"369":[12,29,14,2],"377":[12,29,14,2],"385":[12,1,14,2],"393":[12,1,14,3],"401":[15,1,15,7],"409":[15,1,15,17],"417":[16,12,16,16],"425":[16,12,16,22],"433":[16,12,16,22],"441":[16,5,16,23],"449":[15,28,17,2],"457":[15,28,17,2],"465":[15,28,17,2],"473":[15,1,17,2],"481":[15,1,17,3],"489":[18,1,18,7],"497":[18,1,18,17],"505":[19,12,19,16],"513":[19,12,19,22],"521":[19,12,19,22],"529":[19,5,19,23],"537":[18,27,20,2],"545":[18,27,20,2],"553":[18,27,20,2],"561":[18,1,20,2],"569":[18,1,20,3],"577":[21,1,21,7],"585":[21,18,21,24],"593":[21,1,21,24],"601":[21,1,21,25],"609":[22,1,22,7],"617":[22,1,22,15],"625":[22,25,22,31],"633":[22,1,22,31],"641":[22,1,22,32],"649":[1,1,22,32],"657":[1,1,22,32],"665":[1,1,22,32],"673":[3,1,8,2],"681":[1,1,22,32],"689":[4,5,5,34],"697":[3,1,8,2],"705":[3,1,8,2],"713":[9,28,11,2],"721":[9,28,11,2],"729":[12,29,14,2],"737":[12,29,14,2],"745":[15,28,17,2],"753":[15,28,17,2],"761":[18,27,20,2],"769":[18,27,20,2],"777":[1,1,22,32],"785":[1,1,22,32],"nBranches":2,"originalCodeFileName":"/home/blake/ExpoSE/Targets/harvesterjs/node_modules/harvesterjs/node_modules/mongojs/node_modules/mongodb/node_modules/bson/lib/bson/symbol.js","instrumentedCodeFileName":"/home/blake/ExpoSE/Targets/harvesterjs/node_modules/harvesterjs/node_modules/mongojs/node_modules/mongodb/node_modules/bson/lib/bson/symbol_jalangi_.js","code":"/**\n * A class representation of the BSON Symbol type.\n *\n * @class\n * @deprecated\n * @param {string} value the string representing the symbol.\n * @return {Symbol}\n */\nfunction Symbol(value) {\n  if(!(this instanceof Symbol)) return new Symbol(value);\n  this._bsontype = 'Symbol';\n  this.value = value;\n}\n\n/**\n * Access the wrapped string value.\n *\n * @method\n * @return {String} returns the wrapped string.\n */\nSymbol.prototype.valueOf = function() {\n  return this.value;\n};\n\n/**\n * @ignore\n */\nSymbol.prototype.toString = function() {\n  return this.value;\n}\n\n/**\n * @ignore\n */\nSymbol.prototype.inspect = function() {\n  return this.value;\n}\n\n/**\n * @ignore\n */\nSymbol.prototype.toJSON = function() {\n  return this.value;\n}\n\nmodule.exports = Symbol;\nmodule.exports.Symbol = Symbol;"};
jalangiLabel1461:
    while (true) {
        try {
            J$.Se(649, '/home/blake/ExpoSE/Targets/harvesterjs/node_modules/harvesterjs/node_modules/mongojs/node_modules/mongodb/node_modules/bson/lib/bson/symbol_jalangi_.js', '/home/blake/ExpoSE/Targets/harvesterjs/node_modules/harvesterjs/node_modules/mongojs/node_modules/mongodb/node_modules/bson/lib/bson/symbol.js');
            function Symbol(value) {
                jalangiLabel1456:
                    while (true) {
                        try {
                            J$.Fe(201, arguments.callee, this, arguments);
                            arguments = J$.N(209, 'arguments', arguments, 4);
                            value = J$.N(217, 'value', value, 4);
                            if (J$.X1(689, J$.C(8, J$.U(18, '!', J$.B(10, 'instanceof', J$.R(81, 'this', this, 0), J$.R(89, 'Symbol', Symbol, 1), 0)))))
                                return J$.X1(129, J$.Rt(121, J$.F(113, J$.R(97, 'Symbol', Symbol, 1), 1)(J$.R(105, 'value', value, 0))));
                            J$.X1(161, J$.P(153, J$.R(137, 'this', this, 0), '_bsontype', J$.T(145, 'Symbol', 21, false), 0));
                            J$.X1(193, J$.P(185, J$.R(169, 'this', this, 0), 'value', J$.R(177, 'value', value, 0), 0));
                        } catch (J$e) {
                            J$.Ex(697, J$e);
                        } finally {
                            if (J$.Fr(705))
                                continue jalangiLabel1456;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(657, '__secret__S$', __secret__S$, 0);
            J$.N(665, '__secret__traits__', __secret__traits__, 0);
            Symbol = J$.N(681, 'Symbol', J$.T(673, Symbol, 12, false, 201), 0);
            var __secret__S$ = J$.X1(41, J$.W(33, '__secret__S$', J$.F(25, J$.R(9, 'require', require, 2), 0)(J$.T(17, 'S$', 21, false)), __secret__S$, 3));
            var __secret__traits__ = J$.X1(73, J$.W(65, '__secret__traits__', J$.G(57, J$.R(49, '__secret__S$', __secret__S$, 1), 'Traits', 0), __secret__traits__, 3));
            J$.X1(305, J$.P(297, J$.G(233, J$.R(225, 'Symbol', Symbol, 1), 'prototype', 0), 'valueOf', J$.T(289, function () {
                jalangiLabel1457:
                    while (true) {
                        try {
                            J$.Fe(273, arguments.callee, this, arguments);
                            arguments = J$.N(281, 'arguments', arguments, 4);
                            return J$.X1(265, J$.Rt(257, J$.G(249, J$.R(241, 'this', this, 0), 'value', 0)));
                        } catch (J$e) {
                            J$.Ex(713, J$e);
                        } finally {
                            if (J$.Fr(721))
                                continue jalangiLabel1457;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 273), 0));
            J$.X1(393, J$.P(385, J$.G(321, J$.R(313, 'Symbol', Symbol, 1), 'prototype', 0), 'toString', J$.T(377, function () {
                jalangiLabel1458:
                    while (true) {
                        try {
                            J$.Fe(361, arguments.callee, this, arguments);
                            arguments = J$.N(369, 'arguments', arguments, 4);
                            return J$.X1(353, J$.Rt(345, J$.G(337, J$.R(329, 'this', this, 0), 'value', 0)));
                        } catch (J$e) {
                            J$.Ex(729, J$e);
                        } finally {
                            if (J$.Fr(737))
                                continue jalangiLabel1458;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 361), 0));
            J$.X1(481, J$.P(473, J$.G(409, J$.R(401, 'Symbol', Symbol, 1), 'prototype', 0), 'inspect', J$.T(465, function () {
                jalangiLabel1459:
                    while (true) {
                        try {
                            J$.Fe(449, arguments.callee, this, arguments);
                            arguments = J$.N(457, 'arguments', arguments, 4);
                            return J$.X1(441, J$.Rt(433, J$.G(425, J$.R(417, 'this', this, 0), 'value', 0)));
                        } catch (J$e) {
                            J$.Ex(745, J$e);
                        } finally {
                            if (J$.Fr(753))
                                continue jalangiLabel1459;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 449), 0));
            J$.X1(569, J$.P(561, J$.G(497, J$.R(489, 'Symbol', Symbol, 1), 'prototype', 0), 'toJSON', J$.T(553, function () {
                jalangiLabel1460:
                    while (true) {
                        try {
                            J$.Fe(537, arguments.callee, this, arguments);
                            arguments = J$.N(545, 'arguments', arguments, 4);
                            return J$.X1(529, J$.Rt(521, J$.G(513, J$.R(505, 'this', this, 0), 'value', 0)));
                        } catch (J$e) {
                            J$.Ex(761, J$e);
                        } finally {
                            if (J$.Fr(769))
                                continue jalangiLabel1460;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 537), 0));
            J$.X1(601, J$.P(593, J$.R(577, 'module', module, 2), 'exports', J$.R(585, 'Symbol', Symbol, 1), 0));
            J$.X1(641, J$.P(633, J$.G(617, J$.R(609, 'module', module, 2), 'exports', 0), 'Symbol', J$.R(625, 'Symbol', Symbol, 1), 0));
        } catch (J$e) {
            J$.Ex(777, J$e);
        } finally {
            if (J$.Sr(785)) {
                J$.L();
                continue jalangiLabel1461;
            } else {
                J$.L();
                break jalangiLabel1461;
            }
        }
    }
// JALANGI DO NOT INSTRUMENT
