// jszip/lib/compressions_jalangi_.js
J$.iids = {"9":[1,20,1,27],"17":[1,28,1,32],"25":[1,20,1,33],"33":[1,20,1,33],"41":[1,20,1,33],"49":[2,26,2,38],"57":[2,26,2,45],"65":[2,26,2,45],"73":[2,26,2,45],"81":[3,1,3,13],"89":[3,1,3,14],"97":[4,1,4,8],"105":[5,12,5,18],"113":[7,16,7,23],"121":[7,16,7,23],"129":[7,9,7,24],"137":[6,15,8,6],"145":[6,15,8,6],"153":[6,15,8,6],"161":[6,15,8,6],"169":[6,15,8,6],"177":[10,16,10,23],"185":[10,16,10,23],"193":[10,9,10,24],"201":[9,17,11,6],"209":[9,17,11,6],"217":[9,17,11,6],"225":[9,17,11,6],"233":[12,24,12,28],"241":[13,26,13,30],"249":[4,17,14,2],"257":[4,1,14,2],"265":[4,1,14,3],"273":[15,1,15,8],"281":[15,19,15,26],"289":[15,27,15,36],"297":[15,19,15,37],"305":[15,1,15,37],"313":[15,1,15,38],"321":[1,1,15,38],"329":[1,1,15,38],"337":[1,1,15,38],"345":[6,15,8,6],"353":[6,15,8,6],"361":[9,17,11,6],"369":[9,17,11,6],"377":[1,1,15,38],"385":[1,1,15,38],"nBranches":0,"originalCodeFileName":"/home/blake/ExpoSE/Targets/excel-builder/node_modules/jszip/lib/compressions.js","instrumentedCodeFileName":"/home/blake/ExpoSE/Targets/excel-builder/node_modules/jszip/lib/compressions_jalangi_.js","code":"'use strict';\nexports.STORE = {\n    magic: \"\\x00\\x00\",\n    compress: function(content, compressionOptions) {\n        return content; // no compression\n    },\n    uncompress: function(content) {\n        return content; // no compression\n    },\n    compressInputType: null,\n    uncompressInputType: null\n};\nexports.DEFLATE = require('./flate');\n"};
jalangiLabel984:
    while (true) {
        try {
            J$.Se(321, '/home/blake/ExpoSE/Targets/excel-builder/node_modules/jszip/lib/compressions_jalangi_.js', '/home/blake/ExpoSE/Targets/excel-builder/node_modules/jszip/lib/compressions.js');
            J$.N(329, '__secret__S$', __secret__S$, 0);
            J$.N(337, '__secret__traits__', __secret__traits__, 0);
            var __secret__S$ = J$.X1(41, J$.W(33, '__secret__S$', J$.F(25, J$.R(9, 'require', require, 2), 0)(J$.T(17, 'S$', 21, false)), __secret__S$, 3));
            var __secret__traits__ = J$.X1(73, J$.W(65, '__secret__traits__', J$.G(57, J$.R(49, '__secret__S$', __secret__S$, 1), 'Traits', 0), __secret__traits__, 3));
            J$.X1(89, J$.T(81, 'use strict', 21, false));
            J$.X1(265, J$.P(257, J$.R(97, 'exports', exports, 2), 'STORE', J$.T(249, {
                magic: J$.T(105, '\0\0', 21, false),
                compress: J$.T(169, function (content, compressionOptions) {
                    jalangiLabel982:
                        while (true) {
                            try {
                                J$.Fe(137, arguments.callee, this, arguments);
                                arguments = J$.N(145, 'arguments', arguments, 4);
                                content = J$.N(153, 'content', content, 4);
                                compressionOptions = J$.N(161, 'compressionOptions', compressionOptions, 4);
                                return J$.X1(129, J$.Rt(121, J$.R(113, 'content', content, 0)));
                            } catch (J$e) {
                                J$.Ex(345, J$e);
                            } finally {
                                if (J$.Fr(353))
                                    continue jalangiLabel982;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false, 137),
                uncompress: J$.T(225, function (content) {
                    jalangiLabel983:
                        while (true) {
                            try {
                                J$.Fe(201, arguments.callee, this, arguments);
                                arguments = J$.N(209, 'arguments', arguments, 4);
                                content = J$.N(217, 'content', content, 4);
                                return J$.X1(193, J$.Rt(185, J$.R(177, 'content', content, 0)));
                            } catch (J$e) {
                                J$.Ex(361, J$e);
                            } finally {
                                if (J$.Fr(369))
                                    continue jalangiLabel983;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false, 201),
                compressInputType: J$.T(233, null, 25, false),
                uncompressInputType: J$.T(241, null, 25, false)
            }, 11, false), 0));
            J$.X1(313, J$.P(305, J$.R(273, 'exports', exports, 2), 'DEFLATE', J$.F(297, J$.R(281, 'require', require, 2), 0)(J$.T(289, './flate', 21, false)), 0));
        } catch (J$e) {
            J$.Ex(377, J$e);
        } finally {
            if (J$.Sr(385)) {
                J$.L();
                continue jalangiLabel984;
            } else {
                J$.L();
                break jalangiLabel984;
            }
        }
    }
// JALANGI DO NOT INSTRUMENT
