// pinyinize/lib/tone-map_jalangi_.js
J$.iids = {"9":[1,20,1,27],"17":[1,28,1,32],"25":[1,20,1,33],"33":[1,20,1,33],"41":[1,20,1,33],"49":[2,26,2,38],"57":[2,26,2,45],"65":[2,26,2,45],"73":[2,26,2,45],"81":[3,1,3,7],"89":[5,12,5,15],"97":[6,12,6,15],"105":[7,12,7,15],"113":[8,12,8,15],"121":[4,8,9,6],"129":[11,12,11,15],"137":[12,12,12,15],"145":[13,12,13,15],"153":[14,12,14,15],"161":[10,8,15,6],"169":[17,12,17,15],"177":[18,12,18,15],"185":[19,12,19,15],"193":[20,12,20,15],"201":[16,8,21,6],"209":[23,12,23,15],"217":[24,12,24,15],"225":[25,12,25,15],"233":[26,12,26,15],"241":[22,8,27,6],"249":[29,12,29,15],"257":[30,12,30,15],"265":[31,12,31,15],"273":[32,12,32,15],"281":[28,8,33,6],"289":[35,12,35,15],"297":[36,12,36,15],"305":[37,12,37,15],"313":[38,12,38,15],"321":[34,8,39,6],"329":[41,12,41,15],"337":[42,12,42,15],"345":[43,12,43,15],"353":[44,12,44,15],"361":[40,10,45,6],"369":[3,18,46,2],"377":[3,1,46,2],"385":[3,1,46,3],"393":[1,1,46,3],"401":[1,1,46,3],"409":[1,1,46,3],"417":[1,1,46,3],"425":[1,1,46,3],"nBranches":0,"originalCodeFileName":"/home/blake/ExpoSE/Targets/pinyinize/node_modules/pinyinize/lib/tone-map.js","instrumentedCodeFileName":"/home/blake/ExpoSE/Targets/pinyinize/node_modules/pinyinize/lib/tone-map_jalangi_.js","code":"module.exports = {\n  a: {\n    1: 'ā',\n    2: 'á',\n    3: 'ǎ',\n    4: 'à'\n  },\n  e: {\n    1: 'ē',\n    2: 'é',\n    3: 'ě',\n    4: 'è'\n  },\n  i: {\n    1: 'ī',\n    2: 'í',\n    3: 'ǐ',\n    4: 'ì'\n  },\n  o: {\n    1: 'ō',\n    2: 'ó',\n    3: 'ǒ',\n    4: 'ò'\n  },\n  u: {\n    1: 'ū',\n    2: 'ú',\n    3: 'ǔ',\n    4: 'ù'\n  },\n  v: {\n    1: 'ǖ',\n    2: 'ǘ',\n    3: 'ǚ',\n    4: 'ǜ'\n  },\n  'ü': {\n    1: 'ǖ',\n    2: 'ǘ',\n    3: 'ǚ',\n    4: 'ǜ'\n  }\n};\n"};
jalangiLabel509:
    while (true) {
        try {
            J$.Se(393, '/home/blake/ExpoSE/Targets/pinyinize/node_modules/pinyinize/lib/tone-map_jalangi_.js', '/home/blake/ExpoSE/Targets/pinyinize/node_modules/pinyinize/lib/tone-map.js');
            J$.N(401, '__secret__S$', __secret__S$, 0);
            J$.N(409, '__secret__traits__', __secret__traits__, 0);
            var __secret__S$ = J$.X1(41, J$.W(33, '__secret__S$', J$.F(25, J$.R(9, 'require', require, 2), 0)(J$.T(17, 'S$', 21, false)), __secret__S$, 3));
            var __secret__traits__ = J$.X1(73, J$.W(65, '__secret__traits__', J$.G(57, J$.R(49, '__secret__S$', __secret__S$, 1), 'Traits', 0), __secret__traits__, 3));
            J$.X1(385, J$.P(377, J$.R(81, 'module', module, 2), 'exports', J$.T(369, {
                a: J$.T(121, {
                    1: J$.T(89, 'ā', 21, false),
                    2: J$.T(97, 'á', 21, false),
                    3: J$.T(105, 'ǎ', 21, false),
                    4: J$.T(113, 'à', 21, false)
                }, 11, false),
                e: J$.T(161, {
                    1: J$.T(129, 'ē', 21, false),
                    2: J$.T(137, 'é', 21, false),
                    3: J$.T(145, 'ě', 21, false),
                    4: J$.T(153, 'è', 21, false)
                }, 11, false),
                i: J$.T(201, {
                    1: J$.T(169, 'ī', 21, false),
                    2: J$.T(177, 'í', 21, false),
                    3: J$.T(185, 'ǐ', 21, false),
                    4: J$.T(193, 'ì', 21, false)
                }, 11, false),
                o: J$.T(241, {
                    1: J$.T(209, 'ō', 21, false),
                    2: J$.T(217, 'ó', 21, false),
                    3: J$.T(225, 'ǒ', 21, false),
                    4: J$.T(233, 'ò', 21, false)
                }, 11, false),
                u: J$.T(281, {
                    1: J$.T(249, 'ū', 21, false),
                    2: J$.T(257, 'ú', 21, false),
                    3: J$.T(265, 'ǔ', 21, false),
                    4: J$.T(273, 'ù', 21, false)
                }, 11, false),
                v: J$.T(321, {
                    1: J$.T(289, 'ǖ', 21, false),
                    2: J$.T(297, 'ǘ', 21, false),
                    3: J$.T(305, 'ǚ', 21, false),
                    4: J$.T(313, 'ǜ', 21, false)
                }, 11, false),
                'ü': J$.T(361, {
                    1: J$.T(329, 'ǖ', 21, false),
                    2: J$.T(337, 'ǘ', 21, false),
                    3: J$.T(345, 'ǚ', 21, false),
                    4: J$.T(353, 'ǜ', 21, false)
                }, 11, false)
            }, 11, false), 0));
        } catch (J$e) {
            J$.Ex(417, J$e);
        } finally {
            if (J$.Sr(425)) {
                J$.L();
                continue jalangiLabel509;
            } else {
                J$.L();
                break jalangiLabel509;
            }
        }
    }
// JALANGI DO NOT INSTRUMENT
