// validator/lib/isHash_jalangi_.js
J$.iids = {"8":[1,5,1,17],"9":[1,6,1,12],"10":[1,5,1,17],"16":[10,12,10,33],"17":[1,6,1,17],"18":[29,27,29,60],"24":[10,12,10,33],"25":[2,5,2,11],"26":[29,27,29,67],"33":[2,19,2,26],"41":[2,27,2,31],"49":[2,19,2,32],"57":[2,5,2,32],"65":[2,5,2,33],"73":[4,1,4,13],"81":[4,1,4,14],"89":[5,1,5,7],"97":[5,23,5,30],"105":[5,32,5,44],"113":[5,55,5,59],"121":[5,46,5,61],"129":[5,1,5,62],"131":[5,1,5,22],"137":[5,1,5,63],"145":[6,1,6,8],"153":[6,19,6,25],"161":[6,1,6,25],"169":[6,1,6,26],"177":[7,21,7,28],"185":[7,29,7,50],"193":[7,21,7,51],"201":[7,21,7,51],"209":[7,21,7,51],"217":[8,22,8,44],"225":[8,45,8,58],"233":[8,22,8,59],"241":[8,22,8,59],"249":[8,22,8,59],"257":[10,12,10,15],"265":[10,19,10,22],"273":[10,19,10,33],"281":[10,36,10,39],"289":[10,53,10,56],"297":[10,42,10,58],"305":[10,12,10,58],"313":[10,5,10,59],"321":[9,1,11,2],"329":[9,1,11,2],"337":[9,1,11,2],"345":[13,10,13,12],"353":[14,10,14,12],"361":[15,11,15,13],"369":[16,13,16,15],"377":[17,13,17,15],"385":[18,13,18,16],"393":[19,16,19,18],"401":[20,16,20,18],"409":[21,15,21,17],"417":[22,15,22,17],"425":[23,15,23,17],"433":[24,12,24,13],"441":[25,13,25,14],"449":[12,15,26,2],"457":[12,15,26,2],"465":[12,15,26,2],"473":[28,6,28,7],"481":[28,9,28,23],"489":[28,9,28,31],"497":[28,6,28,7],"505":[28,33,28,36],"513":[28,5,28,37],"521":[28,5,28,38],"529":[29,20,29,26],"537":[29,27,29,39],"545":[29,42,29,49],"553":[29,50,29,59],"561":[29,42,29,60],"569":[29,63,29,67],"577":[29,16,29,68],"585":[29,16,29,68],"593":[29,16,29,68],"601":[30,12,30,16],"609":[30,22,30,25],"617":[30,12,30,26],"619":[30,12,30,21],"625":[30,12,30,26],"633":[30,5,30,27],"641":[27,1,31,2],"649":[27,1,31,2],"657":[27,1,31,2],"665":[27,1,31,2],"673":[27,1,31,2],"681":[32,1,32,7],"689":[32,18,32,25],"697":[32,26,32,35],"705":[32,18,32,36],"713":[32,1,32,36],"721":[32,1,32,37],"729":[1,1,32,37],"737":[1,1,32,37],"745":[1,1,32,37],"753":[9,1,11,2],"761":[1,1,32,37],"769":[1,1,32,37],"777":[27,1,31,2],"785":[1,1,32,37],"793":[1,1,3,2],"801":[10,12,10,58],"809":[9,1,11,2],"817":[9,1,11,2],"825":[27,1,31,2],"833":[27,1,31,2],"841":[1,1,32,37],"849":[1,1,32,37],"nBranches":6,"originalCodeFileName":"/home/blake/ExpoSE/Targets/mario/node_modules/validator/lib/isHash.js","instrumentedCodeFileName":"/home/blake/ExpoSE/Targets/mario/node_modules/validator/lib/isHash_jalangi_.js","code":"'use strict';\n\nObject.defineProperty(exports, \"__esModule\", {\n  value: true\n});\nexports.default = isHash;\n\nvar _assertString = require('./util/assertString');\n\nvar _assertString2 = _interopRequireDefault(_assertString);\n\nfunction _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }\n\nvar lengths = {\n  md5: 32,\n  md4: 32,\n  sha1: 40,\n  sha256: 64,\n  sha384: 96,\n  sha512: 128,\n  ripemd128: 32,\n  ripemd160: 40,\n  tiger128: 32,\n  tiger160: 40,\n  tiger192: 48,\n  crc32: 8,\n  crc32b: 8\n};\n\nfunction isHash(str, algorithm) {\n  (0, _assertString2.default)(str);\n  var hash = new RegExp('^[a-f0-9]{' + lengths[algorithm] + '}$');\n  return hash.test(str);\n}\nmodule.exports = exports['default'];"};
jalangiLabel142:
    while (true) {
        try {
            J$.Se(729, '/home/blake/ExpoSE/Targets/mario/node_modules/validator/lib/isHash_jalangi_.js', '/home/blake/ExpoSE/Targets/mario/node_modules/validator/lib/isHash.js');
            function _interopRequireDefault(obj) {
                jalangiLabel140:
                    while (true) {
                        try {
                            J$.Fe(321, arguments.callee, this, arguments);
                            arguments = J$.N(329, 'arguments', arguments, 4);
                            obj = J$.N(337, 'obj', obj, 4);
                            return J$.X1(313, J$.Rt(305, J$.X1(801, J$.C(24, J$.C(16, J$.R(257, 'obj', obj, 0)) ? J$.G(273, J$.R(265, 'obj', obj, 0), '__esModule', 0) : J$._())) ? J$.R(281, 'obj', obj, 0) : J$.T(297, {
                                default: J$.R(289, 'obj', obj, 0)
                            }, 11, false)));
                        } catch (J$e) {
                            J$.Ex(809, J$e);
                        } finally {
                            if (J$.Fr(817))
                                continue jalangiLabel140;
                            else
                                return J$.Ra();
                        }
                    }
            }
            function isHash(str, algorithm) {
                jalangiLabel141:
                    while (true) {
                        try {
                            J$.Fe(641, arguments.callee, this, arguments);
                            arguments = J$.N(649, 'arguments', arguments, 4);
                            str = J$.N(657, 'str', str, 4);
                            algorithm = J$.N(665, 'algorithm', algorithm, 4);
                            J$.N(673, 'hash', hash, 0);
                            J$.X1(521, J$.F(513, (J$.X1(497, J$.T(473, 0, 22, false)), J$.G(489, J$.R(481, '_assertString2', _assertString2, 1), 'default', 0)), 0)(J$.R(505, 'str', str, 0)));
                            var hash = J$.X1(593, J$.W(585, 'hash', J$.F(577, J$.R(529, 'RegExp', RegExp, 2), 1)(J$.B(26, '+', J$.B(18, '+', J$.T(537, '^[a-f0-9]{', 21, false), J$.G(561, J$.R(545, 'lengths', lengths, 1), J$.R(553, 'algorithm', algorithm, 0), 4), 0), J$.T(569, '}$', 21, false), 0)), hash, 1));
                            return J$.X1(633, J$.Rt(625, J$.M(617, J$.R(601, 'hash', hash, 0), 'test', 0)(J$.R(609, 'str', str, 0))));
                        } catch (J$e) {
                            J$.Ex(825, J$e);
                        } finally {
                            if (J$.Fr(833))
                                continue jalangiLabel141;
                            else
                                return J$.Ra();
                        }
                    }
            }
            J$.N(737, '_assertString', _assertString, 0);
            J$.N(745, '_assertString2', _assertString2, 0);
            _interopRequireDefault = J$.N(761, '_interopRequireDefault', J$.T(753, _interopRequireDefault, 12, false, 321), 0);
            J$.N(769, 'lengths', lengths, 0);
            isHash = J$.N(785, 'isHash', J$.T(777, isHash, 12, false, 641), 0);
            if (J$.X1(793, J$.C(8, J$.U(10, '!', J$.G(17, J$.R(9, 'Object', Object, 2), '_s$_', 0))))) {
                J$.X1(65, J$.P(57, J$.R(25, 'Object', Object, 2), '_s$_', J$.F(49, J$.R(33, 'require', require, 2), 0)(J$.T(41, 'S$', 21, false)), 0));
            }
            J$.X1(81, J$.T(73, 'use strict', 21, false));
            J$.X1(137, J$.M(129, J$.R(89, 'Object', Object, 2), 'defineProperty', 0)(J$.R(97, 'exports', exports, 2), J$.T(105, '__esModule', 21, false), J$.T(121, {
                value: J$.T(113, true, 23, false)
            }, 11, false)));
            J$.X1(169, J$.P(161, J$.R(145, 'exports', exports, 2), 'default', J$.R(153, 'isHash', isHash, 1), 0));
            var _assertString = J$.X1(209, J$.W(201, '_assertString', J$.F(193, J$.R(177, 'require', require, 2), 0)(J$.T(185, './util/assertString', 21, false)), _assertString, 3));
            var _assertString2 = J$.X1(249, J$.W(241, '_assertString2', J$.F(233, J$.R(217, '_interopRequireDefault', _interopRequireDefault, 1), 0)(J$.R(225, '_assertString', _assertString, 1)), _assertString2, 3));
            var lengths = J$.X1(465, J$.W(457, 'lengths', J$.T(449, {
                md5: J$.T(345, 32, 22, false),
                md4: J$.T(353, 32, 22, false),
                sha1: J$.T(361, 40, 22, false),
                sha256: J$.T(369, 64, 22, false),
                sha384: J$.T(377, 96, 22, false),
                sha512: J$.T(385, 128, 22, false),
                ripemd128: J$.T(393, 32, 22, false),
                ripemd160: J$.T(401, 40, 22, false),
                tiger128: J$.T(409, 32, 22, false),
                tiger160: J$.T(417, 40, 22, false),
                tiger192: J$.T(425, 48, 22, false),
                crc32: J$.T(433, 8, 22, false),
                crc32b: J$.T(441, 8, 22, false)
            }, 11, false), lengths, 3));
            J$.X1(721, J$.P(713, J$.R(681, 'module', module, 2), 'exports', J$.G(705, J$.R(689, 'exports', exports, 2), J$.T(697, 'default', 21, false), 4), 0));
        } catch (J$e) {
            J$.Ex(841, J$e);
        } finally {
            if (J$.Sr(849)) {
                J$.L();
                continue jalangiLabel142;
            } else {
                J$.L();
                break jalangiLabel142;
            }
        }
    }
// JALANGI DO NOT INSTRUMENT
