// excel-builder/src/Excel/Drawing/index_jalangi_.js
J$.iids = {"8":[13,18,13,30],"9":[1,20,1,27],"16":[15,17,15,21],"17":[1,28,1,32],"24":[16,14,16,30],"25":[1,20,1,33],"32":[19,14,19,29],"33":[1,20,1,33],"40":[22,14,22,29],"41":[1,20,1,33],"49":[2,26,2,38],"57":[2,26,2,45],"65":[2,26,2,45],"73":[2,26,2,45],"81":[3,1,3,13],"89":[3,1,3,14],"97":[4,9,4,16],"105":[4,17,4,25],"113":[4,9,4,26],"121":[4,9,4,26],"129":[4,9,4,26],"137":[5,22,5,29],"145":[5,30,5,48],"153":[5,22,5,49],"161":[5,22,5,49],"169":[5,22,5,49],"177":[6,21,6,28],"185":[6,29,6,46],"193":[6,21,6,47],"201":[6,21,6,47],"209":[6,21,6,47],"217":[7,21,7,28],"225":[7,29,7,46],"233":[7,21,7,47],"241":[7,21,7,47],"249":[7,21,7,47],"257":[9,5,9,9],"265":[9,15,9,16],"273":[9,26,9,35],"281":[9,15,9,36],"283":[9,15,9,25],"289":[9,5,9,36],"297":[9,5,9,37],"305":[8,15,10,2],"313":[8,15,10,2],"321":[8,15,10,2],"329":[8,15,10,2],"337":[8,15,10,2],"345":[11,1,11,2],"353":[11,10,11,17],"361":[11,10,11,27],"369":[13,18,13,24],"377":[13,28,13,30],"385":[13,18,13,30],"393":[13,9,13,31],"401":[14,9,14,15],"409":[14,26,14,30],"417":[14,9,14,30],"425":[14,9,14,31],"433":[15,17,15,21],"441":[17,13,17,17],"449":[17,31,17,45],"457":[17,46,17,52],"465":[17,27,17,53],"473":[17,13,17,53],"481":[17,13,17,54],"489":[16,14,16,30],"497":[20,13,20,17],"505":[20,31,20,44],"513":[20,45,20,51],"521":[20,27,20,52],"529":[20,13,20,52],"537":[20,13,20,53],"545":[19,14,19,29],"553":[23,13,23,17],"561":[23,31,23,44],"569":[23,45,23,51],"577":[23,27,23,52],"585":[23,13,23,52],"593":[23,13,23,53],"601":[22,14,22,29],"609":[26,16,26,20],"617":[26,16,26,27],"625":[26,16,26,27],"633":[26,9,26,28],"641":[12,19,27,6],"649":[12,19,27,6],"657":[12,19,27,6],"665":[12,19,27,6],"673":[12,19,27,6],"681":[11,29,28,2],"689":[11,1,28,3],"691":[11,1,11,9],"697":[11,1,28,4],"705":[29,1,29,7],"713":[29,25,29,32],"721":[32,20,32,27],"729":[32,28,32,46],"737":[32,20,32,47],"745":[32,20,32,47],"753":[32,13,32,48],"761":[31,14,33,10],"769":[31,14,33,10],"777":[31,14,33,10],"785":[30,21,34,6],"793":[37,20,37,27],"801":[37,28,37,37],"809":[37,20,37,38],"817":[37,20,37,38],"825":[37,13,37,39],"833":[36,14,38,10],"841":[36,14,38,10],"849":[36,14,38,10],"857":[35,12,39,6],"865":[42,20,42,27],"873":[42,28,42,45],"881":[42,20,42,46],"889":[42,20,42,46],"897":[42,13,42,47],"905":[41,14,43,10],"913":[41,14,43,10],"921":[41,14,43,10],"929":[40,20,44,6],"937":[47,20,47,27],"945":[47,28,47,39],"953":[47,20,47,40],"961":[47,20,47,40],"969":[47,13,47,41],"977":[46,14,48,10],"985":[46,14,48,10],"993":[46,14,48,10],"1001":[45,14,49,6],"1009":[52,20,52,27],"1017":[52,28,52,45],"1025":[52,20,52,46],"1033":[52,20,52,46],"1041":[52,13,52,47],"1049":[51,14,53,10],"1057":[51,14,53,10],"1065":[51,14,53,10],"1073":[50,20,54,6],"1081":[29,34,55,2],"1089":[29,1,55,3],"1091":[29,1,29,24],"1097":[29,1,55,4],"1105":[56,1,56,7],"1113":[56,18,56,25],"1121":[56,1,56,25],"1129":[56,1,56,26],"1137":[1,1,56,26],"1145":[1,1,56,26],"1153":[1,1,56,26],"1161":[1,1,56,26],"1169":[1,1,56,26],"1177":[1,1,56,26],"1185":[1,1,56,26],"1193":[1,1,56,26],"1201":[8,15,10,2],"1209":[8,15,10,2],"1217":[15,17,15,21],"1225":[16,14,16,30],"1233":[19,14,19,29],"1241":[22,14,22,29],"1249":[12,19,27,6],"1257":[12,19,27,6],"1265":[31,14,33,10],"1273":[31,14,33,10],"1281":[36,14,38,10],"1289":[36,14,38,10],"1297":[41,14,43,10],"1305":[41,14,43,10],"1313":[46,14,48,10],"1321":[46,14,48,10],"1329":[51,14,53,10],"1337":[51,14,53,10],"1345":[1,1,56,26],"1353":[1,1,56,26],"nBranches":10,"originalCodeFileName":"/home/blake/ExpoSE/Targets/excel-builder/node_modules/excel-builder/src/Excel/Drawing/index.js","instrumentedCodeFileName":"/home/blake/ExpoSE/Targets/excel-builder/node_modules/excel-builder/src/Excel/Drawing/index_jalangi_.js","code":"\"use strict\";\r\nvar _ = require('lodash');\r\nvar AbsoluteAnchor = require('./AbsoluteAnchor');\r\nvar OneCellAnchor = require('./OneCellAnchor');\r\nvar TwoCellAnchor = require('./TwoCellAnchor');\r\n\r\n/**\r\n * This is mostly a global spot where all of the relationship managers can get and set\r\n * path information from/to. \r\n * @module Excel/Drawing\r\n */\r\nvar Drawing = function () {\r\n    this.id = _.uniqueId('Drawing');\r\n};\r\n\r\n_.extend(Drawing.prototype, {\r\n    /**\r\n     *\r\n     * @param {String} type Can be 'absoluteAnchor', 'oneCellAnchor', or 'twoCellAnchor'.\r\n     * @param {Object} config Shorthand - pass the created anchor coords that can normally be used to construct it.\r\n     * @returns {Anchor}\r\n     */\r\n    createAnchor: function (type, config) {\r\n        config = config || {};\r\n        config.drawing = this;\r\n        switch(type) {\r\n            case 'absoluteAnchor':\r\n                this.anchor = new AbsoluteAnchor(config);\r\n                break;\r\n            case 'oneCellAnchor':\r\n                this.anchor = new OneCellAnchor(config);\r\n                break;\r\n            case 'twoCellAnchor':\r\n                this.anchor = new TwoCellAnchor(config);\r\n                break;\r\n        }\r\n        return this.anchor;\r\n    }\r\n});\r\n\r\nObject.defineProperties(Drawing, {\r\n    AbsoluteAnchor: {get: function () { return require('./AbsoluteAnchor'); }},\r\n    Chart: {get: function () { return require('./Chart'); }},\r\n    OneCellAnchor: {get: function () { return require('./OneCellAnchor'); }},\r\n    Picture: {get: function () { return require('./Picture'); }},\r\n    TwoCellAnchor: {get: function () { return require('./TwoCellAnchor'); }}\r\n});\r\n\r\nmodule.exports = Drawing;\r\n"};
jalangiLabel540:
    while (true) {
        try {
            J$.Se(1137, '/home/blake/ExpoSE/Targets/excel-builder/node_modules/excel-builder/src/Excel/Drawing/index_jalangi_.js', '/home/blake/ExpoSE/Targets/excel-builder/node_modules/excel-builder/src/Excel/Drawing/index.js');
            J$.N(1145, '__secret__S$', __secret__S$, 0);
            J$.N(1153, '__secret__traits__', __secret__traits__, 0);
            J$.N(1161, '_', _, 0);
            J$.N(1169, 'AbsoluteAnchor', AbsoluteAnchor, 0);
            J$.N(1177, 'OneCellAnchor', OneCellAnchor, 0);
            J$.N(1185, 'TwoCellAnchor', TwoCellAnchor, 0);
            J$.N(1193, 'Drawing', Drawing, 0);
            var __secret__S$ = J$.X1(41, J$.W(33, '__secret__S$', J$.F(25, J$.R(9, 'require', require, 2), 0)(J$.T(17, 'S$', 21, false)), __secret__S$, 3));
            var __secret__traits__ = J$.X1(73, J$.W(65, '__secret__traits__', J$.G(57, J$.R(49, '__secret__S$', __secret__S$, 1), 'Traits', 0), __secret__traits__, 3));
            J$.X1(89, J$.T(81, 'use strict', 21, false));
            var _ = J$.X1(129, J$.W(121, '_', J$.F(113, J$.R(97, 'require', require, 2), 0)(J$.T(105, 'lodash', 21, false)), _, 3));
            var AbsoluteAnchor = J$.X1(169, J$.W(161, 'AbsoluteAnchor', J$.F(153, J$.R(137, 'require', require, 2), 0)(J$.T(145, './AbsoluteAnchor', 21, false)), AbsoluteAnchor, 3));
            var OneCellAnchor = J$.X1(209, J$.W(201, 'OneCellAnchor', J$.F(193, J$.R(177, 'require', require, 2), 0)(J$.T(185, './OneCellAnchor', 21, false)), OneCellAnchor, 3));
            var TwoCellAnchor = J$.X1(249, J$.W(241, 'TwoCellAnchor', J$.F(233, J$.R(217, 'require', require, 2), 0)(J$.T(225, './TwoCellAnchor', 21, false)), TwoCellAnchor, 3));
            var Drawing = J$.X1(337, J$.W(329, 'Drawing', J$.T(321, function () {
                jalangiLabel533:
                    while (true) {
                        try {
                            J$.Fe(305, arguments.callee, this, arguments);
                            arguments = J$.N(313, 'arguments', arguments, 4);
                            J$.X1(297, J$.P(289, J$.R(257, 'this', this, 0), 'id', J$.M(281, J$.R(265, '_', _, 1), 'uniqueId', 0)(J$.T(273, 'Drawing', 21, false)), 0));
                        } catch (J$e) {
                            J$.Ex(1201, J$e);
                        } finally {
                            if (J$.Fr(1209))
                                continue jalangiLabel533;
                            else
                                return J$.Ra();
                        }
                    }
            }, 12, false, 305), Drawing, 3));
            J$.X1(697, J$.M(689, J$.R(345, '_', _, 1), 'extend', 0)(J$.G(361, J$.R(353, 'Drawing', Drawing, 1), 'prototype', 0), J$.T(681, {
                createAnchor: J$.T(673, function (type, config) {
                    jalangiLabel534:
                        while (true) {
                            try {
                                J$.Fe(641, arguments.callee, this, arguments);
                                arguments = J$.N(649, 'arguments', arguments, 4);
                                type = J$.N(657, 'type', type, 4);
                                config = J$.N(665, 'config', config, 4);
                                J$.X1(393, config = J$.W(385, 'config', J$.C(8, J$.R(369, 'config', config, 0)) ? J$._() : J$.T(377, {}, 11, false), config, 0));
                                J$.X1(425, J$.P(417, J$.R(401, 'config', config, 0), 'drawing', J$.R(409, 'this', this, 0), 0));
                                switch (J$.X1(1217, J$.C1(16, J$.R(433, 'type', type, 0)))) {
                                case J$.X1(1225, J$.C2(24, J$.T(489, 'absoluteAnchor', 21, false))):
                                    J$.X1(481, J$.P(473, J$.R(441, 'this', this, 0), 'anchor', J$.F(465, J$.R(449, 'AbsoluteAnchor', AbsoluteAnchor, 1), 1)(J$.R(457, 'config', config, 0)), 0));
                                    break;
                                case J$.X1(1233, J$.C2(32, J$.T(545, 'oneCellAnchor', 21, false))):
                                    J$.X1(537, J$.P(529, J$.R(497, 'this', this, 0), 'anchor', J$.F(521, J$.R(505, 'OneCellAnchor', OneCellAnchor, 1), 1)(J$.R(513, 'config', config, 0)), 0));
                                    break;
                                case J$.X1(1241, J$.C2(40, J$.T(601, 'twoCellAnchor', 21, false))):
                                    J$.X1(593, J$.P(585, J$.R(553, 'this', this, 0), 'anchor', J$.F(577, J$.R(561, 'TwoCellAnchor', TwoCellAnchor, 1), 1)(J$.R(569, 'config', config, 0)), 0));
                                    break;
                                }
                                return J$.X1(633, J$.Rt(625, J$.G(617, J$.R(609, 'this', this, 0), 'anchor', 0)));
                            } catch (J$e) {
                                J$.Ex(1249, J$e);
                            } finally {
                                if (J$.Fr(1257))
                                    continue jalangiLabel534;
                                else
                                    return J$.Ra();
                            }
                        }
                }, 12, false, 641)
            }, 11, false)));
            J$.X1(1097, J$.M(1089, J$.R(705, 'Object', Object, 2), 'defineProperties', 0)(J$.R(713, 'Drawing', Drawing, 1), J$.T(1081, {
                AbsoluteAnchor: J$.T(785, {
                    get: J$.T(777, function () {
                        jalangiLabel535:
                            while (true) {
                                try {
                                    J$.Fe(761, arguments.callee, this, arguments);
                                    arguments = J$.N(769, 'arguments', arguments, 4);
                                    return J$.X1(753, J$.Rt(745, J$.F(737, J$.R(721, 'require', require, 2), 0)(J$.T(729, './AbsoluteAnchor', 21, false))));
                                } catch (J$e) {
                                    J$.Ex(1265, J$e);
                                } finally {
                                    if (J$.Fr(1273))
                                        continue jalangiLabel535;
                                    else
                                        return J$.Ra();
                                }
                            }
                    }, 12, false, 761)
                }, 11, false),
                Chart: J$.T(857, {
                    get: J$.T(849, function () {
                        jalangiLabel536:
                            while (true) {
                                try {
                                    J$.Fe(833, arguments.callee, this, arguments);
                                    arguments = J$.N(841, 'arguments', arguments, 4);
                                    return J$.X1(825, J$.Rt(817, J$.F(809, J$.R(793, 'require', require, 2), 0)(J$.T(801, './Chart', 21, false))));
                                } catch (J$e) {
                                    J$.Ex(1281, J$e);
                                } finally {
                                    if (J$.Fr(1289))
                                        continue jalangiLabel536;
                                    else
                                        return J$.Ra();
                                }
                            }
                    }, 12, false, 833)
                }, 11, false),
                OneCellAnchor: J$.T(929, {
                    get: J$.T(921, function () {
                        jalangiLabel537:
                            while (true) {
                                try {
                                    J$.Fe(905, arguments.callee, this, arguments);
                                    arguments = J$.N(913, 'arguments', arguments, 4);
                                    return J$.X1(897, J$.Rt(889, J$.F(881, J$.R(865, 'require', require, 2), 0)(J$.T(873, './OneCellAnchor', 21, false))));
                                } catch (J$e) {
                                    J$.Ex(1297, J$e);
                                } finally {
                                    if (J$.Fr(1305))
                                        continue jalangiLabel537;
                                    else
                                        return J$.Ra();
                                }
                            }
                    }, 12, false, 905)
                }, 11, false),
                Picture: J$.T(1001, {
                    get: J$.T(993, function () {
                        jalangiLabel538:
                            while (true) {
                                try {
                                    J$.Fe(977, arguments.callee, this, arguments);
                                    arguments = J$.N(985, 'arguments', arguments, 4);
                                    return J$.X1(969, J$.Rt(961, J$.F(953, J$.R(937, 'require', require, 2), 0)(J$.T(945, './Picture', 21, false))));
                                } catch (J$e) {
                                    J$.Ex(1313, J$e);
                                } finally {
                                    if (J$.Fr(1321))
                                        continue jalangiLabel538;
                                    else
                                        return J$.Ra();
                                }
                            }
                    }, 12, false, 977)
                }, 11, false),
                TwoCellAnchor: J$.T(1073, {
                    get: J$.T(1065, function () {
                        jalangiLabel539:
                            while (true) {
                                try {
                                    J$.Fe(1049, arguments.callee, this, arguments);
                                    arguments = J$.N(1057, 'arguments', arguments, 4);
                                    return J$.X1(1041, J$.Rt(1033, J$.F(1025, J$.R(1009, 'require', require, 2), 0)(J$.T(1017, './TwoCellAnchor', 21, false))));
                                } catch (J$e) {
                                    J$.Ex(1329, J$e);
                                } finally {
                                    if (J$.Fr(1337))
                                        continue jalangiLabel539;
                                    else
                                        return J$.Ra();
                                }
                            }
                    }, 12, false, 1049)
                }, 11, false)
            }, 11, false)));
            J$.X1(1129, J$.P(1121, J$.R(1105, 'module', module, 2), 'exports', J$.R(1113, 'Drawing', Drawing, 1), 0));
        } catch (J$e) {
            J$.Ex(1345, J$e);
        } finally {
            if (J$.Sr(1353)) {
                J$.L();
                continue jalangiLabel540;
            } else {
                J$.L();
                break jalangiLabel540;
            }
        }
    }
// JALANGI DO NOT INSTRUMENT
